<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>14 Alpha Diversity | 3D’omics | Histomonas Turkeys</title>
  <meta name="description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="14 Alpha Diversity | 3D’omics | Histomonas Turkeys" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="github-repo" content="3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="14 Alpha Diversity | 3D’omics | Histomonas Turkeys" />
  
  <meta name="twitter:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  

<meta name="author" content="Amalia Bogri" />
<meta name="author" content="Jorge Langa" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2025-11-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bacterial-load.html"/>
<link rel="next" href="beta-diversity---macro-samples.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#environment"><i class="fa fa-check"></i><b>1.1.1</b> Environment</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#libraries"><i class="fa fa-check"></i><b>1.1.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#helper-functions"><i class="fa fa-check"></i><b>1.2</b> Helper Functions</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#alpha-diversity-calculation"><i class="fa fa-check"></i><b>1.2.1</b> Alpha Diversity Calculation</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#remove-samples-or-taxa"><i class="fa fa-check"></i><b>1.2.2</b> Remove samples or taxa</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#clr_function"><i class="fa fa-check"></i><b>1.2.3</b> clr_function</a></li>
<li class="chapter" data-level="1.2.4" data-path="index.html"><a href="index.html#perform_pca"><i class="fa fa-check"></i><b>1.2.4</b> perform_pca</a></li>
<li class="chapter" data-level="1.2.5" data-path="index.html"><a href="index.html#plot_pca"><i class="fa fa-check"></i><b>1.2.5</b> plot_pca</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notes"><i class="fa fa-check"></i><b>1.3</b> Notes:</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#installation-of-modules"><i class="fa fa-check"></i><b>1.3.1</b> Installation of modules</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#airtable-access"><i class="fa fa-check"></i><b>1.3.2</b> Airtable access:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>2</b> MAG Catalogue</a>
<ul>
<li class="chapter" data-level="2.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#dereplicated-genomes"><i class="fa fa-check"></i><b>2.1</b> Dereplicated genomes</a></li>
<li class="chapter" data-level="2.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-tree"><i class="fa fa-check"></i><b>2.2</b> Genome (MAGs) tree</a></li>
<li class="chapter" data-level="2.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-taxonomy"><i class="fa fa-check"></i><b>2.3</b> Genome (MAGs) taxonomy</a></li>
<li class="chapter" data-level="2.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-quality"><i class="fa fa-check"></i><b>2.4</b> Genome (MAGs) quality</a></li>
<li class="chapter" data-level="2.5" data-path="mag-catalogue.html"><a href="mag-catalogue.html#merged-genome-mags-taxonomy-quality"><i class="fa fa-check"></i><b>2.5</b> Merged Genome (MAGs) taxonomy &amp; quality</a></li>
<li class="chapter" data-level="2.6" data-path="mag-catalogue.html"><a href="mag-catalogue.html#factoring-of-taxonomic-levels"><i class="fa fa-check"></i><b>2.6</b> Factoring of taxonomic levels</a></li>
<li class="chapter" data-level="2.7" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-functional-annotations"><i class="fa fa-check"></i><b>2.7</b> Genome (MAGs) functional annotations</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#distil-functional-annotations"><i class="fa fa-check"></i><b>2.7.1</b> Distil functional annotations</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="mag-catalogue.html"><a href="mag-catalogue.html#bacterial-phyla-color-scheme-data"><i class="fa fa-check"></i><b>2.8</b> Bacterial phyla color scheme data</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-color-schemes-for-phyla-and-orders"><i class="fa fa-check"></i><b>2.8.1</b> Define color schemes for phyla and orders</a></li>
<li class="chapter" data-level="2.8.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-plotting-settings"><i class="fa fa-check"></i><b>2.8.2</b> Define plotting settings</a></li>
<li class="chapter" data-level="2.8.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-treatment-color-schemes"><i class="fa fa-check"></i><b>2.8.3</b> Define treatment color schemes</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="mag-catalogue.html"><a href="mag-catalogue.html#save-working-objects"><i class="fa fa-check"></i><b>2.9</b> Save working objects</a></li>
<li class="chapter" data-level="2.10" data-path="mag-catalogue.html"><a href="mag-catalogue.html#validation"><i class="fa fa-check"></i><b>2.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html"><i class="fa fa-check"></i><b>3</b> Micro-sample Metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#load-and-process-micro-sample-metadata-from-airtable"><i class="fa fa-check"></i><b>3.1</b> Load and process micro-sample metadata from Airtable</a></li>
<li class="chapter" data-level="3.2" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#save-working-objects-1"><i class="fa fa-check"></i><b>3.2</b> Save working objects</a></li>
<li class="chapter" data-level="3.3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#validation-1"><i class="fa fa-check"></i><b>3.3</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html"><i class="fa fa-check"></i><b>4</b> Micro-sample Counts</a>
<ul>
<li class="chapter" data-level="4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#load-required-data"><i class="fa fa-check"></i><b>4.1</b> Load required data</a></li>
<li class="chapter" data-level="4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#bacterial-reads-counts"><i class="fa fa-check"></i><b>4.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#genome-covered-bases"><i class="fa fa-check"></i><b>4.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="4.4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#filter-and-normalise-data"><i class="fa fa-check"></i><b>4.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#coverage-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Coverage filtering</a></li>
<li class="chapter" data-level="4.4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-genome-count-table"><i class="fa fa-check"></i><b>4.4.2</b> Generate genome count table</a></li>
<li class="chapter" data-level="4.4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-length-corrected-read-count-tables"><i class="fa fa-check"></i><b>4.4.3</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#alpha-diversity-calculations"><i class="fa fa-check"></i><b>4.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="4.6" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#save-working-objects-2"><i class="fa fa-check"></i><b>4.6</b> Save working objects</a></li>
<li class="chapter" data-level="4.7" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#validation-2"><i class="fa fa-check"></i><b>4.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>5</b> Micro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#load-required-data-1"><i class="fa fa-check"></i><b>5.1</b> Load required data</a></li>
<li class="chapter" data-level="5.2" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#fastp-sequencing-statistics"><i class="fa fa-check"></i><b>5.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="5.3" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas"><i class="fa fa-check"></i><b>5.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="5.4" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#bacterial-mapped-reads"><i class="fa fa-check"></i><b>5.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="5.5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table"><i class="fa fa-check"></i><b>5.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="5.6" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#sanity-checks"><i class="fa fa-check"></i><b>5.6</b> Sanity checks</a></li>
<li class="chapter" data-level="5.7" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics"><i class="fa fa-check"></i><b>5.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="5.8" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#create-plot_data_stats"><i class="fa fa-check"></i><b>5.8</b> Create plot_data_stats</a></li>
<li class="chapter" data-level="5.9" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#save-working-objects-3"><i class="fa fa-check"></i><b>5.9</b> Save working objects</a></li>
<li class="chapter" data-level="5.10" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#validation-3"><i class="fa fa-check"></i><b>5.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html"><i class="fa fa-check"></i><b>6</b> Macro-sample Metadata</a>
<ul>
<li class="chapter" data-level="6.1" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#save-working-objects-4"><i class="fa fa-check"></i><b>6.1</b> Save working objects</a></li>
<li class="chapter" data-level="6.2" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#validation-4"><i class="fa fa-check"></i><b>6.2</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html"><i class="fa fa-check"></i><b>7</b> Macro-sample Counts</a>
<ul>
<li class="chapter" data-level="7.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#load-required-data-2"><i class="fa fa-check"></i><b>7.1</b> Load required data</a></li>
<li class="chapter" data-level="7.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#bacterial-reads-counts-1"><i class="fa fa-check"></i><b>7.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="7.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#genome-covered-bases-1"><i class="fa fa-check"></i><b>7.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="7.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#filter-and-normalise-data-1"><i class="fa fa-check"></i><b>7.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-coverage-table"><i class="fa fa-check"></i><b>7.4.1</b> Generate coverage table</a></li>
<li class="chapter" data-level="7.4.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#coverage-filtering-1"><i class="fa fa-check"></i><b>7.4.2</b> Coverage filtering</a></li>
<li class="chapter" data-level="7.4.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-genome-count-table-1"><i class="fa fa-check"></i><b>7.4.3</b> Generate genome count table</a></li>
<li class="chapter" data-level="7.4.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-length-corrected-read-count-tables-1"><i class="fa fa-check"></i><b>7.4.4</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#alpha-diversity-calculations-1"><i class="fa fa-check"></i><b>7.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="7.6" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#save-working-objects-5"><i class="fa fa-check"></i><b>7.6</b> Save working objects</a></li>
<li class="chapter" data-level="7.7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#validation-5"><i class="fa fa-check"></i><b>7.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>8</b> Macro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#load-required-data-3"><i class="fa fa-check"></i><b>8.1</b> Load required data</a></li>
<li class="chapter" data-level="8.2" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#fastp-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="8.3" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas-1"><i class="fa fa-check"></i><b>8.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="8.4" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#bacterial-mapped-reads-1"><i class="fa fa-check"></i><b>8.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="8.5" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table-1"><i class="fa fa-check"></i><b>8.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="8.6" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#sanity-checks-1"><i class="fa fa-check"></i><b>8.6</b> Sanity checks</a></li>
<li class="chapter" data-level="8.7" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="8.8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#create-plot_data_stats_macro"><i class="fa fa-check"></i><b>8.8</b> Create plot_data_stats_macro</a></li>
<li class="chapter" data-level="8.9" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#save-working-objects-6"><i class="fa fa-check"></i><b>8.9</b> Save working objects</a></li>
<li class="chapter" data-level="8.10" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#validation-6"><i class="fa fa-check"></i><b>8.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html"><i class="fa fa-check"></i><b>9</b> Metabolomics Data Preparation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-metabolomics-data"><i class="fa fa-check"></i><b>9.1</b> Load metabolomics data</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-abundance-values-and-metabolites-metadata"><i class="fa fa-check"></i><b>9.1.1</b> Load abundance values and metabolites metadata</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#classify-samples-by-type"><i class="fa fa-check"></i><b>9.2</b> Classify samples by type</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#identify-sample-columns-and-classify-by-type"><i class="fa fa-check"></i><b>9.2.1</b> Identify sample columns and classify by type</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#create-dataframes-for-each-sample-type"><i class="fa fa-check"></i><b>9.3</b> Create dataframes for each sample type</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hn-histomonas-chicken-host-tissue"><i class="fa fa-check"></i><b>9.3.1</b> HN: Histomonas Chicken, host tissue</a></li>
<li class="chapter" data-level="9.3.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hm-histomonas-chicken-digesta"><i class="fa fa-check"></i><b>9.3.2</b> HM: Histomonas Chicken, digesta</a></li>
<li class="chapter" data-level="9.3.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mn-histomonas-turkey-host-tissue"><i class="fa fa-check"></i><b>9.3.3</b> MN: Histomonas Turkey, host tissue</a></li>
<li class="chapter" data-level="9.3.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mm-histomonas-turkey-digesta"><i class="fa fa-check"></i><b>9.3.4</b> MM: Histomonas Turkey, digesta</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-processed-data"><i class="fa fa-check"></i><b>9.4</b> Save processed data</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-metabolomics-dataframes"><i class="fa fa-check"></i><b>9.4.1</b> Save metabolomics dataframes</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validation-7"><i class="fa fa-check"></i><b>9.5</b> Validation</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validate-metabolomics-data"><i class="fa fa-check"></i><b>9.5.1</b> Validate metabolomics data</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#notes-on-metabolites-metadata"><i class="fa fa-check"></i><b>9.6</b> Notes on metabolites metadata</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#available-metadata-fields"><i class="fa fa-check"></i><b>9.6.1</b> Available metadata fields:</a></li>
<li class="chapter" data-level="9.6.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#suggested-uses"><i class="fa fa-check"></i><b>9.6.2</b> Suggested uses:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html"><i class="fa fa-check"></i><b>10</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-quality"><i class="fa fa-check"></i><b>10.1</b> Genome quality</a></li>
<li class="chapter" data-level="10.2" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours"><i class="fa fa-check"></i><b>10.2</b> MAG catalogue phyla and order colours</a></li>
<li class="chapter" data-level="10.3" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours-and-genera-number"><i class="fa fa-check"></i><b>10.3</b> MAG catalogue phyla and order colours, and genera number</a></li>
<li class="chapter" data-level="10.4" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot"><i class="fa fa-check"></i><b>10.4</b> Genome phylogeny plot</a></li>
<li class="chapter" data-level="10.5" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot-without-legend"><i class="fa fa-check"></i><b>10.5</b> Genome phylogeny plot without legend</a></li>
<li class="chapter" data-level="10.6" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-overview"><i class="fa fa-check"></i><b>10.6</b> Functional overview</a></li>
<li class="chapter" data-level="10.7" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-distances"><i class="fa fa-check"></i><b>10.7</b> Functional distances</a></li>
<li class="chapter" data-level="10.8" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-ordination"><i class="fa fa-check"></i><b>10.8</b> Functional ordination</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="histomonas-presence.html"><a href="histomonas-presence.html"><i class="fa fa-check"></i><b>11</b> Histomonas Presence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="histomonas-presence.html"><a href="histomonas-presence.html#macro"><i class="fa fa-check"></i><b>11.1</b> Macro</a></li>
<li class="chapter" data-level="11.2" data-path="histomonas-presence.html"><a href="histomonas-presence.html#micro"><i class="fa fa-check"></i><b>11.2</b> Micro</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html"><i class="fa fa-check"></i><b>12</b> Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#prepare-tidy-tables-for-plotting"><i class="fa fa-check"></i><b>12.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="12.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#macro-samples"><i class="fa fa-check"></i><b>12.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases-reads"><i class="fa fa-check"></i><b>12.2.1</b> Number of bases &amp; reads</a></li>
<li class="chapter" data-level="12.2.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped"><i class="fa fa-check"></i><b>12.2.2</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.2.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#composition-of-reads-vs.-references"><i class="fa fa-check"></i><b>12.2.3</b> Composition of reads vs. references</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#micro-samples"><i class="fa fa-check"></i><b>12.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases"><i class="fa fa-check"></i><b>12.3.1</b> Number of bases</a></li>
<li class="chapter" data-level="12.3.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-reads-after-trimming"><i class="fa fa-check"></i><b>12.3.2</b> Number of reads after trimming</a></li>
<li class="chapter" data-level="12.3.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped-1"><i class="fa fa-check"></i><b>12.3.3</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.3.4" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts"><i class="fa fa-check"></i><b>12.3.4</b> Collection attempts</a></li>
<li class="chapter" data-level="12.3.5" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts-vs-filtering-retention"><i class="fa fa-check"></i><b>12.3.5</b> Collection attempts vs filtering retention</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="bacterial-load.html"><a href="bacterial-load.html"><i class="fa fa-check"></i><b>13</b> Bacterial Load</a>
<ul>
<li class="chapter" data-level="13.1" data-path="bacterial-load.html"><a href="bacterial-load.html#macro-samples-1"><i class="fa fa-check"></i><b>13.1</b> Macro-samples</a></li>
<li class="chapter" data-level="13.2" data-path="bacterial-load.html"><a href="bacterial-load.html#micro-samples-1"><i class="fa fa-check"></i><b>13.2</b> Micro-samples</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="14.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#prepare-tidy-tables-for-plotting-1"><i class="fa fa-check"></i><b>14.1</b> Prepare tidy tables for plotting</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-tidy-tables"><i class="fa fa-check"></i><b>14.1.1</b> Alpha diversity tidy tables</a></li>
<li class="chapter" data-level="14.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#genome-counts-tidy-tables"><i class="fa fa-check"></i><b>14.1.2</b> Genome counts tidy tables</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#macro-samples-2"><i class="fa fa-check"></i><b>14.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div"><i class="fa fa-check"></i><b>14.2.1</b> Alpha-div</a></li>
<li class="chapter" data-level="14.2.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size"><i class="fa fa-check"></i><b>14.2.2</b> Alpha-div vs genome size</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#micro-samples-2"><i class="fa fa-check"></i><b>14.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-1"><i class="fa fa-check"></i><b>14.3.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="14.3.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size-1"><i class="fa fa-check"></i><b>14.3.2</b> Alpha-div vs genome size</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html"><i class="fa fa-check"></i><b>15</b> Beta Diversity - Macro Samples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#prepare-tidy-tables-for-plotting-2"><i class="fa fa-check"></i><b>15.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="15.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-level"><i class="fa fa-check"></i><b>15.2</b> Species level</a></li>
<li class="chapter" data-level="15.3" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#genus-level"><i class="fa fa-check"></i><b>15.3</b> Genus level</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#permanova"><i class="fa fa-check"></i><b>15.3.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="15.3.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-representative-genomes-overlap"><i class="fa fa-check"></i><b>15.3.2</b> Species representative genomes overlap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html"><i class="fa fa-check"></i><b>16</b> Beta Diversity - Micro Samples</a>
<ul>
<li class="chapter" data-level="16.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#prepare-tidy-tables-for-plotting-3"><i class="fa fa-check"></i><b>16.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="16.2" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#species-level-1"><i class="fa fa-check"></i><b>16.2</b> Species level</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#permanova-1"><i class="fa fa-check"></i><b>16.2.1</b> PERMANOVA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html"><i class="fa fa-check"></i><b>17</b> Beta Diversity - Macro vs Micro Comparison</a>
<ul>
<li class="chapter" data-level="17.1" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#genomes-overlap"><i class="fa fa-check"></i><b>17.1</b> Genomes overlap</a></li>
<li class="chapter" data-level="17.2" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#micro-vs-macro-genus-level"><i class="fa fa-check"></i><b>17.2</b> Micro vs Macro Genus level</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html"><i class="fa fa-check"></i><b>18</b> Spatial Analysis - Micro Samples</a>
<ul>
<li class="chapter" data-level="18.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#functions"><i class="fa fa-check"></i><b>18.1</b> Functions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#spatial_cryosections"><i class="fa fa-check"></i><b>18.1.1</b> spatial_cryosections</a></li>
<li class="chapter" data-level="18.1.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#additive-diversity-partitioning"><i class="fa fa-check"></i><b>18.1.2</b> Additive diversity partitioning</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#load-data"><i class="fa fa-check"></i><b>18.2</b> Load data</a></li>
<li class="chapter" data-level="18.3" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#extended-rlq-analysis"><i class="fa fa-check"></i><b>18.3</b> Extended RLQ analysis</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html"><i class="fa fa-check"></i><b>19</b> Export Data for MicrobeTag</a>
<ul>
<li class="chapter" data-level="19.1" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#load-required-data-4"><i class="fa fa-check"></i><b>19.1</b> Load required data</a></li>
<li class="chapter" data-level="19.2" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-sample-metadata-macro-samples"><i class="fa fa-check"></i><b>19.2</b> Export sample metadata (macro samples)</a></li>
<li class="chapter" data-level="19.3" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-metabolites-data-mm-samples"><i class="fa fa-check"></i><b>19.3</b> Export metabolites data (MM samples)</a></li>
<li class="chapter" data-level="19.4" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-metadata"><i class="fa fa-check"></i><b>19.4</b> Export genome metadata</a></li>
<li class="chapter" data-level="19.5" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-counts-macro-samples-filtered"><i class="fa fa-check"></i><b>19.5</b> Export genome counts (macro samples, filtered)</a></li>
<li class="chapter" data-level="19.6" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#summary"><i class="fa fa-check"></i><b>19.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-versions.html"><a href="package-versions.html"><i class="fa fa-check"></i><b>20</b> Package Versions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">3D’omics | Histomonas Turkeys</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="alpha-diversity" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">14</span> Alpha Diversity<a href="alpha-diversity.html#alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb194"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb194-1"><a href="alpha-diversity.html#cb194-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/macro/sample_metadata.Rdata&quot;</span>)</span>
<span id="cb194-2"><a href="alpha-diversity.html#cb194-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/macro/counts.Rdata&quot;</span>)</span>
<span id="cb194-3"><a href="alpha-diversity.html#cb194-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/micro/sample_metadata.Rdata&quot;</span>)</span>
<span id="cb194-4"><a href="alpha-diversity.html#cb194-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/micro/counts.Rdata&quot;</span>)</span>
<span id="cb194-5"><a href="alpha-diversity.html#cb194-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/MAG_catalogue/data.Rdata&quot;</span>)</span>
<span id="cb194-6"><a href="alpha-diversity.html#cb194-6" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data_colors.Rdata&quot;</span>)</span></code></pre></div>
<div id="prepare-tidy-tables-for-plotting-1" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> Prepare tidy tables for plotting<a href="alpha-diversity.html#prepare-tidy-tables-for-plotting-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="alpha-diversity-tidy-tables" class="section level3 hasAnchor" number="14.1.1">
<h3><span class="header-section-number">14.1.1</span> Alpha diversity tidy tables<a href="alpha-diversity.html#alpha-diversity-tidy-tables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create tidy table for alpha diversity (micro)</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb195-1"><a href="alpha-diversity.html#cb195-1" tabindex="-1"></a>tidy_plot_alpha_diversity_genomes <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb195-2"><a href="alpha-diversity.html#cb195-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb195-3"><a href="alpha-diversity.html#cb195-3" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(richness, neutral, phylogenetic), </span>
<span id="cb195-4"><a href="alpha-diversity.html#cb195-4" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>,                      </span>
<span id="cb195-5"><a href="alpha-diversity.html#cb195-5" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>                       </span>
<span id="cb195-6"><a href="alpha-diversity.html#cb195-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb195-7"><a href="alpha-diversity.html#cb195-7" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">factor</span>(metric, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;richness&quot;</span>, <span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;phylogenetic&quot;</span>)))</span></code></pre></div>
<p>Create tidy table for alpha diversity (macro)</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb196-1"><a href="alpha-diversity.html#cb196-1" tabindex="-1"></a>tidy_plot_alpha_diversity_genomes_macro <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb196-2"><a href="alpha-diversity.html#cb196-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb196-3"><a href="alpha-diversity.html#cb196-3" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(richness, neutral, phylogenetic), </span>
<span id="cb196-4"><a href="alpha-diversity.html#cb196-4" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>,                      </span>
<span id="cb196-5"><a href="alpha-diversity.html#cb196-5" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>                       </span>
<span id="cb196-6"><a href="alpha-diversity.html#cb196-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb196-7"><a href="alpha-diversity.html#cb196-7" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">factor</span>(metric, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;richness&quot;</span>, <span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;phylogenetic&quot;</span>)))</span></code></pre></div>
</div>
<div id="genome-counts-tidy-tables" class="section level3 hasAnchor" number="14.1.2">
<h3><span class="header-section-number">14.1.2</span> Genome counts tidy tables<a href="alpha-diversity.html#genome-counts-tidy-tables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create tidy tables for filtered genome counts (micro)</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb197-1"><a href="alpha-diversity.html#cb197-1" tabindex="-1"></a>tidy_plot_genome_counts_filt_30_closed <span class="ot">&lt;-</span>  genome_counts_filt_30 <span class="sc">%&gt;%</span></span>
<span id="cb197-2"><a href="alpha-diversity.html#cb197-2" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome), <span class="sc">~</span> . <span class="sc">/</span> <span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb197-3"><a href="alpha-diversity.html#cb197-3" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;microsample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb197-4"><a href="alpha-diversity.html#cb197-4" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> </span>
<span id="cb197-5"><a href="alpha-diversity.html#cb197-5" tabindex="-1"></a>    <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span> </span>
<span id="cb197-6"><a href="alpha-diversity.html#cb197-6" tabindex="-1"></a>    <span class="fu">left_join</span>(final_combined_stats, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span></span>
<span id="cb197-7"><a href="alpha-diversity.html#cb197-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">section =</span> <span class="fu">unlist</span>(section)) <span class="sc">%&gt;%</span> </span>
<span id="cb197-8"><a href="alpha-diversity.html#cb197-8" tabindex="-1"></a>    <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb197-9"><a href="alpha-diversity.html#cb197-9" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> phylum_level_vector),</span>
<span id="cb197-10"><a href="alpha-diversity.html#cb197-10" tabindex="-1"></a>           <span class="at">order =</span> <span class="fu">factor</span>(order, <span class="at">levels =</span> order_level_vector))</span></code></pre></div>
<p>Create tidy tables for filtered genome counts (macro)</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb198-1"><a href="alpha-diversity.html#cb198-1" tabindex="-1"></a>tidy_plot_genome_counts_filt_30_macro_closed <span class="ot">&lt;-</span>  genome_counts_macro_filt_30 <span class="sc">%&gt;%</span></span>
<span id="cb198-2"><a href="alpha-diversity.html#cb198-2" tabindex="-1"></a>    <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome), <span class="sc">~</span> . <span class="sc">/</span> <span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb198-3"><a href="alpha-diversity.html#cb198-3" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;microsample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb198-4"><a href="alpha-diversity.html#cb198-4" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> </span>
<span id="cb198-5"><a href="alpha-diversity.html#cb198-5" tabindex="-1"></a>    <span class="fu">left_join</span>(sample_metadata_macro, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb198-6"><a href="alpha-diversity.html#cb198-6" tabindex="-1"></a>    <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb198-7"><a href="alpha-diversity.html#cb198-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> phylum_level_vector),</span>
<span id="cb198-8"><a href="alpha-diversity.html#cb198-8" tabindex="-1"></a>           <span class="at">order =</span> <span class="fu">factor</span>(order, <span class="at">levels =</span> order_level_vector))</span></code></pre></div>
</div>
</div>
<div id="macro-samples-2" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Macro-samples<a href="alpha-diversity.html#macro-samples-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="alpha-div" class="section level3 hasAnchor" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Alpha-div<a href="alpha-diversity.html#alpha-div" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb199"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb199-1"><a href="alpha-diversity.html#cb199-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;value&quot;</span></span>
<span id="cb199-2"><a href="alpha-diversity.html#cb199-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> tidy_plot_alpha_diversity_genomes_macro <span class="sc">%&gt;%</span></span>
<span id="cb199-3"><a href="alpha-diversity.html#cb199-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb199-4"><a href="alpha-diversity.html#cb199-4" tabindex="-1"></a>  <span class="co"># filter(filter_level == &#39;filtered_30&#39;) %&gt;%</span></span>
<span id="cb199-5"><a href="alpha-diversity.html#cb199-5" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># to remove empty samples</span></span>
<span id="cb199-6"><a href="alpha-diversity.html#cb199-6" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment_expl, age_category, metric) <span class="sc">%&gt;%</span></span>
<span id="cb199-7"><a href="alpha-diversity.html#cb199-7" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb199-8"><a href="alpha-diversity.html#cb199-8" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb199-9"><a href="alpha-diversity.html#cb199-9" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb199-10"><a href="alpha-diversity.html#cb199-10" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb199-11"><a href="alpha-diversity.html#cb199-11" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb199-12"><a href="alpha-diversity.html#cb199-12" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb199-13"><a href="alpha-diversity.html#cb199-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">formatC</span>(median, <span class="at">format =</span> <span class="st">&quot;f&quot;</span>, <span class="at">digits =</span> <span class="dv">1</span>)) <span class="co">#format(median, scientific = TRUE, digits = 5))</span></span>
<span id="cb199-14"><a href="alpha-diversity.html#cb199-14" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table>
<colgroup>
<col width="26%" />
<col width="11%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">treatment_expl</th>
<th align="left">age_category</th>
<th align="left">metric</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="right">210.000000</td>
<td align="right">98.0000000</td>
<td align="right">138.000000</td>
<td align="right">236.000000</td>
<td align="left">210.0</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="right">83.811909</td>
<td align="right">32.8752555</td>
<td align="right">59.397392</td>
<td align="right">92.272648</td>
<td align="left">83.8</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="right">6.696391</td>
<td align="right">0.7547965</td>
<td align="right">6.258387</td>
<td align="right">7.013183</td>
<td align="left">6.7</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">b</td>
<td align="left">richness</td>
<td align="right">262.000000</td>
<td align="right">103.0000000</td>
<td align="right">175.000000</td>
<td align="right">278.000000</td>
<td align="left">262.0</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">b</td>
<td align="left">neutral</td>
<td align="right">85.913403</td>
<td align="right">17.7706682</td>
<td align="right">76.580196</td>
<td align="right">94.350864</td>
<td align="left">85.9</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">b</td>
<td align="left">phylogenetic</td>
<td align="right">7.801862</td>
<td align="right">1.6404552</td>
<td align="right">7.433081</td>
<td align="right">9.073536</td>
<td align="left">7.8</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">c</td>
<td align="left">richness</td>
<td align="right">225.000000</td>
<td align="right">42.7500000</td>
<td align="right">210.750000</td>
<td align="right">253.500000</td>
<td align="left">225.0</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">c</td>
<td align="left">neutral</td>
<td align="right">79.011578</td>
<td align="right">11.0493141</td>
<td align="right">74.089534</td>
<td align="right">85.138848</td>
<td align="left">79.0</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">c</td>
<td align="left">phylogenetic</td>
<td align="right">7.686419</td>
<td align="right">0.5045590</td>
<td align="right">7.324135</td>
<td align="right">7.828694</td>
<td align="left">7.7</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="right">212.000000</td>
<td align="right">18.0000000</td>
<td align="right">205.250000</td>
<td align="right">223.250000</td>
<td align="left">212.0</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="right">77.070687</td>
<td align="right">20.1219134</td>
<td align="right">67.456909</td>
<td align="right">87.578822</td>
<td align="left">77.1</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="right">7.630725</td>
<td align="right">0.5222967</td>
<td align="right">7.222578</td>
<td align="right">7.744874</td>
<td align="left">7.6</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="right">213.000000</td>
<td align="right">29.0000000</td>
<td align="right">198.000000</td>
<td align="right">227.000000</td>
<td align="left">213.0</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="right">92.572615</td>
<td align="right">12.2730843</td>
<td align="right">86.960644</td>
<td align="right">99.233728</td>
<td align="left">92.6</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="right">6.284060</td>
<td align="right">0.6118000</td>
<td align="right">5.889814</td>
<td align="right">6.501614</td>
<td align="left">6.3</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">b</td>
<td align="left">richness</td>
<td align="right">10.000000</td>
<td align="right">8.0000000</td>
<td align="right">4.000000</td>
<td align="right">12.000000</td>
<td align="left">10.0</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">b</td>
<td align="left">neutral</td>
<td align="right">6.828946</td>
<td align="right">6.1434774</td>
<td align="right">3.533646</td>
<td align="right">9.677124</td>
<td align="left">6.8</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">b</td>
<td align="left">phylogenetic</td>
<td align="right">4.036856</td>
<td align="right">0.7414506</td>
<td align="right">3.582019</td>
<td align="right">4.323469</td>
<td align="left">4.0</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">c</td>
<td align="left">richness</td>
<td align="right">25.000000</td>
<td align="right">15.5000000</td>
<td align="right">17.500000</td>
<td align="right">33.000000</td>
<td align="left">25.0</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">c</td>
<td align="left">neutral</td>
<td align="right">9.820644</td>
<td align="right">5.0179333</td>
<td align="right">7.941182</td>
<td align="right">12.959115</td>
<td align="left">9.8</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">c</td>
<td align="left">phylogenetic</td>
<td align="right">4.084129</td>
<td align="right">1.1346151</td>
<td align="right">3.609400</td>
<td align="right">4.744015</td>
<td align="left">4.1</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="right">178.500000</td>
<td align="right">84.0000000</td>
<td align="right">143.500000</td>
<td align="right">227.500000</td>
<td align="left">178.5</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="right">73.283154</td>
<td align="right">9.7468985</td>
<td align="right">66.995456</td>
<td align="right">76.742355</td>
<td align="left">73.3</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="right">5.584873</td>
<td align="right">0.7026976</td>
<td align="right">5.141251</td>
<td align="right">5.843949</td>
<td align="left">5.6</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">richness</td>
<td align="right">9.000000</td>
<td align="right">11.0000000</td>
<td align="right">6.000000</td>
<td align="right">17.000000</td>
<td align="left">9.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">neutral</td>
<td align="right">4.805473</td>
<td align="right">8.9287820</td>
<td align="right">2.829573</td>
<td align="right">11.758355</td>
<td align="left">4.8</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">phylogenetic</td>
<td align="right">3.877432</td>
<td align="right">1.7102515</td>
<td align="right">2.580326</td>
<td align="right">4.290578</td>
<td align="left">3.9</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">richness</td>
<td align="right">61.000000</td>
<td align="right">93.0000000</td>
<td align="right">37.000000</td>
<td align="right">130.000000</td>
<td align="left">61.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">neutral</td>
<td align="right">19.031655</td>
<td align="right">16.5854099</td>
<td align="right">15.355569</td>
<td align="right">31.940978</td>
<td align="left">19.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">phylogenetic</td>
<td align="right">4.742644</td>
<td align="right">1.1812226</td>
<td align="right">4.708435</td>
<td align="right">5.889657</td>
<td align="left">4.7</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="right">203.000000</td>
<td align="right">107.0000000</td>
<td align="right">134.250000</td>
<td align="right">241.250000</td>
<td align="left">203.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="right">72.419728</td>
<td align="right">38.9935100</td>
<td align="right">48.231406</td>
<td align="right">87.224916</td>
<td align="left">72.4</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="right">6.672006</td>
<td align="right">1.3006379</td>
<td align="right">5.902491</td>
<td align="right">7.203129</td>
<td align="left">6.7</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="right">229.500000</td>
<td align="right">93.0000000</td>
<td align="right">182.250000</td>
<td align="right">275.250000</td>
<td align="left">229.5</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="right">97.656057</td>
<td align="right">45.8913381</td>
<td align="right">74.102555</td>
<td align="right">119.993893</td>
<td align="left">97.7</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="right">6.598823</td>
<td align="right">0.3352976</td>
<td align="right">6.366943</td>
<td align="right">6.702240</td>
<td align="left">6.6</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">richness</td>
<td align="right">24.000000</td>
<td align="right">43.5000000</td>
<td align="right">11.500000</td>
<td align="right">55.000000</td>
<td align="left">24.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">neutral</td>
<td align="right">15.375341</td>
<td align="right">21.4031885</td>
<td align="right">6.951692</td>
<td align="right">28.354880</td>
<td align="left">15.4</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">b</td>
<td align="left">phylogenetic</td>
<td align="right">4.971620</td>
<td align="right">1.3565209</td>
<td align="right">4.374426</td>
<td align="right">5.730947</td>
<td align="left">5.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">richness</td>
<td align="right">107.000000</td>
<td align="right">77.0000000</td>
<td align="right">50.000000</td>
<td align="right">127.000000</td>
<td align="left">107.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">neutral</td>
<td align="right">25.883164</td>
<td align="right">4.3196914</td>
<td align="right">22.355050</td>
<td align="right">26.674741</td>
<td align="left">25.9</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">c</td>
<td align="left">phylogenetic</td>
<td align="right">5.519981</td>
<td align="right">0.8902031</td>
<td align="right">4.972761</td>
<td align="right">5.862964</td>
<td align="left">5.5</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="right">147.000000</td>
<td align="right">125.0000000</td>
<td align="right">61.000000</td>
<td align="right">186.000000</td>
<td align="left">147.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="right">47.675856</td>
<td align="right">38.8252638</td>
<td align="right">9.642039</td>
<td align="right">48.467303</td>
<td align="left">47.7</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="right">6.695277</td>
<td align="right">4.1891537</td>
<td align="right">2.902256</td>
<td align="right">7.091409</td>
<td align="left">6.7</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb200"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb200-1"><a href="alpha-diversity.html#cb200-1" tabindex="-1"></a>plot_alpha_div_macro <span class="ot">&lt;-</span> tidy_plot_alpha_diversity_genomes_macro <span class="sc">%&gt;%</span></span>
<span id="cb200-2"><a href="alpha-diversity.html#cb200-2" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb200-3"><a href="alpha-diversity.html#cb200-3" tabindex="-1"></a>  <span class="fu">filter</span>(filter_level <span class="sc">==</span> <span class="st">&#39;filtered_30&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb200-4"><a href="alpha-diversity.html#cb200-4" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># to remove empty samples</span></span>
<span id="cb200-5"><a href="alpha-diversity.html#cb200-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> age_category, <span class="at">y =</span> value, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb200-6"><a href="alpha-diversity.html#cb200-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb200-7"><a href="alpha-diversity.html#cb200-7" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb200-8"><a href="alpha-diversity.html#cb200-8" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">names</span>(treatment_colours_bright),</span>
<span id="cb200-9"><a href="alpha-diversity.html#cb200-9" tabindex="-1"></a>                     <span class="at">values =</span> treatment_colours_bright) <span class="sc">+</span></span>
<span id="cb200-10"><a href="alpha-diversity.html#cb200-10" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb200-11"><a href="alpha-diversity.html#cb200-11" tabindex="-1"></a>  <span class="fu">facet_nested</span>(metric <span class="sc">~</span> treatment_expl, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb200-12"><a href="alpha-diversity.html#cb200-12" tabindex="-1"></a>  <span class="co"># geom_text(aes(label = animal), </span></span>
<span id="cb200-13"><a href="alpha-diversity.html#cb200-13" tabindex="-1"></a>  <span class="co">#           position = position_jitter(width = 0.2, height = 0), </span></span>
<span id="cb200-14"><a href="alpha-diversity.html#cb200-14" tabindex="-1"></a>  <span class="co">#           size = 3.0, alpha = 0.7, check_overlap = TRUE, color=&quot;#8c1c47&quot;) +</span></span>
<span id="cb200-15"><a href="alpha-diversity.html#cb200-15" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Days Post Infection&quot;</span>, <span class="at">y =</span> <span class="st">&quot;a-div value&quot;</span>) <span class="sc">+</span></span>
<span id="cb200-16"><a href="alpha-diversity.html#cb200-16" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.15</span>))) <span class="sc">+</span></span>
<span id="cb200-17"><a href="alpha-diversity.html#cb200-17" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb200-18"><a href="alpha-diversity.html#cb200-18" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb200-19"><a href="alpha-diversity.html#cb200-19" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> age_category, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb200-20"><a href="alpha-diversity.html#cb200-20" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb200-21"><a href="alpha-diversity.html#cb200-21" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">3.0</span>,</span>
<span id="cb200-22"><a href="alpha-diversity.html#cb200-22" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb200-23"><a href="alpha-diversity.html#cb200-23" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb200-24"><a href="alpha-diversity.html#cb200-24" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb200-25"><a href="alpha-diversity.html#cb200-25" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb200-26"><a href="alpha-diversity.html#cb200-26" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb200-27"><a href="alpha-diversity.html#cb200-27" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb200-28"><a href="alpha-diversity.html#cb200-28" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Alpha diversity - 30% coverage filter&quot;</span>)</span>
<span id="cb200-29"><a href="alpha-diversity.html#cb200-29" tabindex="-1"></a>plot_alpha_div_macro</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_alpha_div_macro_main-1.png" width="960" /></p>
</div>
<div id="alpha-div-vs-genome-size" class="section level3 hasAnchor" number="14.2.2">
<h3><span class="header-section-number">14.2.2</span> Alpha-div vs genome size<a href="alpha-diversity.html#alpha-div-vs-genome-size" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb201"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb201-1"><a href="alpha-diversity.html#cb201-1" tabindex="-1"></a>tidy_plot_genome_counts_filt_30_macro_closed_zerosrem <span class="ot">&lt;-</span> tidy_plot_genome_counts_filt_30_macro_closed <span class="sc">%&gt;%</span></span>
<span id="cb201-2"><a href="alpha-diversity.html#cb201-2" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb201-3"><a href="alpha-diversity.html#cb201-3" tabindex="-1"></a></span>
<span id="cb201-4"><a href="alpha-diversity.html#cb201-4" tabindex="-1"></a>median_lengths <span class="ot">&lt;-</span> tidy_plot_genome_counts_filt_30_macro_closed_zerosrem <span class="sc">%&gt;%</span></span>
<span id="cb201-5"><a href="alpha-diversity.html#cb201-5" tabindex="-1"></a>  <span class="fu">group_by</span>(microsample) <span class="sc">%&gt;%</span></span>
<span id="cb201-6"><a href="alpha-diversity.html#cb201-6" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_genome_length =</span> <span class="fu">median</span>(length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb201-7"><a href="alpha-diversity.html#cb201-7" tabindex="-1"></a></span>
<span id="cb201-8"><a href="alpha-diversity.html#cb201-8" tabindex="-1"></a>alpha_div_macro_filtered_30_gen_length <span class="ot">&lt;-</span> alpha_div_macro_filtered_30 <span class="sc">%&gt;%</span></span>
<span id="cb201-9"><a href="alpha-diversity.html#cb201-9" tabindex="-1"></a>  <span class="fu">left_join</span>(median_lengths, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span></span>
<span id="cb201-10"><a href="alpha-diversity.html#cb201-10" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata_macro, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb201-11"><a href="alpha-diversity.html#cb201-11" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">!=</span> <span class="st">&quot;TM0&quot;</span>) </span>
<span id="cb201-12"><a href="alpha-diversity.html#cb201-12" tabindex="-1"></a></span>
<span id="cb201-13"><a href="alpha-diversity.html#cb201-13" tabindex="-1"></a></span>
<span id="cb201-14"><a href="alpha-diversity.html#cb201-14" tabindex="-1"></a>alpha_div_macro_filtered_30_gen_length <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb201-15"><a href="alpha-diversity.html#cb201-15" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb201-16"><a href="alpha-diversity.html#cb201-16" tabindex="-1"></a>  <span class="fu">filter</span>(filter_level <span class="sc">==</span> <span class="st">&quot;filtered_30&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb201-17"><a href="alpha-diversity.html#cb201-17" tabindex="-1"></a>  <span class="fu">left_join</span>(median_lengths, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span></span>
<span id="cb201-18"><a href="alpha-diversity.html#cb201-18" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">!=</span> <span class="st">&quot;TM0&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb201-19"><a href="alpha-diversity.html#cb201-19" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb202"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb202-1"><a href="alpha-diversity.html#cb202-1" tabindex="-1"></a><span class="co"># alpha_div_macro_filtered_30_gen_length %&gt;%</span></span>
<span id="cb202-2"><a href="alpha-diversity.html#cb202-2" tabindex="-1"></a><span class="co">#   ggplot(aes(x = median_genome_length, y = richness, color = treatment, shape = factor(age_category))) +</span></span>
<span id="cb202-3"><a href="alpha-diversity.html#cb202-3" tabindex="-1"></a><span class="co">#   geom_point(size = 3, alpha = 0.8) +</span></span>
<span id="cb202-4"><a href="alpha-diversity.html#cb202-4" tabindex="-1"></a><span class="co">#   geom_smooth(method = &quot;lm&quot;, se = TRUE, aes(group = 1), color = &quot;black&quot;, linewidth = 1) +</span></span>
<span id="cb202-5"><a href="alpha-diversity.html#cb202-5" tabindex="-1"></a><span class="co">#   scale_color_manual(values = treatment_colours_bright, name = &quot;Treatment&quot;) +</span></span>
<span id="cb202-6"><a href="alpha-diversity.html#cb202-6" tabindex="-1"></a><span class="co">#   scale_shape_manual(</span></span>
<span id="cb202-7"><a href="alpha-diversity.html#cb202-7" tabindex="-1"></a><span class="co">#     name = &quot;DPI&quot;,</span></span>
<span id="cb202-8"><a href="alpha-diversity.html#cb202-8" tabindex="-1"></a><span class="co">#     values = c(&quot;a&quot; = 16, &quot;b&quot; = 17, &quot;c&quot; = 15, &quot;d&quot; = 4)  # Example shapes</span></span>
<span id="cb202-9"><a href="alpha-diversity.html#cb202-9" tabindex="-1"></a><span class="co">#   ) +</span></span>
<span id="cb202-10"><a href="alpha-diversity.html#cb202-10" tabindex="-1"></a><span class="co">#   labs(x = &quot;Median genome length (bp) per sample&quot;, y = &quot;Richness&quot;) +</span></span>
<span id="cb202-11"><a href="alpha-diversity.html#cb202-11" tabindex="-1"></a><span class="co">#   ylim(c(-5,300)) +</span></span>
<span id="cb202-12"><a href="alpha-diversity.html#cb202-12" tabindex="-1"></a><span class="co">#   theme_minimal() +</span></span>
<span id="cb202-13"><a href="alpha-diversity.html#cb202-13" tabindex="-1"></a><span class="co">#   custom_ggplot_theme </span></span></code></pre></div>
<div class="sourceCode" id="cb203"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb203-1"><a href="alpha-diversity.html#cb203-1" tabindex="-1"></a><span class="co"># Fit model</span></span>
<span id="cb203-2"><a href="alpha-diversity.html#cb203-2" tabindex="-1"></a>model_nb <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(richness <span class="sc">~</span> median_genome_length, </span>
<span id="cb203-3"><a href="alpha-diversity.html#cb203-3" tabindex="-1"></a>                   <span class="at">data =</span> alpha_div_macro_filtered_30_gen_length)</span>
<span id="cb203-4"><a href="alpha-diversity.html#cb203-4" tabindex="-1"></a></span>
<span id="cb203-5"><a href="alpha-diversity.html#cb203-5" tabindex="-1"></a><span class="co"># Fit null model (intercept only)</span></span>
<span id="cb203-6"><a href="alpha-diversity.html#cb203-6" tabindex="-1"></a>model_null <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(richness <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb203-7"><a href="alpha-diversity.html#cb203-7" tabindex="-1"></a>                     <span class="at">data =</span> alpha_div_macro_filtered_30_gen_length)</span>
<span id="cb203-8"><a href="alpha-diversity.html#cb203-8" tabindex="-1"></a></span>
<span id="cb203-9"><a href="alpha-diversity.html#cb203-9" tabindex="-1"></a><span class="co"># Manual McFadden pseudo-R²</span></span>
<span id="cb203-10"><a href="alpha-diversity.html#cb203-10" tabindex="-1"></a>pseudo_r2 <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> (<span class="fu">logLik</span>(model_nb) <span class="sc">/</span> <span class="fu">logLik</span>(model_null))</span>
<span id="cb203-11"><a href="alpha-diversity.html#cb203-11" tabindex="-1"></a>pseudo_r2</span></code></pre></div>
<pre class="script-output"><code>&#39;log Lik.&#39; 0.1045218 (df=3)</code></pre>
<div class="sourceCode" id="cb205"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb205-1"><a href="alpha-diversity.html#cb205-1" tabindex="-1"></a><span class="co"># Spearman correlation</span></span>
<span id="cb205-2"><a href="alpha-diversity.html#cb205-2" tabindex="-1"></a>cor_test <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb205-3"><a href="alpha-diversity.html#cb205-3" tabindex="-1"></a>  <span class="sc">~</span> richness <span class="sc">+</span> median_genome_length,</span>
<span id="cb205-4"><a href="alpha-diversity.html#cb205-4" tabindex="-1"></a>  <span class="at">data =</span> alpha_div_macro_filtered_30_gen_length,</span>
<span id="cb205-5"><a href="alpha-diversity.html#cb205-5" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span></span>
<span id="cb205-6"><a href="alpha-diversity.html#cb205-6" tabindex="-1"></a>)</span>
<span id="cb205-7"><a href="alpha-diversity.html#cb205-7" tabindex="-1"></a></span>
<span id="cb205-8"><a href="alpha-diversity.html#cb205-8" tabindex="-1"></a>rho  <span class="ot">&lt;-</span> cor_test<span class="sc">$</span>estimate</span>
<span id="cb205-9"><a href="alpha-diversity.html#cb205-9" tabindex="-1"></a>pval <span class="ot">&lt;-</span> cor_test<span class="sc">$</span>p.value</span>
<span id="cb205-10"><a href="alpha-diversity.html#cb205-10" tabindex="-1"></a></span>
<span id="cb205-11"><a href="alpha-diversity.html#cb205-11" tabindex="-1"></a></span>
<span id="cb205-12"><a href="alpha-diversity.html#cb205-12" tabindex="-1"></a><span class="co"># Create new data for prediction</span></span>
<span id="cb205-13"><a href="alpha-diversity.html#cb205-13" tabindex="-1"></a>newdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb205-14"><a href="alpha-diversity.html#cb205-14" tabindex="-1"></a>  <span class="at">median_genome_length =</span> <span class="fu">seq</span>(</span>
<span id="cb205-15"><a href="alpha-diversity.html#cb205-15" tabindex="-1"></a>    <span class="fu">min</span>(alpha_div_macro_filtered_30_gen_length<span class="sc">$</span>median_genome_length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb205-16"><a href="alpha-diversity.html#cb205-16" tabindex="-1"></a>    <span class="fu">max</span>(alpha_div_macro_filtered_30_gen_length<span class="sc">$</span>median_genome_length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb205-17"><a href="alpha-diversity.html#cb205-17" tabindex="-1"></a>    <span class="at">length.out =</span> <span class="dv">200</span></span>
<span id="cb205-18"><a href="alpha-diversity.html#cb205-18" tabindex="-1"></a>  )</span>
<span id="cb205-19"><a href="alpha-diversity.html#cb205-19" tabindex="-1"></a>)</span>
<span id="cb205-20"><a href="alpha-diversity.html#cb205-20" tabindex="-1"></a><span class="co"># Get predictions + SE on link scale</span></span>
<span id="cb205-21"><a href="alpha-diversity.html#cb205-21" tabindex="-1"></a>pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(model_nb, newdata, <span class="at">type =</span> <span class="st">&quot;link&quot;</span>, <span class="at">se.fit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb205-22"><a href="alpha-diversity.html#cb205-22" tabindex="-1"></a></span>
<span id="cb205-23"><a href="alpha-diversity.html#cb205-23" tabindex="-1"></a><span class="co"># Add predicted values to newdata (back-transform with exp)</span></span>
<span id="cb205-24"><a href="alpha-diversity.html#cb205-24" tabindex="-1"></a>newdata<span class="sc">$</span>fit <span class="ot">&lt;-</span> <span class="fu">exp</span>(pred<span class="sc">$</span>fit)</span>
<span id="cb205-25"><a href="alpha-diversity.html#cb205-25" tabindex="-1"></a>newdata<span class="sc">$</span>lwr <span class="ot">&lt;-</span> <span class="fu">exp</span>(pred<span class="sc">$</span>fit <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> pred<span class="sc">$</span>se.fit)   <span class="co"># 95% CI lower</span></span>
<span id="cb205-26"><a href="alpha-diversity.html#cb205-26" tabindex="-1"></a>newdata<span class="sc">$</span>upr <span class="ot">&lt;-</span> <span class="fu">exp</span>(pred<span class="sc">$</span>fit <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> pred<span class="sc">$</span>se.fit)   <span class="co"># 95% CI upper</span></span>
<span id="cb205-27"><a href="alpha-diversity.html#cb205-27" tabindex="-1"></a></span>
<span id="cb205-28"><a href="alpha-diversity.html#cb205-28" tabindex="-1"></a>x_pos <span class="ot">&lt;-</span> <span class="fu">max</span>(alpha_div_macro_filtered_30_gen_length<span class="sc">$</span>median_genome_length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">*</span> <span class="fl">0.98</span></span>
<span id="cb205-29"><a href="alpha-diversity.html#cb205-29" tabindex="-1"></a>y_pos <span class="ot">&lt;-</span> <span class="fu">max</span>(alpha_div_macro_filtered_30_gen_length<span class="sc">$</span>richness, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb205-30"><a href="alpha-diversity.html#cb205-30" tabindex="-1"></a></span>
<span id="cb205-31"><a href="alpha-diversity.html#cb205-31" tabindex="-1"></a></span>
<span id="cb205-32"><a href="alpha-diversity.html#cb205-32" tabindex="-1"></a><span class="co"># Plot with fitted NB curve</span></span>
<span id="cb205-33"><a href="alpha-diversity.html#cb205-33" tabindex="-1"></a><span class="fu">ggplot</span>(alpha_div_macro_filtered_30_gen_length, </span>
<span id="cb205-34"><a href="alpha-diversity.html#cb205-34" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> median_genome_length, <span class="at">y =</span> richness)) <span class="sc">+</span></span>
<span id="cb205-35"><a href="alpha-diversity.html#cb205-35" tabindex="-1"></a>  <span class="co"># NB fit + CI</span></span>
<span id="cb205-36"><a href="alpha-diversity.html#cb205-36" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">data =</span> newdata, </span>
<span id="cb205-37"><a href="alpha-diversity.html#cb205-37" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> median_genome_length, <span class="at">ymin =</span> lwr, <span class="at">ymax =</span> upr),</span>
<span id="cb205-38"><a href="alpha-diversity.html#cb205-38" tabindex="-1"></a>              <span class="at">fill =</span> <span class="st">&quot;grey70&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb205-39"><a href="alpha-diversity.html#cb205-39" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> newdata, </span>
<span id="cb205-40"><a href="alpha-diversity.html#cb205-40" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> median_genome_length, <span class="at">y =</span> fit),</span>
<span id="cb205-41"><a href="alpha-diversity.html#cb205-41" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> <span class="fl">1.2</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb205-42"><a href="alpha-diversity.html#cb205-42" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> treatment, <span class="at">shape =</span> <span class="fu">factor</span>(age_category)), </span>
<span id="cb205-43"><a href="alpha-diversity.html#cb205-43" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb205-44"><a href="alpha-diversity.html#cb205-44" tabindex="-1"></a>  <span class="co"># ➊ Spearman rho + p-value</span></span>
<span id="cb205-45"><a href="alpha-diversity.html#cb205-45" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>,</span>
<span id="cb205-46"><a href="alpha-diversity.html#cb205-46" tabindex="-1"></a>           <span class="at">x =</span> x_pos, <span class="at">y =</span> y_pos,</span>
<span id="cb205-47"><a href="alpha-diversity.html#cb205-47" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Spearman ρ = &quot;</span>, <span class="fu">round</span>(rho, <span class="dv">2</span>),</span>
<span id="cb205-48"><a href="alpha-diversity.html#cb205-48" tabindex="-1"></a>                          <span class="st">&quot;</span><span class="sc">\n</span><span class="st">p = &quot;</span>, <span class="fu">signif</span>(pval, <span class="dv">2</span>)),</span>
<span id="cb205-49"><a href="alpha-diversity.html#cb205-49" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb205-50"><a href="alpha-diversity.html#cb205-50" tabindex="-1"></a>  </span>
<span id="cb205-51"><a href="alpha-diversity.html#cb205-51" tabindex="-1"></a>  <span class="co"># ➋ Pseudo-R²</span></span>
<span id="cb205-52"><a href="alpha-diversity.html#cb205-52" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>,</span>
<span id="cb205-53"><a href="alpha-diversity.html#cb205-53" tabindex="-1"></a>           <span class="at">x =</span> x_pos, <span class="at">y =</span> y_pos <span class="sc">*</span> <span class="fl">0.85</span>,</span>
<span id="cb205-54"><a href="alpha-diversity.html#cb205-54" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Pseudo–R² = &quot;</span>, <span class="fu">round</span>(pseudo_r2, <span class="dv">3</span>)),</span>
<span id="cb205-55"><a href="alpha-diversity.html#cb205-55" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb205-56"><a href="alpha-diversity.html#cb205-56" tabindex="-1"></a></span>
<span id="cb205-57"><a href="alpha-diversity.html#cb205-57" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colours_bright, <span class="at">name =</span> <span class="st">&quot;Treatment&quot;</span>) <span class="sc">+</span></span>
<span id="cb205-58"><a href="alpha-diversity.html#cb205-58" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">name =</span> <span class="st">&quot;DPI&quot;</span>,</span>
<span id="cb205-59"><a href="alpha-diversity.html#cb205-59" tabindex="-1"></a>                     <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;b&quot;</span> <span class="ot">=</span> <span class="dv">17</span>, <span class="st">&quot;c&quot;</span> <span class="ot">=</span> <span class="dv">15</span>, <span class="st">&quot;d&quot;</span> <span class="ot">=</span> <span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb205-60"><a href="alpha-diversity.html#cb205-60" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Median genome length (bp) per sample&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Richness&quot;</span>) <span class="sc">+</span></span>
<span id="cb205-61"><a href="alpha-diversity.html#cb205-61" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb205-62"><a href="alpha-diversity.html#cb205-62" tabindex="-1"></a>  custom_ggplot_theme</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_alpha_div_macro_genome_size_model-1.png" width="672" /></p>
</div>
</div>
<div id="micro-samples-2" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Micro-samples<a href="alpha-diversity.html#micro-samples-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="alpha-diversity-1" class="section level3 hasAnchor" number="14.3.1">
<h3><span class="header-section-number">14.3.1</span> Alpha diversity<a href="alpha-diversity.html#alpha-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb206"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb206-1"><a href="alpha-diversity.html#cb206-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;value&quot;</span></span>
<span id="cb206-2"><a href="alpha-diversity.html#cb206-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> tidy_plot_alpha_diversity_genomes <span class="sc">%&gt;%</span></span>
<span id="cb206-3"><a href="alpha-diversity.html#cb206-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb206-4"><a href="alpha-diversity.html#cb206-4" tabindex="-1"></a>  <span class="co"># filter(filter_level == &#39;filtered_30&#39;) %&gt;%</span></span>
<span id="cb206-5"><a href="alpha-diversity.html#cb206-5" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># to remove empty samples</span></span>
<span id="cb206-6"><a href="alpha-diversity.html#cb206-6" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment_expl, age_category, metric, animal) <span class="sc">%&gt;%</span></span>
<span id="cb206-7"><a href="alpha-diversity.html#cb206-7" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb206-8"><a href="alpha-diversity.html#cb206-8" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb206-9"><a href="alpha-diversity.html#cb206-9" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb206-10"><a href="alpha-diversity.html#cb206-10" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb206-11"><a href="alpha-diversity.html#cb206-11" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb206-12"><a href="alpha-diversity.html#cb206-12" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb206-13"><a href="alpha-diversity.html#cb206-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">formatC</span>(median, <span class="at">format =</span> <span class="st">&quot;f&quot;</span>, <span class="at">digits =</span> <span class="dv">1</span>)) <span class="co">#format(median, scientific = TRUE, digits = 5))</span></span>
<span id="cb206-14"><a href="alpha-diversity.html#cb206-14" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table>
<colgroup>
<col width="25%" />
<col width="11%" />
<col width="11%" />
<col width="6%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">treatment_expl</th>
<th align="left">age_category</th>
<th align="left">metric</th>
<th align="left">animal</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M061a</td>
<td align="right">130.500000</td>
<td align="right">34.0000000</td>
<td align="right">109.250000</td>
<td align="right">143.250000</td>
<td align="left">130.5</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M065a</td>
<td align="right">157.500000</td>
<td align="right">53.0000000</td>
<td align="right">134.000000</td>
<td align="right">187.000000</td>
<td align="left">157.5</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M061a</td>
<td align="right">65.845900</td>
<td align="right">11.6891430</td>
<td align="right">58.483977</td>
<td align="right">70.173120</td>
<td align="left">65.8</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M065a</td>
<td align="right">73.967935</td>
<td align="right">17.5288031</td>
<td align="right">64.139521</td>
<td align="right">81.668324</td>
<td align="left">74.0</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M061a</td>
<td align="right">5.525477</td>
<td align="right">0.2984958</td>
<td align="right">5.320437</td>
<td align="right">5.618932</td>
<td align="left">5.5</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M065a</td>
<td align="right">5.127197</td>
<td align="right">0.6600657</td>
<td align="right">4.771212</td>
<td align="right">5.431278</td>
<td align="left">5.1</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M078d</td>
<td align="right">139.000000</td>
<td align="right">62.0000000</td>
<td align="right">99.000000</td>
<td align="right">161.000000</td>
<td align="left">139.0</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M080d</td>
<td align="right">130.000000</td>
<td align="right">71.5000000</td>
<td align="right">95.500000</td>
<td align="right">167.000000</td>
<td align="left">130.0</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M078d</td>
<td align="right">50.816185</td>
<td align="right">18.3957608</td>
<td align="right">38.670258</td>
<td align="right">57.066019</td>
<td align="left">50.8</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M080d</td>
<td align="right">65.907097</td>
<td align="right">23.7889916</td>
<td align="right">52.513692</td>
<td align="right">76.302684</td>
<td align="left">65.9</td>
</tr>
<tr class="odd">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M078d</td>
<td align="right">4.148053</td>
<td align="right">0.4284805</td>
<td align="right">3.983931</td>
<td align="right">4.412412</td>
<td align="left">4.1</td>
</tr>
<tr class="even">
<td align="left">ControlGroup</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M080d</td>
<td align="right">4.507315</td>
<td align="right">0.5140530</td>
<td align="right">4.223553</td>
<td align="right">4.737606</td>
<td align="left">4.5</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M041a</td>
<td align="right">55.500000</td>
<td align="right">56.2500000</td>
<td align="right">20.750000</td>
<td align="right">77.000000</td>
<td align="left">55.5</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M042a</td>
<td align="right">177.000000</td>
<td align="right">39.0000000</td>
<td align="right">155.000000</td>
<td align="right">194.000000</td>
<td align="left">177.0</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M041a</td>
<td align="right">38.101501</td>
<td align="right">31.3992176</td>
<td align="right">15.789598</td>
<td align="right">47.188815</td>
<td align="left">38.1</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M042a</td>
<td align="right">96.412950</td>
<td align="right">20.5186346</td>
<td align="right">83.608790</td>
<td align="right">104.127425</td>
<td align="left">96.4</td>
</tr>
<tr class="odd">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M041a</td>
<td align="right">3.736384</td>
<td align="right">0.5362499</td>
<td align="right">3.456220</td>
<td align="right">3.992470</td>
<td align="left">3.7</td>
</tr>
<tr class="even">
<td align="left">OnlyHistomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M042a</td>
<td align="right">4.631626</td>
<td align="right">0.4538228</td>
<td align="right">4.365442</td>
<td align="right">4.819264</td>
<td align="left">4.6</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M002a</td>
<td align="right">56.000000</td>
<td align="right">62.2500000</td>
<td align="right">26.500000</td>
<td align="right">88.750000</td>
<td align="left">56.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M003a</td>
<td align="right">77.000000</td>
<td align="right">19.0000000</td>
<td align="right">69.000000</td>
<td align="right">88.000000</td>
<td align="left">77.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M002a</td>
<td align="right">32.958140</td>
<td align="right">25.2620550</td>
<td align="right">17.536660</td>
<td align="right">42.798715</td>
<td align="left">33.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M003a</td>
<td align="right">42.785067</td>
<td align="right">9.5763010</td>
<td align="right">37.868745</td>
<td align="right">47.445046</td>
<td align="left">42.8</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M002a</td>
<td align="right">4.136386</td>
<td align="right">0.8227032</td>
<td align="right">3.607391</td>
<td align="right">4.430094</td>
<td align="left">4.1</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M003a</td>
<td align="right">3.586387</td>
<td align="right">0.4317854</td>
<td align="right">3.295357</td>
<td align="right">3.727143</td>
<td align="left">3.6</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M017d</td>
<td align="right">31.500000</td>
<td align="right">39.7500000</td>
<td align="right">13.500000</td>
<td align="right">53.250000</td>
<td align="left">31.5</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M018d</td>
<td align="right">98.000000</td>
<td align="right">37.0000000</td>
<td align="right">79.000000</td>
<td align="right">116.000000</td>
<td align="left">98.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M017d</td>
<td align="right">22.539126</td>
<td align="right">22.8242044</td>
<td align="right">10.113384</td>
<td align="right">32.937588</td>
<td align="left">22.5</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M018d</td>
<td align="right">52.806886</td>
<td align="right">13.5884550</td>
<td align="right">44.781267</td>
<td align="right">58.369722</td>
<td align="left">52.8</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M017d</td>
<td align="right">4.559298</td>
<td align="right">1.2170197</td>
<td align="right">3.757188</td>
<td align="right">4.974207</td>
<td align="left">4.6</td>
</tr>
<tr class="even">
<td align="left">VaccinatedCloacal&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M018d</td>
<td align="right">5.312937</td>
<td align="right">0.6620478</td>
<td align="right">4.867675</td>
<td align="right">5.529723</td>
<td align="left">5.3</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M021a</td>
<td align="right">55.500000</td>
<td align="right">49.7500000</td>
<td align="right">26.500000</td>
<td align="right">76.250000</td>
<td align="left">55.5</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">richness</td>
<td align="left">M022a</td>
<td align="right">66.000000</td>
<td align="right">51.2500000</td>
<td align="right">40.750000</td>
<td align="right">92.000000</td>
<td align="left">66.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M021a</td>
<td align="right">29.763472</td>
<td align="right">22.6535874</td>
<td align="right">17.093991</td>
<td align="right">39.747579</td>
<td align="left">29.8</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">neutral</td>
<td align="left">M022a</td>
<td align="right">38.110138</td>
<td align="right">19.6758541</td>
<td align="right">27.710731</td>
<td align="right">47.386586</td>
<td align="left">38.1</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M021a</td>
<td align="right">4.377165</td>
<td align="right">0.5961356</td>
<td align="right">4.108973</td>
<td align="right">4.705109</td>
<td align="left">4.4</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">a</td>
<td align="left">phylogenetic</td>
<td align="left">M022a</td>
<td align="right">3.921068</td>
<td align="right">0.6018992</td>
<td align="right">3.668648</td>
<td align="right">4.270547</td>
<td align="left">3.9</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M038d</td>
<td align="right">80.000000</td>
<td align="right">28.5000000</td>
<td align="right">66.000000</td>
<td align="right">94.500000</td>
<td align="left">80.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">richness</td>
<td align="left">M039d</td>
<td align="right">104.000000</td>
<td align="right">25.5000000</td>
<td align="right">88.250000</td>
<td align="right">113.750000</td>
<td align="left">104.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M038d</td>
<td align="right">49.006582</td>
<td align="right">13.0411587</td>
<td align="right">41.853412</td>
<td align="right">54.894570</td>
<td align="left">49.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">neutral</td>
<td align="left">M039d</td>
<td align="right">57.961846</td>
<td align="right">9.0889236</td>
<td align="right">52.534251</td>
<td align="right">61.623175</td>
<td align="left">58.0</td>
</tr>
<tr class="odd">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M038d</td>
<td align="right">6.008672</td>
<td align="right">0.9258856</td>
<td align="right">5.516224</td>
<td align="right">6.442110</td>
<td align="left">6.0</td>
</tr>
<tr class="even">
<td align="left">VaccinatedOral&amp;Histomonas</td>
<td align="left">d</td>
<td align="left">phylogenetic</td>
<td align="left">M039d</td>
<td align="right">5.421190</td>
<td align="right">0.3200240</td>
<td align="right">5.303203</td>
<td align="right">5.623227</td>
<td align="left">5.4</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb207"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb207-1"><a href="alpha-diversity.html#cb207-1" tabindex="-1"></a>plot_alpha_div_micro <span class="ot">&lt;-</span> tidy_plot_alpha_diversity_genomes <span class="sc">%&gt;%</span></span>
<span id="cb207-2"><a href="alpha-diversity.html#cb207-2" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb207-3"><a href="alpha-diversity.html#cb207-3" tabindex="-1"></a>  <span class="fu">filter</span>(filter_level <span class="sc">==</span> <span class="st">&#39;filtered_30&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb207-4"><a href="alpha-diversity.html#cb207-4" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># to remove empty samples</span></span>
<span id="cb207-5"><a href="alpha-diversity.html#cb207-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> animal, <span class="at">y =</span> value, <span class="at">fill =</span> treatment)) <span class="sc">+</span></span>
<span id="cb207-6"><a href="alpha-diversity.html#cb207-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb207-7"><a href="alpha-diversity.html#cb207-7" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb207-8"><a href="alpha-diversity.html#cb207-8" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">names</span>(treatment_colours_bright),</span>
<span id="cb207-9"><a href="alpha-diversity.html#cb207-9" tabindex="-1"></a>                     <span class="at">values =</span> treatment_colours_bright) <span class="sc">+</span></span>
<span id="cb207-10"><a href="alpha-diversity.html#cb207-10" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>) <span class="sc">+</span></span>
<span id="cb207-11"><a href="alpha-diversity.html#cb207-11" tabindex="-1"></a>  <span class="fu">facet_nested</span>(metric <span class="sc">~</span> treatment_expl <span class="sc">+</span> age_category, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb207-12"><a href="alpha-diversity.html#cb207-12" tabindex="-1"></a>  <span class="co"># geom_text(aes(label = animal), </span></span>
<span id="cb207-13"><a href="alpha-diversity.html#cb207-13" tabindex="-1"></a>  <span class="co">#           position = position_jitter(width = 0.2, height = 0), </span></span>
<span id="cb207-14"><a href="alpha-diversity.html#cb207-14" tabindex="-1"></a>  <span class="co">#           size = 3.0, alpha = 0.7, check_overlap = TRUE, color=&quot;#8c1c47&quot;) +</span></span>
<span id="cb207-15"><a href="alpha-diversity.html#cb207-15" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Days Post Infection&quot;</span>, <span class="at">y =</span> <span class="st">&quot;a-div value&quot;</span>) <span class="sc">+</span></span>
<span id="cb207-16"><a href="alpha-diversity.html#cb207-16" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.15</span>))) <span class="sc">+</span></span>
<span id="cb207-17"><a href="alpha-diversity.html#cb207-17" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb207-18"><a href="alpha-diversity.html#cb207-18" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb207-19"><a href="alpha-diversity.html#cb207-19" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> animal, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb207-20"><a href="alpha-diversity.html#cb207-20" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb207-21"><a href="alpha-diversity.html#cb207-21" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">5.0</span>,</span>
<span id="cb207-22"><a href="alpha-diversity.html#cb207-22" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb207-23"><a href="alpha-diversity.html#cb207-23" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb207-24"><a href="alpha-diversity.html#cb207-24" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb207-25"><a href="alpha-diversity.html#cb207-25" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb207-26"><a href="alpha-diversity.html#cb207-26" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb207-27"><a href="alpha-diversity.html#cb207-27" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb207-28"><a href="alpha-diversity.html#cb207-28" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Alpha diversity - 30% coverage filter&quot;</span>)</span>
<span id="cb207-29"><a href="alpha-diversity.html#cb207-29" tabindex="-1"></a>plot_alpha_div_micro</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_alpha_div_micro_main-1.png" width="960" /></p>
</div>
<div id="alpha-div-vs-genome-size-1" class="section level3 hasAnchor" number="14.3.2">
<h3><span class="header-section-number">14.3.2</span> Alpha-div vs genome size<a href="alpha-diversity.html#alpha-div-vs-genome-size-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb208"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb208-1"><a href="alpha-diversity.html#cb208-1" tabindex="-1"></a>tidy_plot_genome_counts_filt_30_closed_zerosrem <span class="ot">&lt;-</span> tidy_plot_genome_counts_filt_30_closed <span class="sc">%&gt;%</span></span>
<span id="cb208-2"><a href="alpha-diversity.html#cb208-2" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb208-3"><a href="alpha-diversity.html#cb208-3" tabindex="-1"></a></span>
<span id="cb208-4"><a href="alpha-diversity.html#cb208-4" tabindex="-1"></a>median_lengths <span class="ot">&lt;-</span> tidy_plot_genome_counts_filt_30_closed_zerosrem <span class="sc">%&gt;%</span></span>
<span id="cb208-5"><a href="alpha-diversity.html#cb208-5" tabindex="-1"></a>  <span class="fu">group_by</span>(microsample) <span class="sc">%&gt;%</span></span>
<span id="cb208-6"><a href="alpha-diversity.html#cb208-6" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_genome_length =</span> <span class="fu">median</span>(length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb208-7"><a href="alpha-diversity.html#cb208-7" tabindex="-1"></a></span>
<span id="cb208-8"><a href="alpha-diversity.html#cb208-8" tabindex="-1"></a>alpha_div_filtered_30_gen_length <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb208-9"><a href="alpha-diversity.html#cb208-9" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb208-10"><a href="alpha-diversity.html#cb208-10" tabindex="-1"></a>  <span class="fu">filter</span>(filter_level <span class="sc">==</span> <span class="st">&quot;filtered_30&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb208-11"><a href="alpha-diversity.html#cb208-11" tabindex="-1"></a>  <span class="fu">left_join</span>(median_lengths, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span></span>
<span id="cb208-12"><a href="alpha-diversity.html#cb208-12" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">!=</span> <span class="st">&quot;TM0&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb208-13"><a href="alpha-diversity.html#cb208-13" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="co">#%&gt;%</span></span>
<span id="cb208-14"><a href="alpha-diversity.html#cb208-14" tabindex="-1"></a>  <span class="co">#filter(richness &gt;10)</span></span></code></pre></div>
<div class="sourceCode" id="cb209"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb209-1"><a href="alpha-diversity.html#cb209-1" tabindex="-1"></a>alpha_div_filtered_30_gen_length <span class="sc">%&gt;%</span></span>
<span id="cb209-2"><a href="alpha-diversity.html#cb209-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> median_genome_length, <span class="at">y =</span> richness, <span class="at">color =</span> treatment, <span class="at">shape =</span> <span class="fu">factor</span>(age_category))) <span class="sc">+</span></span>
<span id="cb209-3"><a href="alpha-diversity.html#cb209-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb209-4"><a href="alpha-diversity.html#cb209-4" tabindex="-1"></a>  <span class="co"># geom_smooth(method = &quot;lm&quot;, se = TRUE, aes(group = 1), color = &quot;black&quot;, linewidth = 1) +</span></span>
<span id="cb209-5"><a href="alpha-diversity.html#cb209-5" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colours_bright, <span class="at">name =</span> <span class="st">&quot;Treatment&quot;</span>) <span class="sc">+</span></span>
<span id="cb209-6"><a href="alpha-diversity.html#cb209-6" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(</span>
<span id="cb209-7"><a href="alpha-diversity.html#cb209-7" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;DPI&quot;</span>,</span>
<span id="cb209-8"><a href="alpha-diversity.html#cb209-8" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;b&quot;</span> <span class="ot">=</span> <span class="dv">17</span>, <span class="st">&quot;c&quot;</span> <span class="ot">=</span> <span class="dv">15</span>, <span class="st">&quot;d&quot;</span> <span class="ot">=</span> <span class="dv">4</span>)  <span class="co"># Example shapes</span></span>
<span id="cb209-9"><a href="alpha-diversity.html#cb209-9" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb209-10"><a href="alpha-diversity.html#cb209-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Median genome length (bp) per sample&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Richness&quot;</span>) <span class="sc">+</span></span>
<span id="cb209-11"><a href="alpha-diversity.html#cb209-11" tabindex="-1"></a>  <span class="co"># ylim(c(-5,300)) +</span></span>
<span id="cb209-12"><a href="alpha-diversity.html#cb209-12" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb209-13"><a href="alpha-diversity.html#cb209-13" tabindex="-1"></a>  custom_ggplot_theme </span></code></pre></div>
<p><img src="_main_files/figure-html/plot_alpha_div_micro_genome_size_plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb210-1"><a href="alpha-diversity.html#cb210-1" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-2"><a href="alpha-diversity.html#cb210-2" tabindex="-1"></a><span class="co"># # Fit model</span></span>
<span id="cb210-3"><a href="alpha-diversity.html#cb210-3" tabindex="-1"></a><span class="co"># model_nb &lt;- glm.nb(richness ~ median_genome_length, </span></span>
<span id="cb210-4"><a href="alpha-diversity.html#cb210-4" tabindex="-1"></a><span class="co">#                    data = alpha_div_filtered_30_gen_length)</span></span>
<span id="cb210-5"><a href="alpha-diversity.html#cb210-5" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-6"><a href="alpha-diversity.html#cb210-6" tabindex="-1"></a><span class="co"># # Fit null model (intercept only)</span></span>
<span id="cb210-7"><a href="alpha-diversity.html#cb210-7" tabindex="-1"></a><span class="co"># model_null &lt;- glm.nb(richness ~ 1,</span></span>
<span id="cb210-8"><a href="alpha-diversity.html#cb210-8" tabindex="-1"></a><span class="co">#                      data = alpha_div_filtered_30_gen_length)</span></span>
<span id="cb210-9"><a href="alpha-diversity.html#cb210-9" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-10"><a href="alpha-diversity.html#cb210-10" tabindex="-1"></a><span class="co"># # Manual McFadden pseudo-R²</span></span>
<span id="cb210-11"><a href="alpha-diversity.html#cb210-11" tabindex="-1"></a><span class="co"># pseudo_r2 &lt;- 1 - (logLik(model_nb) / logLik(model_null))</span></span>
<span id="cb210-12"><a href="alpha-diversity.html#cb210-12" tabindex="-1"></a><span class="co"># pseudo_r2</span></span>
<span id="cb210-13"><a href="alpha-diversity.html#cb210-13" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-14"><a href="alpha-diversity.html#cb210-14" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-15"><a href="alpha-diversity.html#cb210-15" tabindex="-1"></a><span class="co"># # Spearman correlation</span></span>
<span id="cb210-16"><a href="alpha-diversity.html#cb210-16" tabindex="-1"></a><span class="co"># cor_test &lt;- cor.test(</span></span>
<span id="cb210-17"><a href="alpha-diversity.html#cb210-17" tabindex="-1"></a><span class="co">#   ~ richness + median_genome_length,</span></span>
<span id="cb210-18"><a href="alpha-diversity.html#cb210-18" tabindex="-1"></a><span class="co">#   data = alpha_div_filtered_30_gen_length,</span></span>
<span id="cb210-19"><a href="alpha-diversity.html#cb210-19" tabindex="-1"></a><span class="co">#   method = &quot;spearman&quot;</span></span>
<span id="cb210-20"><a href="alpha-diversity.html#cb210-20" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb210-21"><a href="alpha-diversity.html#cb210-21" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-22"><a href="alpha-diversity.html#cb210-22" tabindex="-1"></a><span class="co"># rho  &lt;- cor_test$estimate</span></span>
<span id="cb210-23"><a href="alpha-diversity.html#cb210-23" tabindex="-1"></a><span class="co"># pval &lt;- cor_test$p.value</span></span>
<span id="cb210-24"><a href="alpha-diversity.html#cb210-24" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-25"><a href="alpha-diversity.html#cb210-25" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-26"><a href="alpha-diversity.html#cb210-26" tabindex="-1"></a><span class="co"># # Create new data for prediction</span></span>
<span id="cb210-27"><a href="alpha-diversity.html#cb210-27" tabindex="-1"></a><span class="co"># newdata &lt;- data.frame(</span></span>
<span id="cb210-28"><a href="alpha-diversity.html#cb210-28" tabindex="-1"></a><span class="co">#   median_genome_length = seq(</span></span>
<span id="cb210-29"><a href="alpha-diversity.html#cb210-29" tabindex="-1"></a><span class="co">#     min(alpha_div_filtered_30_gen_length$median_genome_length, na.rm = TRUE),</span></span>
<span id="cb210-30"><a href="alpha-diversity.html#cb210-30" tabindex="-1"></a><span class="co">#     max(alpha_div_filtered_30_gen_length$median_genome_length, na.rm = TRUE),</span></span>
<span id="cb210-31"><a href="alpha-diversity.html#cb210-31" tabindex="-1"></a><span class="co">#     length.out = 200</span></span>
<span id="cb210-32"><a href="alpha-diversity.html#cb210-32" tabindex="-1"></a><span class="co">#   )</span></span>
<span id="cb210-33"><a href="alpha-diversity.html#cb210-33" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb210-34"><a href="alpha-diversity.html#cb210-34" tabindex="-1"></a><span class="co"># # Get predictions + SE on link scale</span></span>
<span id="cb210-35"><a href="alpha-diversity.html#cb210-35" tabindex="-1"></a><span class="co"># pred &lt;- predict(model_nb, newdata, type = &quot;link&quot;, se.fit = TRUE)</span></span>
<span id="cb210-36"><a href="alpha-diversity.html#cb210-36" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-37"><a href="alpha-diversity.html#cb210-37" tabindex="-1"></a><span class="co"># # Add predicted values to newdata (back-transform with exp)</span></span>
<span id="cb210-38"><a href="alpha-diversity.html#cb210-38" tabindex="-1"></a><span class="co"># newdata$fit &lt;- exp(pred$fit)</span></span>
<span id="cb210-39"><a href="alpha-diversity.html#cb210-39" tabindex="-1"></a><span class="co"># newdata$lwr &lt;- exp(pred$fit - 1.96 * pred$se.fit)   # 95% CI lower</span></span>
<span id="cb210-40"><a href="alpha-diversity.html#cb210-40" tabindex="-1"></a><span class="co"># newdata$upr &lt;- exp(pred$fit + 1.96 * pred$se.fit)   # 95% CI upper</span></span>
<span id="cb210-41"><a href="alpha-diversity.html#cb210-41" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-42"><a href="alpha-diversity.html#cb210-42" tabindex="-1"></a><span class="co"># x_pos &lt;- max(alpha_div_filtered_30_gen_length$median_genome_length, na.rm = TRUE) * 0.98</span></span>
<span id="cb210-43"><a href="alpha-diversity.html#cb210-43" tabindex="-1"></a><span class="co"># y_pos &lt;- max(alpha_div_filtered_30_gen_length$richness, na.rm = TRUE) * 0.95</span></span>
<span id="cb210-44"><a href="alpha-diversity.html#cb210-44" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-45"><a href="alpha-diversity.html#cb210-45" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-46"><a href="alpha-diversity.html#cb210-46" tabindex="-1"></a><span class="co"># # Plot with fitted NB curve</span></span>
<span id="cb210-47"><a href="alpha-diversity.html#cb210-47" tabindex="-1"></a><span class="co"># ggplot(alpha_div_filtered_30_gen_length, </span></span>
<span id="cb210-48"><a href="alpha-diversity.html#cb210-48" tabindex="-1"></a><span class="co">#        aes(x = median_genome_length, y = richness)) +</span></span>
<span id="cb210-49"><a href="alpha-diversity.html#cb210-49" tabindex="-1"></a><span class="co">#   # NB fit + CI</span></span>
<span id="cb210-50"><a href="alpha-diversity.html#cb210-50" tabindex="-1"></a><span class="co">#   geom_ribbon(data = newdata, </span></span>
<span id="cb210-51"><a href="alpha-diversity.html#cb210-51" tabindex="-1"></a><span class="co">#               aes(x = median_genome_length, ymin = lwr, ymax = upr),</span></span>
<span id="cb210-52"><a href="alpha-diversity.html#cb210-52" tabindex="-1"></a><span class="co">#               fill = &quot;grey70&quot;, alpha = 0.4, inherit.aes = FALSE) +</span></span>
<span id="cb210-53"><a href="alpha-diversity.html#cb210-53" tabindex="-1"></a><span class="co">#   geom_line(data = newdata, </span></span>
<span id="cb210-54"><a href="alpha-diversity.html#cb210-54" tabindex="-1"></a><span class="co">#             aes(x = median_genome_length, y = fit),</span></span>
<span id="cb210-55"><a href="alpha-diversity.html#cb210-55" tabindex="-1"></a><span class="co">#             color = &quot;black&quot;, linewidth = 1.2, inherit.aes = FALSE) +</span></span>
<span id="cb210-56"><a href="alpha-diversity.html#cb210-56" tabindex="-1"></a><span class="co">#   geom_point(aes(color = treatment, shape = factor(age_category)), </span></span>
<span id="cb210-57"><a href="alpha-diversity.html#cb210-57" tabindex="-1"></a><span class="co">#              size = 3, alpha = 0.8) +</span></span>
<span id="cb210-58"><a href="alpha-diversity.html#cb210-58" tabindex="-1"></a><span class="co">#   # ➊ Spearman rho + p-value</span></span>
<span id="cb210-59"><a href="alpha-diversity.html#cb210-59" tabindex="-1"></a><span class="co">#   annotate(&quot;text&quot;,</span></span>
<span id="cb210-60"><a href="alpha-diversity.html#cb210-60" tabindex="-1"></a><span class="co">#            x = x_pos, y = y_pos,</span></span>
<span id="cb210-61"><a href="alpha-diversity.html#cb210-61" tabindex="-1"></a><span class="co">#            label = paste0(&quot;Spearman ρ = &quot;, round(rho, 2),</span></span>
<span id="cb210-62"><a href="alpha-diversity.html#cb210-62" tabindex="-1"></a><span class="co">#                           &quot;\np = &quot;, signif(pval, 2)),</span></span>
<span id="cb210-63"><a href="alpha-diversity.html#cb210-63" tabindex="-1"></a><span class="co">#            hjust = 1, vjust = 1, size = 3) +</span></span>
<span id="cb210-64"><a href="alpha-diversity.html#cb210-64" tabindex="-1"></a><span class="co">#   </span></span>
<span id="cb210-65"><a href="alpha-diversity.html#cb210-65" tabindex="-1"></a><span class="co">#   # ➋ Pseudo-R²</span></span>
<span id="cb210-66"><a href="alpha-diversity.html#cb210-66" tabindex="-1"></a><span class="co">#   annotate(&quot;text&quot;,</span></span>
<span id="cb210-67"><a href="alpha-diversity.html#cb210-67" tabindex="-1"></a><span class="co">#            x = x_pos, y = y_pos * 0.85,</span></span>
<span id="cb210-68"><a href="alpha-diversity.html#cb210-68" tabindex="-1"></a><span class="co">#            label = paste0(&quot;\nPseudo–R² = &quot;, round(pseudo_r2, 3)),</span></span>
<span id="cb210-69"><a href="alpha-diversity.html#cb210-69" tabindex="-1"></a><span class="co">#            hjust = 1, vjust = 1, size = 3) +</span></span>
<span id="cb210-70"><a href="alpha-diversity.html#cb210-70" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb210-71"><a href="alpha-diversity.html#cb210-71" tabindex="-1"></a><span class="co">#     scale_color_manual(values = treatment_colours_bright, name = &quot;Treatment&quot;) +</span></span>
<span id="cb210-72"><a href="alpha-diversity.html#cb210-72" tabindex="-1"></a><span class="co">#   scale_shape_manual(name = &quot;DPI&quot;,</span></span>
<span id="cb210-73"><a href="alpha-diversity.html#cb210-73" tabindex="-1"></a><span class="co">#                      values = c(&quot;a&quot; = 16, &quot;b&quot; = 17, &quot;c&quot; = 15, &quot;d&quot; = 4)) +</span></span>
<span id="cb210-74"><a href="alpha-diversity.html#cb210-74" tabindex="-1"></a><span class="co">#   labs(x = &quot;Median genome length (bp) per sample&quot;, y = &quot;Richness&quot;) +</span></span>
<span id="cb210-75"><a href="alpha-diversity.html#cb210-75" tabindex="-1"></a><span class="co">#   theme_minimal() +</span></span>
<span id="cb210-76"><a href="alpha-diversity.html#cb210-76" tabindex="-1"></a><span class="co">#   custom_ggplot_theme</span></span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bacterial-load.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="beta-diversity---macro-samples.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
