<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3D’omics | Histomonas Turkeys</title>
  <meta name="description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="3D’omics | Histomonas Turkeys" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="github-repo" content="3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3D’omics | Histomonas Turkeys" />
  
  <meta name="twitter:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  

<meta name="author" content="Amalia Bogri" />
<meta name="author" content="Jorge Langa" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2025-11-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="mag-catalogue.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#environment"><i class="fa fa-check"></i><b>1.1.1</b> Environment</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#libraries"><i class="fa fa-check"></i><b>1.1.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#helper-functions"><i class="fa fa-check"></i><b>1.2</b> Helper Functions</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#alpha-diversity-calculation"><i class="fa fa-check"></i><b>1.2.1</b> Alpha Diversity Calculation</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#remove-samples-or-taxa"><i class="fa fa-check"></i><b>1.2.2</b> Remove samples or taxa</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#clr_function"><i class="fa fa-check"></i><b>1.2.3</b> clr_function</a></li>
<li class="chapter" data-level="1.2.4" data-path="index.html"><a href="index.html#perform_pca"><i class="fa fa-check"></i><b>1.2.4</b> perform_pca</a></li>
<li class="chapter" data-level="1.2.5" data-path="index.html"><a href="index.html#plot_pca"><i class="fa fa-check"></i><b>1.2.5</b> plot_pca</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notes"><i class="fa fa-check"></i><b>1.3</b> Notes:</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#installation-of-modules"><i class="fa fa-check"></i><b>1.3.1</b> Installation of modules</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#airtable-access"><i class="fa fa-check"></i><b>1.3.2</b> Airtable access:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>2</b> MAG Catalogue</a>
<ul>
<li class="chapter" data-level="2.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#dereplicated-genomes"><i class="fa fa-check"></i><b>2.1</b> Dereplicated genomes</a></li>
<li class="chapter" data-level="2.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-tree"><i class="fa fa-check"></i><b>2.2</b> Genome (MAGs) tree</a></li>
<li class="chapter" data-level="2.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-taxonomy"><i class="fa fa-check"></i><b>2.3</b> Genome (MAGs) taxonomy</a></li>
<li class="chapter" data-level="2.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-quality"><i class="fa fa-check"></i><b>2.4</b> Genome (MAGs) quality</a></li>
<li class="chapter" data-level="2.5" data-path="mag-catalogue.html"><a href="mag-catalogue.html#merged-genome-mags-taxonomy-quality"><i class="fa fa-check"></i><b>2.5</b> Merged Genome (MAGs) taxonomy &amp; quality</a></li>
<li class="chapter" data-level="2.6" data-path="mag-catalogue.html"><a href="mag-catalogue.html#factoring-of-taxonomic-levels"><i class="fa fa-check"></i><b>2.6</b> Factoring of taxonomic levels</a></li>
<li class="chapter" data-level="2.7" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-functional-annotations"><i class="fa fa-check"></i><b>2.7</b> Genome (MAGs) functional annotations</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#distil-functional-annotations"><i class="fa fa-check"></i><b>2.7.1</b> Distil functional annotations</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="mag-catalogue.html"><a href="mag-catalogue.html#bacterial-phyla-color-scheme-data"><i class="fa fa-check"></i><b>2.8</b> Bacterial phyla color scheme data</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-color-schemes-for-phyla-and-orders"><i class="fa fa-check"></i><b>2.8.1</b> Define color schemes for phyla and orders</a></li>
<li class="chapter" data-level="2.8.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-plotting-settings"><i class="fa fa-check"></i><b>2.8.2</b> Define plotting settings</a></li>
<li class="chapter" data-level="2.8.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-treatment-color-schemes"><i class="fa fa-check"></i><b>2.8.3</b> Define treatment color schemes</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="mag-catalogue.html"><a href="mag-catalogue.html#save-working-objects"><i class="fa fa-check"></i><b>2.9</b> Save working objects</a></li>
<li class="chapter" data-level="2.10" data-path="mag-catalogue.html"><a href="mag-catalogue.html#validation"><i class="fa fa-check"></i><b>2.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html"><i class="fa fa-check"></i><b>3</b> Micro-sample Metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#load-and-process-micro-sample-metadata-from-airtable"><i class="fa fa-check"></i><b>3.1</b> Load and process micro-sample metadata from Airtable</a></li>
<li class="chapter" data-level="3.2" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#save-working-objects-1"><i class="fa fa-check"></i><b>3.2</b> Save working objects</a></li>
<li class="chapter" data-level="3.3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#validation-1"><i class="fa fa-check"></i><b>3.3</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html"><i class="fa fa-check"></i><b>4</b> Micro-sample Counts</a>
<ul>
<li class="chapter" data-level="4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#load-required-data"><i class="fa fa-check"></i><b>4.1</b> Load required data</a></li>
<li class="chapter" data-level="4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#bacterial-reads-counts"><i class="fa fa-check"></i><b>4.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#genome-covered-bases"><i class="fa fa-check"></i><b>4.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="4.4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#filter-and-normalise-data"><i class="fa fa-check"></i><b>4.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#coverage-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Coverage filtering</a></li>
<li class="chapter" data-level="4.4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-genome-count-table"><i class="fa fa-check"></i><b>4.4.2</b> Generate genome count table</a></li>
<li class="chapter" data-level="4.4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-length-corrected-read-count-tables"><i class="fa fa-check"></i><b>4.4.3</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#alpha-diversity-calculations"><i class="fa fa-check"></i><b>4.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="4.6" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#save-working-objects-2"><i class="fa fa-check"></i><b>4.6</b> Save working objects</a></li>
<li class="chapter" data-level="4.7" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#validation-2"><i class="fa fa-check"></i><b>4.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>5</b> Micro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#load-required-data-1"><i class="fa fa-check"></i><b>5.1</b> Load required data</a></li>
<li class="chapter" data-level="5.2" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#fastp-sequencing-statistics"><i class="fa fa-check"></i><b>5.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="5.3" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas"><i class="fa fa-check"></i><b>5.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="5.4" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#bacterial-mapped-reads"><i class="fa fa-check"></i><b>5.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="5.5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table"><i class="fa fa-check"></i><b>5.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="5.6" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#sanity-checks"><i class="fa fa-check"></i><b>5.6</b> Sanity checks</a></li>
<li class="chapter" data-level="5.7" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics"><i class="fa fa-check"></i><b>5.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="5.8" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#create-plot_data_stats"><i class="fa fa-check"></i><b>5.8</b> Create plot_data_stats</a></li>
<li class="chapter" data-level="5.9" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#save-working-objects-3"><i class="fa fa-check"></i><b>5.9</b> Save working objects</a></li>
<li class="chapter" data-level="5.10" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#validation-3"><i class="fa fa-check"></i><b>5.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html"><i class="fa fa-check"></i><b>6</b> Macro-sample Metadata</a>
<ul>
<li class="chapter" data-level="6.1" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#save-working-objects-4"><i class="fa fa-check"></i><b>6.1</b> Save working objects</a></li>
<li class="chapter" data-level="6.2" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#validation-4"><i class="fa fa-check"></i><b>6.2</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html"><i class="fa fa-check"></i><b>7</b> Macro-sample Counts</a>
<ul>
<li class="chapter" data-level="7.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#load-required-data-2"><i class="fa fa-check"></i><b>7.1</b> Load required data</a></li>
<li class="chapter" data-level="7.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#bacterial-reads-counts-1"><i class="fa fa-check"></i><b>7.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="7.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#genome-covered-bases-1"><i class="fa fa-check"></i><b>7.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="7.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#filter-and-normalise-data-1"><i class="fa fa-check"></i><b>7.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-coverage-table"><i class="fa fa-check"></i><b>7.4.1</b> Generate coverage table</a></li>
<li class="chapter" data-level="7.4.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#coverage-filtering-1"><i class="fa fa-check"></i><b>7.4.2</b> Coverage filtering</a></li>
<li class="chapter" data-level="7.4.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-genome-count-table-1"><i class="fa fa-check"></i><b>7.4.3</b> Generate genome count table</a></li>
<li class="chapter" data-level="7.4.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-length-corrected-read-count-tables-1"><i class="fa fa-check"></i><b>7.4.4</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#alpha-diversity-calculations-1"><i class="fa fa-check"></i><b>7.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="7.6" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#save-working-objects-5"><i class="fa fa-check"></i><b>7.6</b> Save working objects</a></li>
<li class="chapter" data-level="7.7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#validation-5"><i class="fa fa-check"></i><b>7.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>8</b> Macro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#load-required-data-3"><i class="fa fa-check"></i><b>8.1</b> Load required data</a></li>
<li class="chapter" data-level="8.2" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#fastp-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="8.3" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas-1"><i class="fa fa-check"></i><b>8.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="8.4" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#bacterial-mapped-reads-1"><i class="fa fa-check"></i><b>8.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="8.5" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table-1"><i class="fa fa-check"></i><b>8.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="8.6" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#sanity-checks-1"><i class="fa fa-check"></i><b>8.6</b> Sanity checks</a></li>
<li class="chapter" data-level="8.7" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="8.8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#create-plot_data_stats_macro"><i class="fa fa-check"></i><b>8.8</b> Create plot_data_stats_macro</a></li>
<li class="chapter" data-level="8.9" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#save-working-objects-6"><i class="fa fa-check"></i><b>8.9</b> Save working objects</a></li>
<li class="chapter" data-level="8.10" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#validation-6"><i class="fa fa-check"></i><b>8.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html"><i class="fa fa-check"></i><b>9</b> Metabolomics Data Preparation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-metabolomics-data"><i class="fa fa-check"></i><b>9.1</b> Load metabolomics data</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-abundance-values-and-metabolites-metadata"><i class="fa fa-check"></i><b>9.1.1</b> Load abundance values and metabolites metadata</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#classify-samples-by-type"><i class="fa fa-check"></i><b>9.2</b> Classify samples by type</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#identify-sample-columns-and-classify-by-type"><i class="fa fa-check"></i><b>9.2.1</b> Identify sample columns and classify by type</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#create-dataframes-for-each-sample-type"><i class="fa fa-check"></i><b>9.3</b> Create dataframes for each sample type</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hn-histomonas-chicken-host-tissue"><i class="fa fa-check"></i><b>9.3.1</b> HN: Histomonas Chicken, host tissue</a></li>
<li class="chapter" data-level="9.3.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hm-histomonas-chicken-digesta"><i class="fa fa-check"></i><b>9.3.2</b> HM: Histomonas Chicken, digesta</a></li>
<li class="chapter" data-level="9.3.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mn-histomonas-turkey-host-tissue"><i class="fa fa-check"></i><b>9.3.3</b> MN: Histomonas Turkey, host tissue</a></li>
<li class="chapter" data-level="9.3.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mm-histomonas-turkey-digesta"><i class="fa fa-check"></i><b>9.3.4</b> MM: Histomonas Turkey, digesta</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-processed-data"><i class="fa fa-check"></i><b>9.4</b> Save processed data</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-metabolomics-dataframes"><i class="fa fa-check"></i><b>9.4.1</b> Save metabolomics dataframes</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validation-7"><i class="fa fa-check"></i><b>9.5</b> Validation</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validate-metabolomics-data"><i class="fa fa-check"></i><b>9.5.1</b> Validate metabolomics data</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#notes-on-metabolites-metadata"><i class="fa fa-check"></i><b>9.6</b> Notes on metabolites metadata</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#available-metadata-fields"><i class="fa fa-check"></i><b>9.6.1</b> Available metadata fields:</a></li>
<li class="chapter" data-level="9.6.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#suggested-uses"><i class="fa fa-check"></i><b>9.6.2</b> Suggested uses:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html"><i class="fa fa-check"></i><b>10</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-quality"><i class="fa fa-check"></i><b>10.1</b> Genome quality</a></li>
<li class="chapter" data-level="10.2" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours"><i class="fa fa-check"></i><b>10.2</b> MAG catalogue phyla and order colours</a></li>
<li class="chapter" data-level="10.3" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours-and-genera-number"><i class="fa fa-check"></i><b>10.3</b> MAG catalogue phyla and order colours, and genera number</a></li>
<li class="chapter" data-level="10.4" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot"><i class="fa fa-check"></i><b>10.4</b> Genome phylogeny plot</a></li>
<li class="chapter" data-level="10.5" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot-without-legend"><i class="fa fa-check"></i><b>10.5</b> Genome phylogeny plot without legend</a></li>
<li class="chapter" data-level="10.6" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-overview"><i class="fa fa-check"></i><b>10.6</b> Functional overview</a></li>
<li class="chapter" data-level="10.7" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-distances"><i class="fa fa-check"></i><b>10.7</b> Functional distances</a></li>
<li class="chapter" data-level="10.8" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-ordination"><i class="fa fa-check"></i><b>10.8</b> Functional ordination</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="histomonas-presence.html"><a href="histomonas-presence.html"><i class="fa fa-check"></i><b>11</b> Histomonas Presence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="histomonas-presence.html"><a href="histomonas-presence.html#macro"><i class="fa fa-check"></i><b>11.1</b> Macro</a></li>
<li class="chapter" data-level="11.2" data-path="histomonas-presence.html"><a href="histomonas-presence.html#micro"><i class="fa fa-check"></i><b>11.2</b> Micro</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html"><i class="fa fa-check"></i><b>12</b> Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#prepare-tidy-tables-for-plotting"><i class="fa fa-check"></i><b>12.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="12.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#macro-samples"><i class="fa fa-check"></i><b>12.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases-reads"><i class="fa fa-check"></i><b>12.2.1</b> Number of bases &amp; reads</a></li>
<li class="chapter" data-level="12.2.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped"><i class="fa fa-check"></i><b>12.2.2</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.2.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#composition-of-reads-vs.-references"><i class="fa fa-check"></i><b>12.2.3</b> Composition of reads vs. references</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#micro-samples"><i class="fa fa-check"></i><b>12.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases"><i class="fa fa-check"></i><b>12.3.1</b> Number of bases</a></li>
<li class="chapter" data-level="12.3.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-reads-after-trimming"><i class="fa fa-check"></i><b>12.3.2</b> Number of reads after trimming</a></li>
<li class="chapter" data-level="12.3.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped-1"><i class="fa fa-check"></i><b>12.3.3</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.3.4" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts"><i class="fa fa-check"></i><b>12.3.4</b> Collection attempts</a></li>
<li class="chapter" data-level="12.3.5" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts-vs-filtering-retention"><i class="fa fa-check"></i><b>12.3.5</b> Collection attempts vs filtering retention</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="bacterial-load.html"><a href="bacterial-load.html"><i class="fa fa-check"></i><b>13</b> Bacterial Load</a>
<ul>
<li class="chapter" data-level="13.1" data-path="bacterial-load.html"><a href="bacterial-load.html#macro-samples-1"><i class="fa fa-check"></i><b>13.1</b> Macro-samples</a></li>
<li class="chapter" data-level="13.2" data-path="bacterial-load.html"><a href="bacterial-load.html#micro-samples-1"><i class="fa fa-check"></i><b>13.2</b> Micro-samples</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="14.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#prepare-tidy-tables-for-plotting-1"><i class="fa fa-check"></i><b>14.1</b> Prepare tidy tables for plotting</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-tidy-tables"><i class="fa fa-check"></i><b>14.1.1</b> Alpha diversity tidy tables</a></li>
<li class="chapter" data-level="14.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#genome-counts-tidy-tables"><i class="fa fa-check"></i><b>14.1.2</b> Genome counts tidy tables</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#macro-samples-2"><i class="fa fa-check"></i><b>14.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div"><i class="fa fa-check"></i><b>14.2.1</b> Alpha-div</a></li>
<li class="chapter" data-level="14.2.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size"><i class="fa fa-check"></i><b>14.2.2</b> Alpha-div vs genome size</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#micro-samples-2"><i class="fa fa-check"></i><b>14.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-1"><i class="fa fa-check"></i><b>14.3.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="14.3.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size-1"><i class="fa fa-check"></i><b>14.3.2</b> Alpha-div vs genome size</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html"><i class="fa fa-check"></i><b>15</b> Beta Diversity - Macro Samples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#prepare-tidy-tables-for-plotting-2"><i class="fa fa-check"></i><b>15.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="15.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-level"><i class="fa fa-check"></i><b>15.2</b> Species level</a></li>
<li class="chapter" data-level="15.3" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#genus-level"><i class="fa fa-check"></i><b>15.3</b> Genus level</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#permanova"><i class="fa fa-check"></i><b>15.3.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="15.3.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-representative-genomes-overlap"><i class="fa fa-check"></i><b>15.3.2</b> Species representative genomes overlap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html"><i class="fa fa-check"></i><b>16</b> Beta Diversity - Micro Samples</a>
<ul>
<li class="chapter" data-level="16.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#prepare-tidy-tables-for-plotting-3"><i class="fa fa-check"></i><b>16.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="16.2" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#species-level-1"><i class="fa fa-check"></i><b>16.2</b> Species level</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#permanova-1"><i class="fa fa-check"></i><b>16.2.1</b> PERMANOVA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html"><i class="fa fa-check"></i><b>17</b> Beta Diversity - Macro vs Micro Comparison</a>
<ul>
<li class="chapter" data-level="17.1" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#genomes-overlap"><i class="fa fa-check"></i><b>17.1</b> Genomes overlap</a></li>
<li class="chapter" data-level="17.2" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#micro-vs-macro-genus-level"><i class="fa fa-check"></i><b>17.2</b> Micro vs Macro Genus level</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html"><i class="fa fa-check"></i><b>18</b> Spatial Analysis - Micro Samples</a>
<ul>
<li class="chapter" data-level="18.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#functions"><i class="fa fa-check"></i><b>18.1</b> Functions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#spatial_cryosections"><i class="fa fa-check"></i><b>18.1.1</b> spatial_cryosections</a></li>
<li class="chapter" data-level="18.1.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#additive-diversity-partitioning"><i class="fa fa-check"></i><b>18.1.2</b> Additive diversity partitioning</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#load-data"><i class="fa fa-check"></i><b>18.2</b> Load data</a></li>
<li class="chapter" data-level="18.3" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#extended-rlq-analysis"><i class="fa fa-check"></i><b>18.3</b> Extended RLQ analysis</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html"><i class="fa fa-check"></i><b>19</b> Export Data for MicrobeTag</a>
<ul>
<li class="chapter" data-level="19.1" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#load-required-data-4"><i class="fa fa-check"></i><b>19.1</b> Load required data</a></li>
<li class="chapter" data-level="19.2" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-sample-metadata-macro-samples"><i class="fa fa-check"></i><b>19.2</b> Export sample metadata (macro samples)</a></li>
<li class="chapter" data-level="19.3" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-metabolites-data-mm-samples"><i class="fa fa-check"></i><b>19.3</b> Export metabolites data (MM samples)</a></li>
<li class="chapter" data-level="19.4" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-metadata"><i class="fa fa-check"></i><b>19.4</b> Export genome metadata</a></li>
<li class="chapter" data-level="19.5" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-counts-macro-samples-filtered"><i class="fa fa-check"></i><b>19.5</b> Export genome counts (macro samples, filtered)</a></li>
<li class="chapter" data-level="19.6" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#summary"><i class="fa fa-check"></i><b>19.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-versions.html"><a href="package-versions.html"><i class="fa fa-check"></i><b>20</b> Package Versions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">3D’omics | Histomonas Turkeys</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">3D’omics | Histomonas Turkeys</h1>
<h2 class="subtitle"><em>Work in progress</em></h2>
<p class="author"><em>Amalia Bogri<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></em></p>
<p class="author"><em>Jorge Langa<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></em></p>
<p class="author"><em>Antton Alberdi<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></em></p>
<p class="date"><em>2025-11-17</em></p>
</div>
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This webbook contains all the code used for the analyses of for Deliverable 6.3 for the Histomonas trials on turkeys.</p>
<div id="prepare-the-r-environment" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Prepare the R environment<a href="index.html#prepare-the-r-environment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="environment" class="section level3 hasAnchor" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Environment<a href="index.html#environment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To reproduce all the analyses locally, clone this repository in your computer using:</p>
<pre><code>RStudio &gt; New Project &gt; Version Control &gt; Git</code></pre>
<p>And indicating the following git repository:</p>
<blockquote>
<p><a href="https://github.com/3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro.git" class="uri">https://github.com/3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro.git</a></p>
</blockquote>
<p>Once the R project has been created, follow the instructions and code chunks shown in this webbook.</p>
</div>
<div id="libraries" class="section level3 hasAnchor" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Libraries<a href="index.html#libraries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following R packages are required for the data analysis.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" tabindex="-1"></a><span class="co"># Base</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(R.utils)</span>
<span id="cb2-3"><a href="index.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-4"><a href="index.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb2-5"><a href="index.html#cb2-5" tabindex="-1"></a><span class="fu">library</span>(tinytable)</span>
<span id="cb2-6"><a href="index.html#cb2-6" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb2-7"><a href="index.html#cb2-7" tabindex="-1"></a><span class="fu">library</span>(broom.mixed)</span>
<span id="cb2-8"><a href="index.html#cb2-8" tabindex="-1"></a><span class="fu">library</span>(rairtable)</span>
<span id="cb2-9"><a href="index.html#cb2-9" tabindex="-1"></a><span class="fu">library</span>(jsonlite)</span>
<span id="cb2-10"><a href="index.html#cb2-10" tabindex="-1"></a></span>
<span id="cb2-11"><a href="index.html#cb2-11" tabindex="-1"></a><span class="co"># For tree handling</span></span>
<span id="cb2-12"><a href="index.html#cb2-12" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb2-13"><a href="index.html#cb2-13" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb2-14"><a href="index.html#cb2-14" tabindex="-1"></a><span class="fu">library</span>(phytools)</span>
<span id="cb2-15"><a href="index.html#cb2-15" tabindex="-1"></a></span>
<span id="cb2-16"><a href="index.html#cb2-16" tabindex="-1"></a><span class="co"># For plotting</span></span>
<span id="cb2-17"><a href="index.html#cb2-17" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-18"><a href="index.html#cb2-18" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb2-19"><a href="index.html#cb2-19" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb2-20"><a href="index.html#cb2-20" tabindex="-1"></a><span class="fu">library</span>(ggnewscale)</span>
<span id="cb2-21"><a href="index.html#cb2-21" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb2-22"><a href="index.html#cb2-22" tabindex="-1"></a><span class="fu">library</span>(ggtreeExtra)</span>
<span id="cb2-23"><a href="index.html#cb2-23" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb2-24"><a href="index.html#cb2-24" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb2-25"><a href="index.html#cb2-25" tabindex="-1"></a><span class="fu">library</span>(jpeg)</span>
<span id="cb2-26"><a href="index.html#cb2-26" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-27"><a href="index.html#cb2-27" tabindex="-1"></a></span>
<span id="cb2-28"><a href="index.html#cb2-28" tabindex="-1"></a><span class="co"># For statistics</span></span>
<span id="cb2-29"><a href="index.html#cb2-29" tabindex="-1"></a><span class="fu">library</span>(spaa)</span>
<span id="cb2-30"><a href="index.html#cb2-30" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb2-31"><a href="index.html#cb2-31" tabindex="-1"></a><span class="fu">library</span>(Rtsne)</span>
<span id="cb2-32"><a href="index.html#cb2-32" tabindex="-1"></a><span class="fu">library</span>(geiger)</span>
<span id="cb2-33"><a href="index.html#cb2-33" tabindex="-1"></a><span class="fu">library</span>(hilldiv2)</span>
<span id="cb2-34"><a href="index.html#cb2-34" tabindex="-1"></a><span class="fu">library</span>(distillR)</span>
<span id="cb2-35"><a href="index.html#cb2-35" tabindex="-1"></a><span class="fu">library</span>(lme4)</span>
<span id="cb2-36"><a href="index.html#cb2-36" tabindex="-1"></a><span class="fu">library</span>(Hmsc)</span>
<span id="cb2-37"><a href="index.html#cb2-37" tabindex="-1"></a><span class="fu">library</span>(psych)</span>
<span id="cb2-38"><a href="index.html#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="index.html#cb2-39" tabindex="-1"></a><span class="co"># For compositional analyses</span></span>
<span id="cb2-40"><a href="index.html#cb2-40" tabindex="-1"></a><span class="fu">library</span>(zCompositions)</span>
<span id="cb2-41"><a href="index.html#cb2-41" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb2-42"><a href="index.html#cb2-42" tabindex="-1"></a><span class="fu">library</span>(ALDEx2)</span>
<span id="cb2-43"><a href="index.html#cb2-43" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb2-44"><a href="index.html#cb2-44" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb2-45"><a href="index.html#cb2-45" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb2-46"><a href="index.html#cb2-46" tabindex="-1"></a><span class="fu">library</span>(propr)</span>
<span id="cb2-47"><a href="index.html#cb2-47" tabindex="-1"></a></span>
<span id="cb2-48"><a href="index.html#cb2-48" tabindex="-1"></a><span class="fu">library</span>(MASS) <span class="co"># for Negative Binomial Fit</span></span>
<span id="cb2-49"><a href="index.html#cb2-49" tabindex="-1"></a><span class="fu">library</span>(ggVennDiagram)</span>
<span id="cb2-50"><a href="index.html#cb2-50" tabindex="-1"></a></span>
<span id="cb2-51"><a href="index.html#cb2-51" tabindex="-1"></a></span>
<span id="cb2-52"><a href="index.html#cb2-52" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb2-53"><a href="index.html#cb2-53" tabindex="-1"></a><span class="fu">library</span>(permuco)</span>
<span id="cb2-54"><a href="index.html#cb2-54" tabindex="-1"></a><span class="fu">library</span>(magick)</span>
<span id="cb2-55"><a href="index.html#cb2-55" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb2-56"><a href="index.html#cb2-56" tabindex="-1"></a><span class="fu">library</span>(spdep)</span>
<span id="cb2-57"><a href="index.html#cb2-57" tabindex="-1"></a></span>
<span id="cb2-58"><a href="index.html#cb2-58" tabindex="-1"></a></span>
<span id="cb2-59"><a href="index.html#cb2-59" tabindex="-1"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb2-60"><a href="index.html#cb2-60" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb2-61"><a href="index.html#cb2-61" tabindex="-1"></a><span class="co"># Tidyverse libraries (load dplyr last to avoid conflicts)</span></span>
<span id="cb2-62"><a href="index.html#cb2-62" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-63"><a href="index.html#cb2-63" tabindex="-1"></a><span class="fu">library</span>(dplyr) <span class="co"># Explicitly load dplyr last</span></span></code></pre></div>
</div>
</div>
<div id="helper-functions" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Helper Functions<a href="index.html#helper-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="alpha-diversity-calculation" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Alpha Diversity Calculation<a href="index.html#alpha-diversity-calculation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Function to estimate the genomes’ alpha diversity measurements from genome counts.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" tabindex="-1"></a>calculate_alpha_diversity <span class="ot">&lt;-</span> <span class="cf">function</span>(input_data, dataset_name) {</span>
<span id="cb3-2"><a href="index.html#cb3-2" tabindex="-1"></a>  <span class="co"># Step 1: Transform the input data (remove rownames if needed)</span></span>
<span id="cb3-3"><a href="index.html#cb3-3" tabindex="-1"></a>  input_data_matrix <span class="ot">&lt;-</span> input_data <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="index.html#cb3-4" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>)  <span class="co">#%&gt;%</span></span>
<span id="cb3-5"><a href="index.html#cb3-5" tabindex="-1"></a>   <span class="co"># mutate_at(vars(-genome), ~ . / sum(.))  # No reason to do this closure to 1. Results are the same if you do it or not.</span></span>
<span id="cb3-6"><a href="index.html#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="index.html#cb3-7" tabindex="-1"></a>  <span class="co"># Step 2: Calculate richness (q = 0)</span></span>
<span id="cb3-8"><a href="index.html#cb3-8" tabindex="-1"></a>  richness <span class="ot">&lt;-</span> <span class="fu">hilldiv</span>(input_data_matrix, <span class="at">q =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="index.html#cb3-9" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="index.html#cb3-10" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-11"><a href="index.html#cb3-11" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">richness =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-12"><a href="index.html#cb3-12" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;microsample&quot;</span>)</span>
<span id="cb3-13"><a href="index.html#cb3-13" tabindex="-1"></a></span>
<span id="cb3-14"><a href="index.html#cb3-14" tabindex="-1"></a>  <span class="co"># Step 3: Calculate neutral diversity (q = 1)</span></span>
<span id="cb3-15"><a href="index.html#cb3-15" tabindex="-1"></a>  neutral <span class="ot">&lt;-</span> <span class="fu">hilldiv</span>(input_data_matrix, <span class="at">q =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-16"><a href="index.html#cb3-16" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-17"><a href="index.html#cb3-17" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-18"><a href="index.html#cb3-18" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">neutral =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-19"><a href="index.html#cb3-19" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;microsample&quot;</span>)</span>
<span id="cb3-20"><a href="index.html#cb3-20" tabindex="-1"></a></span>
<span id="cb3-21"><a href="index.html#cb3-21" tabindex="-1"></a>  <span class="co"># Step 4: Calculate phylogenetic diversity (q = 1, with genome tree)</span></span>
<span id="cb3-22"><a href="index.html#cb3-22" tabindex="-1"></a>  phylogenetic <span class="ot">&lt;-</span> <span class="fu">hilldiv</span>(input_data_matrix, <span class="at">q =</span> <span class="dv">1</span>, <span class="at">tree =</span> genome_tree) <span class="sc">%&gt;%</span></span>
<span id="cb3-23"><a href="index.html#cb3-23" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-24"><a href="index.html#cb3-24" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-25"><a href="index.html#cb3-25" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">phylogenetic =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-26"><a href="index.html#cb3-26" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;microsample&quot;</span>)</span>
<span id="cb3-27"><a href="index.html#cb3-27" tabindex="-1"></a></span>
<span id="cb3-28"><a href="index.html#cb3-28" tabindex="-1"></a>  <span class="co"># Step 5: Merge all diversity metrics</span></span>
<span id="cb3-29"><a href="index.html#cb3-29" tabindex="-1"></a>  alpha_diversity <span class="ot">&lt;-</span> richness <span class="sc">%&gt;%</span></span>
<span id="cb3-30"><a href="index.html#cb3-30" tabindex="-1"></a>    <span class="fu">full_join</span>(neutral, <span class="at">by =</span> <span class="st">&quot;microsample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-31"><a href="index.html#cb3-31" tabindex="-1"></a>    <span class="fu">full_join</span>(phylogenetic, <span class="at">by =</span> <span class="st">&quot;microsample&quot;</span>) <span class="co"># %&gt;%</span></span>
<span id="cb3-32"><a href="index.html#cb3-32" tabindex="-1"></a>    <span class="co"># left_join(sample_metadata, by = &quot;microsample&quot;) %&gt;%</span></span>
<span id="cb3-33"><a href="index.html#cb3-33" tabindex="-1"></a>    <span class="co"># left_join(final_combined_stats, by = &quot;microsample&quot;) %&gt;%</span></span>
<span id="cb3-34"><a href="index.html#cb3-34" tabindex="-1"></a></span>
<span id="cb3-35"><a href="index.html#cb3-35" tabindex="-1"></a>  <span class="co"># Return the alpha_diversity data frame</span></span>
<span id="cb3-36"><a href="index.html#cb3-36" tabindex="-1"></a>  <span class="fu">return</span>(alpha_diversity)</span>
<span id="cb3-37"><a href="index.html#cb3-37" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="remove-samples-or-taxa" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Remove samples or taxa<a href="index.html#remove-samples-or-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" tabindex="-1"></a><span class="co"># Input: A wide df with samples as rows and taxa as columns</span></span>
<span id="cb4-2"><a href="index.html#cb4-2" tabindex="-1"></a><span class="co"># Result: Remove taxa that are in too few samples, or samples that have too few taxa</span></span>
<span id="cb4-3"><a href="index.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="index.html#cb4-4" tabindex="-1"></a>remove_samples_or_taxa <span class="ot">&lt;-</span> <span class="cf">function</span>(df, min_samples_per_taxon, min_taxa_per_sample){</span>
<span id="cb4-5"><a href="index.html#cb4-5" tabindex="-1"></a>  <span class="co"># Store &amp; print original df dimensions</span></span>
<span id="cb4-6"><a href="index.html#cb4-6" tabindex="-1"></a>  original_rows <span class="ot">&lt;-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb4-7"><a href="index.html#cb4-7" tabindex="-1"></a>  original_cols <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df)</span>
<span id="cb4-8"><a href="index.html#cb4-8" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Initial df: Rows (samples):&quot;</span>, original_rows, <span class="st">&quot;, Columns (taxa):&quot;</span>, original_cols, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb4-9"><a href="index.html#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="index.html#cb4-10" tabindex="-1"></a>  <span class="co"># Remove taxa that are in less than x samples</span></span>
<span id="cb4-11"><a href="index.html#cb4-11" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">sum</span>(. <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">&gt;</span> min_samples_per_taxon))</span>
<span id="cb4-12"><a href="index.html#cb4-12" tabindex="-1"></a>  </span>
<span id="cb4-13"><a href="index.html#cb4-13" tabindex="-1"></a>  <span class="co"># Remove samples that contain less than x taxa</span></span>
<span id="cb4-14"><a href="index.html#cb4-14" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">&gt;</span> min_taxa_per_sample)</span>
<span id="cb4-15"><a href="index.html#cb4-15" tabindex="-1"></a>  </span>
<span id="cb4-16"><a href="index.html#cb4-16" tabindex="-1"></a>  df <span class="sc">&lt;</span> df <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">any</span>(. <span class="sc">!=</span> <span class="dv">0</span>)) <span class="sc">|</span> <span class="fu">where</span>(is.character) <span class="sc">|</span> <span class="fu">where</span>(is.factor))  </span>
<span id="cb4-17"><a href="index.html#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="index.html#cb4-18" tabindex="-1"></a>  removed_rows <span class="ot">&lt;-</span> original_rows <span class="sc">-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb4-19"><a href="index.html#cb4-19" tabindex="-1"></a>  removed_cols <span class="ot">&lt;-</span> original_cols <span class="sc">-</span> <span class="fu">ncol</span>(df)</span>
<span id="cb4-20"><a href="index.html#cb4-20" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Removed: Rows (samples):&quot;</span>, removed_rows, <span class="st">&quot;, Columns (taxa):&quot;</span>, removed_cols, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb4-21"><a href="index.html#cb4-21" tabindex="-1"></a></span>
<span id="cb4-22"><a href="index.html#cb4-22" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Resulting df: Rows (samples):&quot;</span>, <span class="fu">nrow</span>(df), <span class="st">&quot;, Columns (taxa):&quot;</span>, <span class="fu">ncol</span>(df), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb4-23"><a href="index.html#cb4-23" tabindex="-1"></a>  </span>
<span id="cb4-24"><a href="index.html#cb4-24" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb4-25"><a href="index.html#cb4-25" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="clr_function" class="section level3 hasAnchor" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> clr_function<a href="index.html#clr_function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb5-1"><a href="index.html#cb5-1" tabindex="-1"></a>  clr_transform <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb5-2"><a href="index.html#cb5-2" tabindex="-1"></a>    <span class="fu">log</span>(x) <span class="sc">-</span> <span class="fu">mean</span>(<span class="fu">log</span>(x), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-3"><a href="index.html#cb5-3" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="perform_pca" class="section level3 hasAnchor" number="1.2.4">
<h3><span class="header-section-number">1.2.4</span> perform_pca<a href="index.html#perform_pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb6-1"><a href="index.html#cb6-1" tabindex="-1"></a>perform_pca <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">zero_method =</span> <span class="st">&quot;GBM&quot;</span>, <span class="at">z_delete =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb6-2"><a href="index.html#cb6-2" tabindex="-1"></a>  <span class="co"># Store original dimensions</span></span>
<span id="cb6-3"><a href="index.html#cb6-3" tabindex="-1"></a>  original_rows <span class="ot">&lt;-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb6-4"><a href="index.html#cb6-4" tabindex="-1"></a>  original_cols <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df)</span>
<span id="cb6-5"><a href="index.html#cb6-5" tabindex="-1"></a></span>
<span id="cb6-6"><a href="index.html#cb6-6" tabindex="-1"></a>  <span class="co"># 1. Zero replacement</span></span>
<span id="cb6-7"><a href="index.html#cb6-7" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">any</span>(df <span class="sc">==</span> <span class="dv">0</span>)) { <span class="co"># I think cmultRepl already does that</span></span>
<span id="cb6-8"><a href="index.html#cb6-8" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&quot;Zeros found&quot;</span>)</span>
<span id="cb6-9"><a href="index.html#cb6-9" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">cmultRepl</span>(df,</span>
<span id="cb6-10"><a href="index.html#cb6-10" tabindex="-1"></a>      <span class="at">method =</span> zero_method, <span class="at">output =</span> <span class="st">&quot;prop&quot;</span>,</span>
<span id="cb6-11"><a href="index.html#cb6-11" tabindex="-1"></a>      <span class="at">z.warning =</span> <span class="fl">0.8</span>, <span class="at">z.delete =</span> z_delete</span>
<span id="cb6-12"><a href="index.html#cb6-12" tabindex="-1"></a>    )</span>
<span id="cb6-13"><a href="index.html#cb6-13" tabindex="-1"></a>    df <span class="ot">&lt;-</span> df <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb6-14"><a href="index.html#cb6-14" tabindex="-1"></a>  }</span>
<span id="cb6-15"><a href="index.html#cb6-15" tabindex="-1"></a></span>
<span id="cb6-16"><a href="index.html#cb6-16" tabindex="-1"></a>  <span class="co"># Print removed rows and columns</span></span>
<span id="cb6-17"><a href="index.html#cb6-17" tabindex="-1"></a>  removed_rows <span class="ot">&lt;-</span> original_rows <span class="sc">-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb6-18"><a href="index.html#cb6-18" tabindex="-1"></a>  removed_cols <span class="ot">&lt;-</span> original_cols <span class="sc">-</span> <span class="fu">ncol</span>(df)</span>
<span id="cb6-19"><a href="index.html#cb6-19" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Rows (samples) removed after zero replacement:&quot;</span>, removed_rows, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-20"><a href="index.html#cb6-20" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Columns (taxa) removed after zero replacement:&quot;</span>, removed_cols, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-21"><a href="index.html#cb6-21" tabindex="-1"></a></span>
<span id="cb6-22"><a href="index.html#cb6-22" tabindex="-1"></a>  <span class="co"># Geometric mean function</span></span>
<span id="cb6-23"><a href="index.html#cb6-23" tabindex="-1"></a>  geometric_mean <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb6-24"><a href="index.html#cb6-24" tabindex="-1"></a>    <span class="co"># Use log to avoid underflow</span></span>
<span id="cb6-25"><a href="index.html#cb6-25" tabindex="-1"></a>    <span class="fu">exp</span>(<span class="fu">mean</span>(<span class="fu">log</span>(x), <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb6-26"><a href="index.html#cb6-26" tabindex="-1"></a>  }</span>
<span id="cb6-27"><a href="index.html#cb6-27" tabindex="-1"></a></span>
<span id="cb6-28"><a href="index.html#cb6-28" tabindex="-1"></a>  <span class="co"># 2. Calculate geometric mean of the parts (taxa) of the data set.</span></span>
<span id="cb6-29"><a href="index.html#cb6-29" tabindex="-1"></a>  taxa_geometric_means <span class="ot">&lt;-</span> <span class="fu">apply</span>(df, <span class="dv">2</span>, geometric_mean)</span>
<span id="cb6-30"><a href="index.html#cb6-30" tabindex="-1"></a></span>
<span id="cb6-31"><a href="index.html#cb6-31" tabindex="-1"></a>  <span class="co"># 3. Center data</span></span>
<span id="cb6-32"><a href="index.html#cb6-32" tabindex="-1"></a>  df_centered <span class="ot">&lt;-</span> <span class="fu">sweep</span>(df, <span class="dv">2</span>, taxa_geometric_means, <span class="at">FUN =</span> <span class="st">&quot;/&quot;</span>)</span>
<span id="cb6-33"><a href="index.html#cb6-33" tabindex="-1"></a></span>
<span id="cb6-34"><a href="index.html#cb6-34" tabindex="-1"></a>  df_centered <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(df_centered)</span>
<span id="cb6-35"><a href="index.html#cb6-35" tabindex="-1"></a></span>
<span id="cb6-36"><a href="index.html#cb6-36" tabindex="-1"></a>  <span class="co"># Compute the Variation Matrix</span></span>
<span id="cb6-37"><a href="index.html#cb6-37" tabindex="-1"></a>  variation_matrix <span class="ot">&lt;-</span> <span class="fu">outer</span>(</span>
<span id="cb6-38"><a href="index.html#cb6-38" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(df_centered), <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(df_centered),</span>
<span id="cb6-39"><a href="index.html#cb6-39" tabindex="-1"></a>    <span class="fu">Vectorize</span>(<span class="cf">function</span>(i, j) <span class="fu">var</span>(<span class="fu">log</span>(df_centered[, i] <span class="sc">/</span> df_centered[, j]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb6-40"><a href="index.html#cb6-40" tabindex="-1"></a>  )</span>
<span id="cb6-41"><a href="index.html#cb6-41" tabindex="-1"></a></span>
<span id="cb6-42"><a href="index.html#cb6-42" tabindex="-1"></a>  <span class="co"># Calculate Total Variance</span></span>
<span id="cb6-43"><a href="index.html#cb6-43" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df_centered) <span class="co"># Number of taxa (columns)</span></span>
<span id="cb6-44"><a href="index.html#cb6-44" tabindex="-1"></a>  totvar <span class="ot">&lt;-</span> (<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">2</span> <span class="sc">*</span> D)) <span class="sc">*</span> <span class="fu">sum</span>(variation_matrix, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-45"><a href="index.html#cb6-45" tabindex="-1"></a></span>
<span id="cb6-46"><a href="index.html#cb6-46" tabindex="-1"></a>  <span class="co"># 4. Scale data</span></span>
<span id="cb6-47"><a href="index.html#cb6-47" tabindex="-1"></a>  power_exponent <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> <span class="fu">sqrt</span>(totvar)</span>
<span id="cb6-48"><a href="index.html#cb6-48" tabindex="-1"></a>  df_scaled <span class="ot">&lt;-</span> df_centered<span class="sc">^</span>power_exponent</span>
<span id="cb6-49"><a href="index.html#cb6-49" tabindex="-1"></a></span>
<span id="cb6-50"><a href="index.html#cb6-50" tabindex="-1"></a>  <span class="co"># CLR transform data</span></span>
<span id="cb6-51"><a href="index.html#cb6-51" tabindex="-1"></a>  df_clr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(df_scaled, <span class="dv">1</span>, clr_transform)))</span>
<span id="cb6-52"><a href="index.html#cb6-52" tabindex="-1"></a></span>
<span id="cb6-53"><a href="index.html#cb6-53" tabindex="-1"></a>  df_clr_dist <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(df, <span class="dv">1</span>, clr_transform)))</span>
<span id="cb6-54"><a href="index.html#cb6-54" tabindex="-1"></a></span>
<span id="cb6-55"><a href="index.html#cb6-55" tabindex="-1"></a>  <span class="co"># Perform PCA on zero replaced, centered, scaled, and CLR transformed df</span></span>
<span id="cb6-56"><a href="index.html#cb6-56" tabindex="-1"></a>  pca_result <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(df_clr, <span class="at">center =</span> <span class="cn">FALSE</span>, <span class="at">scale. =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-57"><a href="index.html#cb6-57" tabindex="-1"></a></span>
<span id="cb6-58"><a href="index.html#cb6-58" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(</span>
<span id="cb6-59"><a href="index.html#cb6-59" tabindex="-1"></a>    <span class="at">df_clr =</span> df_clr,</span>
<span id="cb6-60"><a href="index.html#cb6-60" tabindex="-1"></a>    <span class="at">df_clr_dist =</span> df_clr_dist,</span>
<span id="cb6-61"><a href="index.html#cb6-61" tabindex="-1"></a>    <span class="at">pca_result =</span> pca_result</span>
<span id="cb6-62"><a href="index.html#cb6-62" tabindex="-1"></a>  ))</span>
<span id="cb6-63"><a href="index.html#cb6-63" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_pca" class="section level3 hasAnchor" number="1.2.5">
<h3><span class="header-section-number">1.2.5</span> plot_pca<a href="index.html#plot_pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb7-1"><a href="index.html#cb7-1" tabindex="-1"></a><span class="co"># Use the &#39;pca_result&#39; df produced from &#39;perform_pca&#39; function to make the PCA plot</span></span>
<span id="cb7-2"><a href="index.html#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="index.html#cb7-3" tabindex="-1"></a>plot_pca <span class="ot">&lt;-</span> <span class="cf">function</span>(df, </span>
<span id="cb7-4"><a href="index.html#cb7-4" tabindex="-1"></a>                     samples_color_metadata, samples_shape_metadata, </span>
<span id="cb7-5"><a href="index.html#cb7-5" tabindex="-1"></a>                     samples_color_value, loadings_color_metadata, </span>
<span id="cb7-6"><a href="index.html#cb7-6" tabindex="-1"></a>                     loadings_color_value, loadings_taxon_level,</span>
<span id="cb7-7"><a href="index.html#cb7-7" tabindex="-1"></a>                     sample_metadata, genome_metadata, order_colors, </span>
<span id="cb7-8"><a href="index.html#cb7-8" tabindex="-1"></a>                     custom_ggplot_theme, <span class="at">scaling_factor_value =</span> <span class="fl">1.5</span>, </span>
<span id="cb7-9"><a href="index.html#cb7-9" tabindex="-1"></a>                     <span class="at">loadings_number =</span> <span class="dv">10</span>, <span class="at">show_labels =</span> <span class="cn">FALSE</span>, <span class="at">add_arrows =</span> <span class="cn">FALSE</span>, <span class="at">add_centroids =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb7-10"><a href="index.html#cb7-10" tabindex="-1"></a>  </span>
<span id="cb7-11"><a href="index.html#cb7-11" tabindex="-1"></a>  <span class="co"># Extract scores from PCA results</span></span>
<span id="cb7-12"><a href="index.html#cb7-12" tabindex="-1"></a>  scores <span class="ot">&lt;-</span> <span class="fu">rownames_to_column</span>(<span class="fu">as.data.frame</span>(df<span class="sc">$</span>x), <span class="at">var =</span> <span class="st">&quot;microsample&quot;</span>)</span>
<span id="cb7-13"><a href="index.html#cb7-13" tabindex="-1"></a>  scores <span class="ot">&lt;-</span> <span class="fu">left_join</span>(scores, sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample))</span>
<span id="cb7-14"><a href="index.html#cb7-14" tabindex="-1"></a>  </span>
<span id="cb7-15"><a href="index.html#cb7-15" tabindex="-1"></a>  <span class="co"># Calculate limits for x and y axes</span></span>
<span id="cb7-16"><a href="index.html#cb7-16" tabindex="-1"></a>  x_limit <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">abs</span>(scores<span class="sc">$</span>PC1))</span>
<span id="cb7-17"><a href="index.html#cb7-17" tabindex="-1"></a>  y_limit <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">abs</span>(scores<span class="sc">$</span>PC2))</span>
<span id="cb7-18"><a href="index.html#cb7-18" tabindex="-1"></a>  </span>
<span id="cb7-19"><a href="index.html#cb7-19" tabindex="-1"></a>  <span class="co"># Calculate variance explained by each PC (principal component) &amp; create labels for plot</span></span>
<span id="cb7-20"><a href="index.html#cb7-20" tabindex="-1"></a>  variance_explained <span class="ot">&lt;-</span> (df<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">/</span> <span class="fu">sum</span>(df<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb7-21"><a href="index.html#cb7-21" tabindex="-1"></a>  pc1_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>, <span class="fu">round</span>(variance_explained[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;% variance explained&quot;</span>)</span>
<span id="cb7-22"><a href="index.html#cb7-22" tabindex="-1"></a>  pc2_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>, <span class="fu">round</span>(variance_explained[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;% variance explained&quot;</span>)</span>
<span id="cb7-23"><a href="index.html#cb7-23" tabindex="-1"></a>  </span>
<span id="cb7-24"><a href="index.html#cb7-24" tabindex="-1"></a>  <span class="co"># Set a scaling factor for loadings (arrows)</span></span>
<span id="cb7-25"><a href="index.html#cb7-25" tabindex="-1"></a>  scaling_factor <span class="ot">&lt;-</span> scaling_factor_value</span>
<span id="cb7-26"><a href="index.html#cb7-26" tabindex="-1"></a>  </span>
<span id="cb7-27"><a href="index.html#cb7-27" tabindex="-1"></a>  <span class="co"># Extract and scale the loadings</span></span>
<span id="cb7-28"><a href="index.html#cb7-28" tabindex="-1"></a>  loadings <span class="ot">&lt;-</span> df<span class="sc">$</span>rotation[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">%&gt;%</span></span>
<span id="cb7-29"><a href="index.html#cb7-29" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-30"><a href="index.html#cb7-30" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">genome =</span> <span class="fu">rownames</span>(.)) <span class="sc">%&gt;%</span>  </span>
<span id="cb7-31"><a href="index.html#cb7-31" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">PC1 =</span> PC1 <span class="sc">*</span> scaling_factor,</span>
<span id="cb7-32"><a href="index.html#cb7-32" tabindex="-1"></a>           <span class="at">PC2 =</span> PC2 <span class="sc">*</span> scaling_factor) <span class="sc">%&gt;%</span></span>
<span id="cb7-33"><a href="index.html#cb7-33" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span>  </span>
<span id="cb7-34"><a href="index.html#cb7-34" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">abs_loading =</span> <span class="fu">sqrt</span>(PC1<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> PC2<span class="sc">^</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span>  </span>
<span id="cb7-35"><a href="index.html#cb7-35" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(abs_loading)) <span class="sc">%&gt;%</span></span>
<span id="cb7-36"><a href="index.html#cb7-36" tabindex="-1"></a>    <span class="fu">slice_max</span>(<span class="at">order_by =</span> abs_loading, <span class="at">n =</span> loadings_number) <span class="sc">%&gt;%</span></span>
<span id="cb7-37"><a href="index.html#cb7-37" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">order_color =</span> order_colors[order])</span>
<span id="cb7-38"><a href="index.html#cb7-38" tabindex="-1"></a>  </span>
<span id="cb7-39"><a href="index.html#cb7-39" tabindex="-1"></a>  <span class="co"># Create ggplot</span></span>
<span id="cb7-40"><a href="index.html#cb7-40" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb7-41"><a href="index.html#cb7-41" tabindex="-1"></a>    <span class="co"># Plot the samples (points)</span></span>
<span id="cb7-42"><a href="index.html#cb7-42" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> scores, </span>
<span id="cb7-43"><a href="index.html#cb7-43" tabindex="-1"></a>               <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, </span>
<span id="cb7-44"><a href="index.html#cb7-44" tabindex="-1"></a>                   <span class="at">fill =</span> .data[[samples_color_metadata]],</span>
<span id="cb7-45"><a href="index.html#cb7-45" tabindex="-1"></a>                   <span class="at">shape =</span> .data[[samples_shape_metadata]]), <span class="co">#.data[[]] tells ggplot2 to look up the column dynamically.</span></span>
<span id="cb7-46"><a href="index.html#cb7-46" tabindex="-1"></a>               <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb7-47"><a href="index.html#cb7-47" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">stroke =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb7-48"><a href="index.html#cb7-48" tabindex="-1"></a>    </span>
<span id="cb7-49"><a href="index.html#cb7-49" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> samples_color_value) <span class="sc">+</span> </span>
<span id="cb7-50"><a href="index.html#cb7-50" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">24</span>, <span class="dv">23</span>, <span class="dv">22</span>, <span class="dv">25</span>)) <span class="sc">+</span></span>
<span id="cb7-51"><a href="index.html#cb7-51" tabindex="-1"></a>    <span class="fu">new_scale_color</span>() <span class="sc">+</span> <span class="co">#  new_scale_fill() ?</span></span>
<span id="cb7-52"><a href="index.html#cb7-52" tabindex="-1"></a>    <span class="co"># Plot the loadings (taxa, i.e. arrows)</span></span>
<span id="cb7-53"><a href="index.html#cb7-53" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="at">data =</span> loadings, </span>
<span id="cb7-54"><a href="index.html#cb7-54" tabindex="-1"></a>                 <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> PC1, <span class="at">yend =</span> PC2, <span class="at">color =</span> .data[[loadings_color_metadata]]),</span>
<span id="cb7-55"><a href="index.html#cb7-55" tabindex="-1"></a>                 <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.2</span>, <span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb7-56"><a href="index.html#cb7-56" tabindex="-1"></a>                 <span class="at">size =</span> <span class="fl">0.7</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb7-57"><a href="index.html#cb7-57" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">name =</span> <span class="st">&quot;Classification&quot;</span>, <span class="at">values =</span> loadings_color_value) <span class="sc">+</span></span>
<span id="cb7-58"><a href="index.html#cb7-58" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> loadings, </span>
<span id="cb7-59"><a href="index.html#cb7-59" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> .data[[loadings_taxon_level]]),</span>
<span id="cb7-60"><a href="index.html#cb7-60" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.7</span>, <span class="at">max.overlaps =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb7-61"><a href="index.html#cb7-61" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;PCA Ordination Plot&quot;</span>,</span>
<span id="cb7-62"><a href="index.html#cb7-62" tabindex="-1"></a>         <span class="at">x =</span> pc1_label,</span>
<span id="cb7-63"><a href="index.html#cb7-63" tabindex="-1"></a>         <span class="at">y =</span> pc2_label) <span class="sc">+</span></span>
<span id="cb7-64"><a href="index.html#cb7-64" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span>x_limit, x_limit)) <span class="sc">+</span></span>
<span id="cb7-65"><a href="index.html#cb7-65" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span>y_limit, y_limit)) <span class="sc">+</span></span>
<span id="cb7-66"><a href="index.html#cb7-66" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">&quot;darkgrey&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-67"><a href="index.html#cb7-67" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">&quot;darkgrey&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-68"><a href="index.html#cb7-68" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb7-69"><a href="index.html#cb7-69" tabindex="-1"></a>    custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb7-70"><a href="index.html#cb7-70" tabindex="-1"></a>    <span class="co"># coord_fixed(ratio = 1) +</span></span>
<span id="cb7-71"><a href="index.html#cb7-71" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>))) <span class="co"># to print the legend for colour correctly!</span></span>
<span id="cb7-72"><a href="index.html#cb7-72" tabindex="-1"></a>  </span>
<span id="cb7-73"><a href="index.html#cb7-73" tabindex="-1"></a>  </span>
<span id="cb7-74"><a href="index.html#cb7-74" tabindex="-1"></a>      <span class="co"># ADD text to points</span></span>
<span id="cb7-75"><a href="index.html#cb7-75" tabindex="-1"></a>     <span class="cf">if</span> (show_labels) {</span>
<span id="cb7-76"><a href="index.html#cb7-76" tabindex="-1"></a>        p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb7-77"><a href="index.html#cb7-77" tabindex="-1"></a>          <span class="fu">geom_text</span>(</span>
<span id="cb7-78"><a href="index.html#cb7-78" tabindex="-1"></a>            <span class="at">data =</span> scores, </span>
<span id="cb7-79"><a href="index.html#cb7-79" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">label =</span> animal),</span>
<span id="cb7-80"><a href="index.html#cb7-80" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">y =</span> <span class="sc">-</span><span class="fl">0.02</span>),</span>
<span id="cb7-81"><a href="index.html#cb7-81" tabindex="-1"></a>            <span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span></span>
<span id="cb7-82"><a href="index.html#cb7-82" tabindex="-1"></a>          )</span>
<span id="cb7-83"><a href="index.html#cb7-83" tabindex="-1"></a>     }</span>
<span id="cb7-84"><a href="index.html#cb7-84" tabindex="-1"></a>  </span>
<span id="cb7-85"><a href="index.html#cb7-85" tabindex="-1"></a>  <span class="cf">if</span> (add_arrows) {</span>
<span id="cb7-86"><a href="index.html#cb7-86" tabindex="-1"></a>  <span class="co"># Prepare arrow data</span></span>
<span id="cb7-87"><a href="index.html#cb7-87" tabindex="-1"></a>  arrows_df <span class="ot">&lt;-</span> scores <span class="sc">%&gt;%</span></span>
<span id="cb7-88"><a href="index.html#cb7-88" tabindex="-1"></a>    <span class="fu">arrange</span>(treatment, age) <span class="sc">%&gt;%</span></span>
<span id="cb7-89"><a href="index.html#cb7-89" tabindex="-1"></a>    <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span></span>
<span id="cb7-90"><a href="index.html#cb7-90" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb7-91"><a href="index.html#cb7-91" tabindex="-1"></a>      <span class="at">PC1_next =</span> <span class="fu">lead</span>(PC1),</span>
<span id="cb7-92"><a href="index.html#cb7-92" tabindex="-1"></a>      <span class="at">PC2_next =</span> <span class="fu">lead</span>(PC2)</span>
<span id="cb7-93"><a href="index.html#cb7-93" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-94"><a href="index.html#cb7-94" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(PC1_next))</span>
<span id="cb7-95"><a href="index.html#cb7-95" tabindex="-1"></a>  </span>
<span id="cb7-96"><a href="index.html#cb7-96" tabindex="-1"></a>  <span class="co"># Add arrows to the plot</span></span>
<span id="cb7-97"><a href="index.html#cb7-97" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb7-98"><a href="index.html#cb7-98" tabindex="-1"></a>    <span class="fu">geom_segment</span>(</span>
<span id="cb7-99"><a href="index.html#cb7-99" tabindex="-1"></a>      <span class="at">data =</span> arrows_df,</span>
<span id="cb7-100"><a href="index.html#cb7-100" tabindex="-1"></a>      <span class="fu">aes</span>(</span>
<span id="cb7-101"><a href="index.html#cb7-101" tabindex="-1"></a>        <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb7-102"><a href="index.html#cb7-102" tabindex="-1"></a>        <span class="at">xend =</span> PC1_next, <span class="at">yend =</span> PC2_next,</span>
<span id="cb7-103"><a href="index.html#cb7-103" tabindex="-1"></a>        <span class="at">color =</span> treatment</span>
<span id="cb7-104"><a href="index.html#cb7-104" tabindex="-1"></a>      ),</span>
<span id="cb7-105"><a href="index.html#cb7-105" tabindex="-1"></a>      <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.15</span>, <span class="st">&quot;cm&quot;</span>)),</span>
<span id="cb7-106"><a href="index.html#cb7-106" tabindex="-1"></a>      <span class="at">linewidth =</span> <span class="fl">0.6</span>,</span>
<span id="cb7-107"><a href="index.html#cb7-107" tabindex="-1"></a>      <span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb7-108"><a href="index.html#cb7-108" tabindex="-1"></a>      <span class="at">inherit.aes =</span> <span class="cn">FALSE</span></span>
<span id="cb7-109"><a href="index.html#cb7-109" tabindex="-1"></a>    )</span>
<span id="cb7-110"><a href="index.html#cb7-110" tabindex="-1"></a>  }</span>
<span id="cb7-111"><a href="index.html#cb7-111" tabindex="-1"></a>  </span>
<span id="cb7-112"><a href="index.html#cb7-112" tabindex="-1"></a><span class="cf">if</span> (add_centroids) {</span>
<span id="cb7-113"><a href="index.html#cb7-113" tabindex="-1"></a></span>
<span id="cb7-114"><a href="index.html#cb7-114" tabindex="-1"></a>  <span class="co"># Compute centroids per treatment × age</span></span>
<span id="cb7-115"><a href="index.html#cb7-115" tabindex="-1"></a>  centroids <span class="ot">&lt;-</span> scores <span class="sc">%&gt;%</span></span>
<span id="cb7-116"><a href="index.html#cb7-116" tabindex="-1"></a>    <span class="fu">group_by</span>(.data[[samples_color_metadata]], .data[[samples_shape_metadata]]) <span class="sc">%&gt;%</span></span>
<span id="cb7-117"><a href="index.html#cb7-117" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb7-118"><a href="index.html#cb7-118" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="fu">mean</span>(PC1),</span>
<span id="cb7-119"><a href="index.html#cb7-119" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="fu">mean</span>(PC2),</span>
<span id="cb7-120"><a href="index.html#cb7-120" tabindex="-1"></a>      <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span></span>
<span id="cb7-121"><a href="index.html#cb7-121" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-122"><a href="index.html#cb7-122" tabindex="-1"></a>    <span class="fu">rename</span>(</span>
<span id="cb7-123"><a href="index.html#cb7-123" tabindex="-1"></a>      <span class="at">color_group =</span> <span class="dv">1</span>,</span>
<span id="cb7-124"><a href="index.html#cb7-124" tabindex="-1"></a>      <span class="at">shape_group =</span> <span class="dv">2</span></span>
<span id="cb7-125"><a href="index.html#cb7-125" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-126"><a href="index.html#cb7-126" tabindex="-1"></a>    <span class="fu">arrange</span>(color_group, shape_group)</span>
<span id="cb7-127"><a href="index.html#cb7-127" tabindex="-1"></a></span>
<span id="cb7-128"><a href="index.html#cb7-128" tabindex="-1"></a>  centroid_arrows <span class="ot">&lt;-</span> centroids <span class="sc">%&gt;%</span></span>
<span id="cb7-129"><a href="index.html#cb7-129" tabindex="-1"></a>    <span class="fu">group_by</span>(color_group) <span class="sc">%&gt;%</span></span>
<span id="cb7-130"><a href="index.html#cb7-130" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb7-131"><a href="index.html#cb7-131" tabindex="-1"></a>      <span class="at">PC1_next =</span> <span class="fu">lead</span>(PC1),</span>
<span id="cb7-132"><a href="index.html#cb7-132" tabindex="-1"></a>      <span class="at">PC2_next =</span> <span class="fu">lead</span>(PC2)</span>
<span id="cb7-133"><a href="index.html#cb7-133" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-134"><a href="index.html#cb7-134" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(PC1_next))</span>
<span id="cb7-135"><a href="index.html#cb7-135" tabindex="-1"></a></span>
<span id="cb7-136"><a href="index.html#cb7-136" tabindex="-1"></a>  <span class="co"># Add new color scale so centroids use treatment palette again</span></span>
<span id="cb7-137"><a href="index.html#cb7-137" tabindex="-1"></a>  p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb7-138"><a href="index.html#cb7-138" tabindex="-1"></a>    <span class="fu">new_scale_color</span>() <span class="sc">+</span></span>
<span id="cb7-139"><a href="index.html#cb7-139" tabindex="-1"></a>    <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb7-140"><a href="index.html#cb7-140" tabindex="-1"></a></span>
<span id="cb7-141"><a href="index.html#cb7-141" tabindex="-1"></a>    <span class="fu">geom_segment</span>(</span>
<span id="cb7-142"><a href="index.html#cb7-142" tabindex="-1"></a>      <span class="at">data =</span> centroid_arrows,</span>
<span id="cb7-143"><a href="index.html#cb7-143" tabindex="-1"></a>      <span class="fu">aes</span>(</span>
<span id="cb7-144"><a href="index.html#cb7-144" tabindex="-1"></a>        <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb7-145"><a href="index.html#cb7-145" tabindex="-1"></a>        <span class="at">xend =</span> PC1_next, <span class="at">yend =</span> PC2_next,</span>
<span id="cb7-146"><a href="index.html#cb7-146" tabindex="-1"></a>        <span class="at">color =</span> color_group</span>
<span id="cb7-147"><a href="index.html#cb7-147" tabindex="-1"></a>      ),</span>
<span id="cb7-148"><a href="index.html#cb7-148" tabindex="-1"></a>      <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;cm&quot;</span>), <span class="at">type =</span> <span class="st">&quot;closed&quot;</span>),</span>
<span id="cb7-149"><a href="index.html#cb7-149" tabindex="-1"></a>      <span class="at">linewidth =</span> <span class="fl">1.2</span>,</span>
<span id="cb7-150"><a href="index.html#cb7-150" tabindex="-1"></a>      <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb7-151"><a href="index.html#cb7-151" tabindex="-1"></a>      <span class="at">inherit.aes =</span> <span class="cn">FALSE</span></span>
<span id="cb7-152"><a href="index.html#cb7-152" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-153"><a href="index.html#cb7-153" tabindex="-1"></a>    <span class="fu">geom_point</span>(</span>
<span id="cb7-154"><a href="index.html#cb7-154" tabindex="-1"></a>      <span class="at">data =</span> centroids,</span>
<span id="cb7-155"><a href="index.html#cb7-155" tabindex="-1"></a>      <span class="fu">aes</span>(</span>
<span id="cb7-156"><a href="index.html#cb7-156" tabindex="-1"></a>        <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb7-157"><a href="index.html#cb7-157" tabindex="-1"></a>        <span class="at">shape =</span> <span class="fu">factor</span>(shape_group),</span>
<span id="cb7-158"><a href="index.html#cb7-158" tabindex="-1"></a>        <span class="at">color =</span> color_group,</span>
<span id="cb7-159"><a href="index.html#cb7-159" tabindex="-1"></a>        <span class="at">fill  =</span> color_group</span>
<span id="cb7-160"><a href="index.html#cb7-160" tabindex="-1"></a>      ),</span>
<span id="cb7-161"><a href="index.html#cb7-161" tabindex="-1"></a>      <span class="at">size =</span> <span class="fl">4.5</span>,</span>
<span id="cb7-162"><a href="index.html#cb7-162" tabindex="-1"></a>      <span class="at">stroke =</span> <span class="dv">1</span>,</span>
<span id="cb7-163"><a href="index.html#cb7-163" tabindex="-1"></a>      <span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb7-164"><a href="index.html#cb7-164" tabindex="-1"></a>      <span class="at">fill =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb7-165"><a href="index.html#cb7-165" tabindex="-1"></a>      <span class="at">inherit.aes =</span> <span class="cn">FALSE</span></span>
<span id="cb7-166"><a href="index.html#cb7-166" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-167"><a href="index.html#cb7-167" tabindex="-1"></a>    <span class="fu">geom_text</span>(</span>
<span id="cb7-168"><a href="index.html#cb7-168" tabindex="-1"></a>      <span class="at">data =</span> centroids,</span>
<span id="cb7-169"><a href="index.html#cb7-169" tabindex="-1"></a>      <span class="fu">aes</span>(</span>
<span id="cb7-170"><a href="index.html#cb7-170" tabindex="-1"></a>        <span class="at">x =</span> PC1, <span class="at">y =</span> PC2,</span>
<span id="cb7-171"><a href="index.html#cb7-171" tabindex="-1"></a>        <span class="at">label =</span> shape_group,</span>
<span id="cb7-172"><a href="index.html#cb7-172" tabindex="-1"></a>        <span class="at">color =</span> color_group</span>
<span id="cb7-173"><a href="index.html#cb7-173" tabindex="-1"></a>      ),</span>
<span id="cb7-174"><a href="index.html#cb7-174" tabindex="-1"></a>      <span class="at">size =</span> <span class="fl">3.3</span>,</span>
<span id="cb7-175"><a href="index.html#cb7-175" tabindex="-1"></a>      <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">1.2</span>,</span>
<span id="cb7-176"><a href="index.html#cb7-176" tabindex="-1"></a>      <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>,</span>
<span id="cb7-177"><a href="index.html#cb7-177" tabindex="-1"></a>      <span class="at">inherit.aes =</span> <span class="cn">FALSE</span></span>
<span id="cb7-178"><a href="index.html#cb7-178" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-179"><a href="index.html#cb7-179" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> samples_color_value) <span class="sc">+</span></span>
<span id="cb7-180"><a href="index.html#cb7-180" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> samples_color_value)</span>
<span id="cb7-181"><a href="index.html#cb7-181" tabindex="-1"></a>}</span>
<span id="cb7-182"><a href="index.html#cb7-182" tabindex="-1"></a>  </span>
<span id="cb7-183"><a href="index.html#cb7-183" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb7-184"><a href="index.html#cb7-184" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="notes" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Notes:<a href="index.html#notes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="installation-of-modules" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Installation of modules<a href="index.html#installation-of-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Most of the modules can be downloaded from CRAN with ‘install.packages(’XXX’, dependencies = TRUE)‘.
Four modules are from Bioconductor.
Install with: ’install.packages(’BiocManager’, dependencies = TRUE)
BiocManager::install(c(“phyloseq”, “ggtreeExtra”,“ggtree”,“ANCOMBC”))’
Two modules are from Antton’s github.
Install with: ‘install.packages(’remotes’, dependencies = TRUE)
remotes::install_github(‘anttonalberdi/hilldiv2’)
remotes::install_github(‘anttonalberdi/distillR’)’
# if (!require(“BiocManager”, quietly = TRUE))
# install.packages(“BiocManager”)
# BiocManager::install(“ALDEx2”)
# remotes::install_github(‘tpq/propr’)</p>
</div>
<div id="airtable-access" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Airtable access:<a href="index.html#airtable-access" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Currently the metadata is loaded from the 3D’omics airtable. In order to access it you need your own API key (from Antton). set_airtable_api_key(‘XXXXX’, install = TRUE)</p>

</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>University of Copenhagen, <a href="mailto:amalia.bogri@sund.ku.dk" class="email">amalia.bogri@sund.ku.dk</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>University of Copenhagen, <a href="mailto:jorge.langa@sund.ku.dk" class="email">jorge.langa@sund.ku.dk</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>University of Copenhagen, <a href="mailto:antton.alberdi@sund.ku.dk" class="email">antton.alberdi@sund.ku.dk</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="mag-catalogue.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
