<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 MAG Catalogue | 3D’omics | Histomonas Turkeys</title>
  <meta name="description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="2 MAG Catalogue | 3D’omics | Histomonas Turkeys" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="github-repo" content="3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 MAG Catalogue | 3D’omics | Histomonas Turkeys" />
  
  <meta name="twitter:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  

<meta name="author" content="Amalia Bogri" />
<meta name="author" content="Jorge Langa" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2025-11-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="micro-sample-metadata.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#environment"><i class="fa fa-check"></i><b>1.1.1</b> Environment</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#libraries"><i class="fa fa-check"></i><b>1.1.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#helper-functions"><i class="fa fa-check"></i><b>1.2</b> Helper Functions</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#alpha-diversity-calculation"><i class="fa fa-check"></i><b>1.2.1</b> Alpha Diversity Calculation</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#remove-samples-or-taxa"><i class="fa fa-check"></i><b>1.2.2</b> Remove samples or taxa</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#clr_function"><i class="fa fa-check"></i><b>1.2.3</b> clr_function</a></li>
<li class="chapter" data-level="1.2.4" data-path="index.html"><a href="index.html#perform_pca"><i class="fa fa-check"></i><b>1.2.4</b> perform_pca</a></li>
<li class="chapter" data-level="1.2.5" data-path="index.html"><a href="index.html#plot_pca"><i class="fa fa-check"></i><b>1.2.5</b> plot_pca</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notes"><i class="fa fa-check"></i><b>1.3</b> Notes:</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#installation-of-modules"><i class="fa fa-check"></i><b>1.3.1</b> Installation of modules</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#airtable-access"><i class="fa fa-check"></i><b>1.3.2</b> Airtable access:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>2</b> MAG Catalogue</a>
<ul>
<li class="chapter" data-level="2.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#dereplicated-genomes"><i class="fa fa-check"></i><b>2.1</b> Dereplicated genomes</a></li>
<li class="chapter" data-level="2.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-tree"><i class="fa fa-check"></i><b>2.2</b> Genome (MAGs) tree</a></li>
<li class="chapter" data-level="2.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-taxonomy"><i class="fa fa-check"></i><b>2.3</b> Genome (MAGs) taxonomy</a></li>
<li class="chapter" data-level="2.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-quality"><i class="fa fa-check"></i><b>2.4</b> Genome (MAGs) quality</a></li>
<li class="chapter" data-level="2.5" data-path="mag-catalogue.html"><a href="mag-catalogue.html#merged-genome-mags-taxonomy-quality"><i class="fa fa-check"></i><b>2.5</b> Merged Genome (MAGs) taxonomy &amp; quality</a></li>
<li class="chapter" data-level="2.6" data-path="mag-catalogue.html"><a href="mag-catalogue.html#factoring-of-taxonomic-levels"><i class="fa fa-check"></i><b>2.6</b> Factoring of taxonomic levels</a></li>
<li class="chapter" data-level="2.7" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-functional-annotations"><i class="fa fa-check"></i><b>2.7</b> Genome (MAGs) functional annotations</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#distil-functional-annotations"><i class="fa fa-check"></i><b>2.7.1</b> Distil functional annotations</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="mag-catalogue.html"><a href="mag-catalogue.html#bacterial-phyla-color-scheme-data"><i class="fa fa-check"></i><b>2.8</b> Bacterial phyla color scheme data</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-color-schemes-for-phyla-and-orders"><i class="fa fa-check"></i><b>2.8.1</b> Define color schemes for phyla and orders</a></li>
<li class="chapter" data-level="2.8.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-plotting-settings"><i class="fa fa-check"></i><b>2.8.2</b> Define plotting settings</a></li>
<li class="chapter" data-level="2.8.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-treatment-color-schemes"><i class="fa fa-check"></i><b>2.8.3</b> Define treatment color schemes</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="mag-catalogue.html"><a href="mag-catalogue.html#save-working-objects"><i class="fa fa-check"></i><b>2.9</b> Save working objects</a></li>
<li class="chapter" data-level="2.10" data-path="mag-catalogue.html"><a href="mag-catalogue.html#validation"><i class="fa fa-check"></i><b>2.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html"><i class="fa fa-check"></i><b>3</b> Micro-sample Metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#load-and-process-micro-sample-metadata-from-airtable"><i class="fa fa-check"></i><b>3.1</b> Load and process micro-sample metadata from Airtable</a></li>
<li class="chapter" data-level="3.2" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#save-working-objects-1"><i class="fa fa-check"></i><b>3.2</b> Save working objects</a></li>
<li class="chapter" data-level="3.3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#validation-1"><i class="fa fa-check"></i><b>3.3</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html"><i class="fa fa-check"></i><b>4</b> Micro-sample Counts</a>
<ul>
<li class="chapter" data-level="4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#load-required-data"><i class="fa fa-check"></i><b>4.1</b> Load required data</a></li>
<li class="chapter" data-level="4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#bacterial-reads-counts"><i class="fa fa-check"></i><b>4.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#genome-covered-bases"><i class="fa fa-check"></i><b>4.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="4.4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#filter-and-normalise-data"><i class="fa fa-check"></i><b>4.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#coverage-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Coverage filtering</a></li>
<li class="chapter" data-level="4.4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-genome-count-table"><i class="fa fa-check"></i><b>4.4.2</b> Generate genome count table</a></li>
<li class="chapter" data-level="4.4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-length-corrected-read-count-tables"><i class="fa fa-check"></i><b>4.4.3</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#alpha-diversity-calculations"><i class="fa fa-check"></i><b>4.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="4.6" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#save-working-objects-2"><i class="fa fa-check"></i><b>4.6</b> Save working objects</a></li>
<li class="chapter" data-level="4.7" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#validation-2"><i class="fa fa-check"></i><b>4.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>5</b> Micro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#load-required-data-1"><i class="fa fa-check"></i><b>5.1</b> Load required data</a></li>
<li class="chapter" data-level="5.2" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#fastp-sequencing-statistics"><i class="fa fa-check"></i><b>5.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="5.3" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas"><i class="fa fa-check"></i><b>5.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="5.4" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#bacterial-mapped-reads"><i class="fa fa-check"></i><b>5.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="5.5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table"><i class="fa fa-check"></i><b>5.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="5.6" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#sanity-checks"><i class="fa fa-check"></i><b>5.6</b> Sanity checks</a></li>
<li class="chapter" data-level="5.7" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics"><i class="fa fa-check"></i><b>5.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="5.8" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#create-plot_data_stats"><i class="fa fa-check"></i><b>5.8</b> Create plot_data_stats</a></li>
<li class="chapter" data-level="5.9" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#save-working-objects-3"><i class="fa fa-check"></i><b>5.9</b> Save working objects</a></li>
<li class="chapter" data-level="5.10" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#validation-3"><i class="fa fa-check"></i><b>5.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html"><i class="fa fa-check"></i><b>6</b> Macro-sample Metadata</a>
<ul>
<li class="chapter" data-level="6.1" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#save-working-objects-4"><i class="fa fa-check"></i><b>6.1</b> Save working objects</a></li>
<li class="chapter" data-level="6.2" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#validation-4"><i class="fa fa-check"></i><b>6.2</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html"><i class="fa fa-check"></i><b>7</b> Macro-sample Counts</a>
<ul>
<li class="chapter" data-level="7.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#load-required-data-2"><i class="fa fa-check"></i><b>7.1</b> Load required data</a></li>
<li class="chapter" data-level="7.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#bacterial-reads-counts-1"><i class="fa fa-check"></i><b>7.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="7.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#genome-covered-bases-1"><i class="fa fa-check"></i><b>7.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="7.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#filter-and-normalise-data-1"><i class="fa fa-check"></i><b>7.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-coverage-table"><i class="fa fa-check"></i><b>7.4.1</b> Generate coverage table</a></li>
<li class="chapter" data-level="7.4.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#coverage-filtering-1"><i class="fa fa-check"></i><b>7.4.2</b> Coverage filtering</a></li>
<li class="chapter" data-level="7.4.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-genome-count-table-1"><i class="fa fa-check"></i><b>7.4.3</b> Generate genome count table</a></li>
<li class="chapter" data-level="7.4.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-length-corrected-read-count-tables-1"><i class="fa fa-check"></i><b>7.4.4</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#alpha-diversity-calculations-1"><i class="fa fa-check"></i><b>7.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="7.6" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#save-working-objects-5"><i class="fa fa-check"></i><b>7.6</b> Save working objects</a></li>
<li class="chapter" data-level="7.7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#validation-5"><i class="fa fa-check"></i><b>7.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>8</b> Macro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#load-required-data-3"><i class="fa fa-check"></i><b>8.1</b> Load required data</a></li>
<li class="chapter" data-level="8.2" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#fastp-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="8.3" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas-1"><i class="fa fa-check"></i><b>8.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="8.4" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#bacterial-mapped-reads-1"><i class="fa fa-check"></i><b>8.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="8.5" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table-1"><i class="fa fa-check"></i><b>8.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="8.6" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#sanity-checks-1"><i class="fa fa-check"></i><b>8.6</b> Sanity checks</a></li>
<li class="chapter" data-level="8.7" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="8.8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#create-plot_data_stats_macro"><i class="fa fa-check"></i><b>8.8</b> Create plot_data_stats_macro</a></li>
<li class="chapter" data-level="8.9" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#save-working-objects-6"><i class="fa fa-check"></i><b>8.9</b> Save working objects</a></li>
<li class="chapter" data-level="8.10" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#validation-6"><i class="fa fa-check"></i><b>8.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html"><i class="fa fa-check"></i><b>9</b> Metabolomics Data Preparation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-metabolomics-data"><i class="fa fa-check"></i><b>9.1</b> Load metabolomics data</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-abundance-values-and-metabolites-metadata"><i class="fa fa-check"></i><b>9.1.1</b> Load abundance values and metabolites metadata</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#classify-samples-by-type"><i class="fa fa-check"></i><b>9.2</b> Classify samples by type</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#identify-sample-columns-and-classify-by-type"><i class="fa fa-check"></i><b>9.2.1</b> Identify sample columns and classify by type</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#create-dataframes-for-each-sample-type"><i class="fa fa-check"></i><b>9.3</b> Create dataframes for each sample type</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hn-histomonas-chicken-host-tissue"><i class="fa fa-check"></i><b>9.3.1</b> HN: Histomonas Chicken, host tissue</a></li>
<li class="chapter" data-level="9.3.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hm-histomonas-chicken-digesta"><i class="fa fa-check"></i><b>9.3.2</b> HM: Histomonas Chicken, digesta</a></li>
<li class="chapter" data-level="9.3.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mn-histomonas-turkey-host-tissue"><i class="fa fa-check"></i><b>9.3.3</b> MN: Histomonas Turkey, host tissue</a></li>
<li class="chapter" data-level="9.3.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mm-histomonas-turkey-digesta"><i class="fa fa-check"></i><b>9.3.4</b> MM: Histomonas Turkey, digesta</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-processed-data"><i class="fa fa-check"></i><b>9.4</b> Save processed data</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-metabolomics-dataframes"><i class="fa fa-check"></i><b>9.4.1</b> Save metabolomics dataframes</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validation-7"><i class="fa fa-check"></i><b>9.5</b> Validation</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validate-metabolomics-data"><i class="fa fa-check"></i><b>9.5.1</b> Validate metabolomics data</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#notes-on-metabolites-metadata"><i class="fa fa-check"></i><b>9.6</b> Notes on metabolites metadata</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#available-metadata-fields"><i class="fa fa-check"></i><b>9.6.1</b> Available metadata fields:</a></li>
<li class="chapter" data-level="9.6.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#suggested-uses"><i class="fa fa-check"></i><b>9.6.2</b> Suggested uses:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html"><i class="fa fa-check"></i><b>10</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-quality"><i class="fa fa-check"></i><b>10.1</b> Genome quality</a></li>
<li class="chapter" data-level="10.2" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours"><i class="fa fa-check"></i><b>10.2</b> MAG catalogue phyla and order colours</a></li>
<li class="chapter" data-level="10.3" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours-and-genera-number"><i class="fa fa-check"></i><b>10.3</b> MAG catalogue phyla and order colours, and genera number</a></li>
<li class="chapter" data-level="10.4" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot"><i class="fa fa-check"></i><b>10.4</b> Genome phylogeny plot</a></li>
<li class="chapter" data-level="10.5" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot-without-legend"><i class="fa fa-check"></i><b>10.5</b> Genome phylogeny plot without legend</a></li>
<li class="chapter" data-level="10.6" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-overview"><i class="fa fa-check"></i><b>10.6</b> Functional overview</a></li>
<li class="chapter" data-level="10.7" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-distances"><i class="fa fa-check"></i><b>10.7</b> Functional distances</a></li>
<li class="chapter" data-level="10.8" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-ordination"><i class="fa fa-check"></i><b>10.8</b> Functional ordination</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="histomonas-presence.html"><a href="histomonas-presence.html"><i class="fa fa-check"></i><b>11</b> Histomonas Presence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="histomonas-presence.html"><a href="histomonas-presence.html#macro"><i class="fa fa-check"></i><b>11.1</b> Macro</a></li>
<li class="chapter" data-level="11.2" data-path="histomonas-presence.html"><a href="histomonas-presence.html#micro"><i class="fa fa-check"></i><b>11.2</b> Micro</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html"><i class="fa fa-check"></i><b>12</b> Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#prepare-tidy-tables-for-plotting"><i class="fa fa-check"></i><b>12.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="12.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#macro-samples"><i class="fa fa-check"></i><b>12.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases-reads"><i class="fa fa-check"></i><b>12.2.1</b> Number of bases &amp; reads</a></li>
<li class="chapter" data-level="12.2.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped"><i class="fa fa-check"></i><b>12.2.2</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.2.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#composition-of-reads-vs.-references"><i class="fa fa-check"></i><b>12.2.3</b> Composition of reads vs. references</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#micro-samples"><i class="fa fa-check"></i><b>12.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases"><i class="fa fa-check"></i><b>12.3.1</b> Number of bases</a></li>
<li class="chapter" data-level="12.3.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-reads-after-trimming"><i class="fa fa-check"></i><b>12.3.2</b> Number of reads after trimming</a></li>
<li class="chapter" data-level="12.3.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped-1"><i class="fa fa-check"></i><b>12.3.3</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.3.4" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts"><i class="fa fa-check"></i><b>12.3.4</b> Collection attempts</a></li>
<li class="chapter" data-level="12.3.5" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts-vs-filtering-retention"><i class="fa fa-check"></i><b>12.3.5</b> Collection attempts vs filtering retention</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="bacterial-load.html"><a href="bacterial-load.html"><i class="fa fa-check"></i><b>13</b> Bacterial Load</a>
<ul>
<li class="chapter" data-level="13.1" data-path="bacterial-load.html"><a href="bacterial-load.html#macro-samples-1"><i class="fa fa-check"></i><b>13.1</b> Macro-samples</a></li>
<li class="chapter" data-level="13.2" data-path="bacterial-load.html"><a href="bacterial-load.html#micro-samples-1"><i class="fa fa-check"></i><b>13.2</b> Micro-samples</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="14.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#prepare-tidy-tables-for-plotting-1"><i class="fa fa-check"></i><b>14.1</b> Prepare tidy tables for plotting</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-tidy-tables"><i class="fa fa-check"></i><b>14.1.1</b> Alpha diversity tidy tables</a></li>
<li class="chapter" data-level="14.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#genome-counts-tidy-tables"><i class="fa fa-check"></i><b>14.1.2</b> Genome counts tidy tables</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#macro-samples-2"><i class="fa fa-check"></i><b>14.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div"><i class="fa fa-check"></i><b>14.2.1</b> Alpha-div</a></li>
<li class="chapter" data-level="14.2.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size"><i class="fa fa-check"></i><b>14.2.2</b> Alpha-div vs genome size</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#micro-samples-2"><i class="fa fa-check"></i><b>14.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-1"><i class="fa fa-check"></i><b>14.3.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="14.3.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size-1"><i class="fa fa-check"></i><b>14.3.2</b> Alpha-div vs genome size</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html"><i class="fa fa-check"></i><b>15</b> Beta Diversity - Macro Samples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#prepare-tidy-tables-for-plotting-2"><i class="fa fa-check"></i><b>15.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="15.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-level"><i class="fa fa-check"></i><b>15.2</b> Species level</a></li>
<li class="chapter" data-level="15.3" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#genus-level"><i class="fa fa-check"></i><b>15.3</b> Genus level</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#permanova"><i class="fa fa-check"></i><b>15.3.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="15.3.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-representative-genomes-overlap"><i class="fa fa-check"></i><b>15.3.2</b> Species representative genomes overlap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html"><i class="fa fa-check"></i><b>16</b> Beta Diversity - Micro Samples</a>
<ul>
<li class="chapter" data-level="16.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#prepare-tidy-tables-for-plotting-3"><i class="fa fa-check"></i><b>16.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="16.2" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#species-level-1"><i class="fa fa-check"></i><b>16.2</b> Species level</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#permanova-1"><i class="fa fa-check"></i><b>16.2.1</b> PERMANOVA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html"><i class="fa fa-check"></i><b>17</b> Beta Diversity - Macro vs Micro Comparison</a>
<ul>
<li class="chapter" data-level="17.1" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#genomes-overlap"><i class="fa fa-check"></i><b>17.1</b> Genomes overlap</a></li>
<li class="chapter" data-level="17.2" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#micro-vs-macro-genus-level"><i class="fa fa-check"></i><b>17.2</b> Micro vs Macro Genus level</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html"><i class="fa fa-check"></i><b>18</b> Spatial Analysis - Micro Samples</a>
<ul>
<li class="chapter" data-level="18.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#functions"><i class="fa fa-check"></i><b>18.1</b> Functions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#spatial_cryosections"><i class="fa fa-check"></i><b>18.1.1</b> spatial_cryosections</a></li>
<li class="chapter" data-level="18.1.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#additive-diversity-partitioning"><i class="fa fa-check"></i><b>18.1.2</b> Additive diversity partitioning</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#load-data"><i class="fa fa-check"></i><b>18.2</b> Load data</a></li>
<li class="chapter" data-level="18.3" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#extended-rlq-analysis"><i class="fa fa-check"></i><b>18.3</b> Extended RLQ analysis</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html"><i class="fa fa-check"></i><b>19</b> Export Data for MicrobeTag</a>
<ul>
<li class="chapter" data-level="19.1" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#load-required-data-4"><i class="fa fa-check"></i><b>19.1</b> Load required data</a></li>
<li class="chapter" data-level="19.2" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-sample-metadata-macro-samples"><i class="fa fa-check"></i><b>19.2</b> Export sample metadata (macro samples)</a></li>
<li class="chapter" data-level="19.3" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-metabolites-data-mm-samples"><i class="fa fa-check"></i><b>19.3</b> Export metabolites data (MM samples)</a></li>
<li class="chapter" data-level="19.4" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-metadata"><i class="fa fa-check"></i><b>19.4</b> Export genome metadata</a></li>
<li class="chapter" data-level="19.5" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-counts-macro-samples-filtered"><i class="fa fa-check"></i><b>19.5</b> Export genome counts (macro samples, filtered)</a></li>
<li class="chapter" data-level="19.6" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#summary"><i class="fa fa-check"></i><b>19.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-versions.html"><a href="package-versions.html"><i class="fa fa-check"></i><b>20</b> Package Versions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">3D’omics | Histomonas Turkeys</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mag-catalogue" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> MAG Catalogue<a href="mag-catalogue.html#mag-catalogue" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Relevant metadata of genomes are fetched and merged into one genome metadata object for downstream analyses.</p>
<p>Location on ERDA:
ID: glS5qQbDc2
Path: 3D-omics/references/REF0029-m_mg_hybrid/
Access: read
genome_taxonomy: <a href="https://sid.erda.dk/share_redirect/glS5qQbDc2/gtdbtk.summary.tsv" class="uri">https://sid.erda.dk/share_redirect/glS5qQbDc2/gtdbtk.summary.tsv</a>
genome_quality: <a href="https://sid.erda.dk/share_redirect/glS5qQbDc2/checkm2.quality_report.tsv" class="uri">https://sid.erda.dk/share_redirect/glS5qQbDc2/checkm2.quality_report.tsv</a>
genome_tree: <a href="https://sid.erda.dk/share_redirect/glS5qQbDc2/gtdbtk.backbone.bac120.classify.tree" class="uri">https://sid.erda.dk/share_redirect/glS5qQbDc2/gtdbtk.backbone.bac120.classify.tree</a>
genome_annotations: <a href="https://sid.erda.dk/share_redirect/glS5qQbDc2/dram.annotations.tsv.gz" class="uri">https://sid.erda.dk/share_redirect/glS5qQbDc2/dram.annotations.tsv.gz</a></p>
<p>The number of genomes in these files should match the number of genomes in the count tables, after accounting for the appropriate dereplication at species level.</p>
<div id="dereplicated-genomes" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Dereplicated genomes<a href="mag-catalogue.html#dereplicated-genomes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Bins were initially dereplicated at 99%, and annotations generated for all resulting MAGs.
A second dereplication was performed at 95% to yield species-representative genomes. This analysis is performed on the 95% subset.
This is how to extract the names of the genomes that remain in the 95% dereplication:
1. Download the file from ERDA: 3D-omics/references/REF0029-m_mg_hybrid/drep.0.95.fa.gz
2. Save it in a directory, and navigate into the directory from the command line.
3. Execute the 3 commands:
gunzip drep.0.95.fa.gz
grep “^&gt;” drep.0.95.fa | sed ‘s/&gt;//’ &gt; genome_names.tsv
cut -d’@’ -f1 genome_names.tsv | sort -u &gt; unique_genome_names.tsv</p>
<p>Load the list of dereplicated genome names</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb8-1"><a href="mag-catalogue.html#cb8-1" tabindex="-1"></a>selected_genomes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/MAG_catalogue/unique_genome_names.tsv&quot;</span>, <span class="at">col_names =</span> <span class="st">&quot;genomes&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="mag-catalogue.html#cb8-2" tabindex="-1"></a>  <span class="fu">pull</span>(genomes)</span></code></pre></div>
<p>The dereplication yielded 527 species-representative genomes.</p>
</div>
<div id="genome-mags-tree" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Genome (MAGs) tree<a href="mag-catalogue.html#genome-mags-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is the raw tree generated by GTDBtk, which needs to be pruned to obtain the phylogenetic tree of the genomes.
Note that the archaeal tree is only generated if any archaeans are detected among the genomes.</p>
<p>Load and prune the phylogenetic tree to include only selected genomes</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb9-1"><a href="mag-catalogue.html#cb9-1" tabindex="-1"></a>genome_tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;data/MAG_catalogue/gtdbtk.backbone.bac120.classify.tree&quot;</span>)</span>
<span id="cb9-2"><a href="mag-catalogue.html#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="mag-catalogue.html#cb9-3" tabindex="-1"></a><span class="co"># remove single quotes in MAG names</span></span>
<span id="cb9-4"><a href="mag-catalogue.html#cb9-4" tabindex="-1"></a>genome_tree<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(genome_tree<span class="sc">$</span>tip.label, <span class="st">&quot;&#39;&quot;</span>, <span class="st">&quot;&quot;</span>) </span>
<span id="cb9-5"><a href="mag-catalogue.html#cb9-5" tabindex="-1"></a></span>
<span id="cb9-6"><a href="mag-catalogue.html#cb9-6" tabindex="-1"></a></span>
<span id="cb9-7"><a href="mag-catalogue.html#cb9-7" tabindex="-1"></a><span class="co"># There is a genome present in the selected genomes but absent in the tree. This causes the keep.tree to crash.</span></span>
<span id="cb9-8"><a href="mag-catalogue.html#cb9-8" tabindex="-1"></a>missing_genomes <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(selected_genomes, genome_tree<span class="sc">$</span>tip.label)</span>
<span id="cb9-9"><a href="mag-catalogue.html#cb9-9" tabindex="-1"></a>missing_genomes</span>
<span id="cb9-10"><a href="mag-catalogue.html#cb9-10" tabindex="-1"></a><span class="co"># The missing genome (from the tree): MPB:bin_000147</span></span>
<span id="cb9-11"><a href="mag-catalogue.html#cb9-11" tabindex="-1"></a></span>
<span id="cb9-12"><a href="mag-catalogue.html#cb9-12" tabindex="-1"></a><span class="co"># In the selected_genomes list, select only genomes present in the tree</span></span>
<span id="cb9-13"><a href="mag-catalogue.html#cb9-13" tabindex="-1"></a>selected_genomes <span class="ot">&lt;-</span> selected_genomes[selected_genomes <span class="sc">%in%</span> genome_tree<span class="sc">$</span>tip.label]</span>
<span id="cb9-14"><a href="mag-catalogue.html#cb9-14" tabindex="-1"></a></span>
<span id="cb9-15"><a href="mag-catalogue.html#cb9-15" tabindex="-1"></a><span class="co"># Then, in the tree, select only the selected_genomes at 95% dereplication</span></span>
<span id="cb9-16"><a href="mag-catalogue.html#cb9-16" tabindex="-1"></a>genome_tree <span class="ot">&lt;-</span> <span class="fu">keep.tip</span>(genome_tree, <span class="at">tip =</span> selected_genomes) </span></code></pre></div>
</div>
<div id="genome-mags-taxonomy" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Genome (MAGs) taxonomy<a href="mag-catalogue.html#genome-mags-taxonomy" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is the raw taxonomy table generated by GTDBtk, which is simplified for downstream analyses.</p>
<p>Load and process genome taxonomy data</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb10-1"><a href="mag-catalogue.html#cb10-1" tabindex="-1"></a>genome_taxonomy <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/MAG_catalogue/gtdbtk.summary.tsv&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="mag-catalogue.html#cb10-2" tabindex="-1"></a>  <span class="co"># rename first column</span></span>
<span id="cb10-3"><a href="mag-catalogue.html#cb10-3" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">genome =</span> user_genome) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="mag-catalogue.html#cb10-4" tabindex="-1"></a>  <span class="co"># remove .fa from the end of each genome name</span></span>
<span id="cb10-5"><a href="mag-catalogue.html#cb10-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genome =</span> <span class="fu">str_replace_all</span>(genome, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.fa&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="mag-catalogue.html#cb10-6" tabindex="-1"></a>  <span class="co"># select the species-representative genomes</span></span>
<span id="cb10-7"><a href="mag-catalogue.html#cb10-7" tabindex="-1"></a>  <span class="fu">filter</span>(genome <span class="sc">%in%</span> selected_genomes) <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="mag-catalogue.html#cb10-8" tabindex="-1"></a>  <span class="co"># separate column &#39;classification&#39; into columns for the different taxonomic levels</span></span>
<span id="cb10-9"><a href="mag-catalogue.html#cb10-9" tabindex="-1"></a>  <span class="fu">separate</span>(classification, <span class="fu">c</span>(<span class="st">&quot;domain&quot;</span>, <span class="st">&quot;phylum&quot;</span>, <span class="st">&quot;class&quot;</span>, <span class="st">&quot;order&quot;</span>, <span class="st">&quot;family&quot;</span>, </span>
<span id="cb10-10"><a href="mag-catalogue.html#cb10-10" tabindex="-1"></a>                             <span class="st">&quot;genus&quot;</span>, <span class="st">&quot;species&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-11"><a href="mag-catalogue.html#cb10-11" tabindex="-1"></a>  <span class="co"># remove redundant letters before taxon names</span></span>
<span id="cb10-12"><a href="mag-catalogue.html#cb10-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(domain<span class="sc">:</span>species, <span class="sc">~</span> <span class="fu">str_sub</span>(., <span class="dv">4</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb10-13"><a href="mag-catalogue.html#cb10-13" tabindex="-1"></a>  <span class="co"># select the relevant taxonomic columns</span></span>
<span id="cb10-14"><a href="mag-catalogue.html#cb10-14" tabindex="-1"></a>  <span class="fu">select</span>(genome, domain, phylum, class, order, family, genus, species) <span class="sc">%&gt;%</span></span>
<span id="cb10-15"><a href="mag-catalogue.html#cb10-15" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-16"><a href="mag-catalogue.html#cb10-16" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">if_else</span>(</span>
<span id="cb10-17"><a href="mag-catalogue.html#cb10-17" tabindex="-1"></a>      <span class="fu">is.na</span>(species) <span class="sc">|</span> species <span class="sc">==</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb10-18"><a href="mag-catalogue.html#cb10-18" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">&quot;unclassified_&quot;</span>, genus),</span>
<span id="cb10-19"><a href="mag-catalogue.html#cb10-19" tabindex="-1"></a>      species</span>
<span id="cb10-20"><a href="mag-catalogue.html#cb10-20" tabindex="-1"></a>    )</span>
<span id="cb10-21"><a href="mag-catalogue.html#cb10-21" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb10-22"><a href="mag-catalogue.html#cb10-22" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-23"><a href="mag-catalogue.html#cb10-23" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">if_else</span>(</span>
<span id="cb10-24"><a href="mag-catalogue.html#cb10-24" tabindex="-1"></a>      <span class="fu">is.na</span>(species) <span class="sc">|</span> species <span class="sc">==</span> <span class="st">&quot;unclassified_&quot;</span>,</span>
<span id="cb10-25"><a href="mag-catalogue.html#cb10-25" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">&quot;unclassified_&quot;</span>, family),</span>
<span id="cb10-26"><a href="mag-catalogue.html#cb10-26" tabindex="-1"></a>      species</span>
<span id="cb10-27"><a href="mag-catalogue.html#cb10-27" tabindex="-1"></a>    )</span>
<span id="cb10-28"><a href="mag-catalogue.html#cb10-28" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb10-29"><a href="mag-catalogue.html#cb10-29" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb10-30"><a href="mag-catalogue.html#cb10-30" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">if_else</span>(</span>
<span id="cb10-31"><a href="mag-catalogue.html#cb10-31" tabindex="-1"></a>      <span class="fu">is.na</span>(genus) <span class="sc">|</span> genus <span class="sc">==</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb10-32"><a href="mag-catalogue.html#cb10-32" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">&quot;unclassified_&quot;</span>, family),</span>
<span id="cb10-33"><a href="mag-catalogue.html#cb10-33" tabindex="-1"></a>      genus</span>
<span id="cb10-34"><a href="mag-catalogue.html#cb10-34" tabindex="-1"></a>    )</span>
<span id="cb10-35"><a href="mag-catalogue.html#cb10-35" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="genome-mags-quality" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Genome (MAGs) quality<a href="mag-catalogue.html#genome-mags-quality" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Quality properties of the genomes.</p>
<p>Load and process genome quality metrics</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb11-1"><a href="mag-catalogue.html#cb11-1" tabindex="-1"></a>genome_quality <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/MAG_catalogue/checkm2.quality_report.tsv&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="mag-catalogue.html#cb11-2" tabindex="-1"></a>  <span class="co"># rename first column</span></span>
<span id="cb11-3"><a href="mag-catalogue.html#cb11-3" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">genome =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="mag-catalogue.html#cb11-4" tabindex="-1"></a>  <span class="co"># remove .fa from the end of each genome name</span></span>
<span id="cb11-5"><a href="mag-catalogue.html#cb11-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genome =</span> <span class="fu">str_replace_all</span>(genome, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.fa&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="mag-catalogue.html#cb11-6" tabindex="-1"></a>  <span class="co"># select the species-representative genomes</span></span>
<span id="cb11-7"><a href="mag-catalogue.html#cb11-7" tabindex="-1"></a>  <span class="fu">filter</span>(genome <span class="sc">%in%</span> selected_genomes) <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="mag-catalogue.html#cb11-8" tabindex="-1"></a>  <span class="co"># select relevant columns and rename</span></span>
<span id="cb11-9"><a href="mag-catalogue.html#cb11-9" tabindex="-1"></a>  <span class="fu">select</span>(genome, Completeness, Contamination, Coding_Density, Genome_Size, Contig_N50) <span class="sc">%&gt;%</span> </span>
<span id="cb11-10"><a href="mag-catalogue.html#cb11-10" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">completeness =</span> Completeness, </span>
<span id="cb11-11"><a href="mag-catalogue.html#cb11-11" tabindex="-1"></a>         <span class="at">contamination =</span> Contamination, </span>
<span id="cb11-12"><a href="mag-catalogue.html#cb11-12" tabindex="-1"></a>         <span class="at">coding_density =</span> Coding_Density, </span>
<span id="cb11-13"><a href="mag-catalogue.html#cb11-13" tabindex="-1"></a>         <span class="at">length =</span> Genome_Size,</span>
<span id="cb11-14"><a href="mag-catalogue.html#cb11-14" tabindex="-1"></a>         <span class="at">N50_length =</span> Contig_N50) <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="mag-catalogue.html#cb11-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">circularity =</span> N50_length<span class="sc">/</span>length) <span class="sc">%&gt;%</span></span>
<span id="cb11-16"><a href="mag-catalogue.html#cb11-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">circularity =</span> <span class="fu">ifelse</span>(circularity <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="st">&quot;1&quot;</span>, <span class="st">&quot;0&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-17"><a href="mag-catalogue.html#cb11-17" tabindex="-1"></a>  <span class="co"># update completeness and contamination for genomes with circularity of 1</span></span>
<span id="cb11-18"><a href="mag-catalogue.html#cb11-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">completeness =</span> <span class="fu">ifelse</span>(circularity <span class="sc">==</span> <span class="st">&quot;1&quot;</span>, <span class="dv">100</span>, completeness),</span>
<span id="cb11-19"><a href="mag-catalogue.html#cb11-19" tabindex="-1"></a>         <span class="at">contamination =</span> <span class="fu">ifelse</span>(circularity <span class="sc">==</span> <span class="st">&quot;1&quot;</span>, <span class="dv">0</span>, contamination)) <span class="sc">%&gt;%</span>   </span>
<span id="cb11-20"><a href="mag-catalogue.html#cb11-20" tabindex="-1"></a><span class="co"># estimate ratio of: genome length / median bacterial genome length</span></span>
<span id="cb11-21"><a href="mag-catalogue.html#cb11-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">length_ratio =</span> length<span class="sc">/</span><span class="fu">median</span>(length, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb11-22"><a href="mag-catalogue.html#cb11-22" tabindex="-1"></a></span>
<span id="cb11-23"><a href="mag-catalogue.html#cb11-23" tabindex="-1"></a></span>
<span id="cb11-24"><a href="mag-catalogue.html#cb11-24" tabindex="-1"></a><span class="co"># How many genomes are circularised?</span></span>
<span id="cb11-25"><a href="mag-catalogue.html#cb11-25" tabindex="-1"></a><span class="fu">sum</span>(genome_quality<span class="sc">$</span>circularity <span class="sc">==</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="merged-genome-mags-taxonomy-quality" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Merged Genome (MAGs) taxonomy &amp; quality<a href="mag-catalogue.html#merged-genome-mags-taxonomy-quality" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Merge taxonomy, length and quality information.</p>
<p>Combine taxonomy and quality data into a single metadata object</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb12-1"><a href="mag-catalogue.html#cb12-1" tabindex="-1"></a><span class="co"># combine taxonomy &amp; quality into one df</span></span>
<span id="cb12-2"><a href="mag-catalogue.html#cb12-2" tabindex="-1"></a>genome_metadata <span class="ot">&lt;-</span> genome_taxonomy <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="mag-catalogue.html#cb12-3" tabindex="-1"></a>  <span class="fu">left_join</span>(genome_quality, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome))</span>
<span id="cb12-4"><a href="mag-catalogue.html#cb12-4" tabindex="-1"></a><span class="co"># replace - with _ in the order names</span></span>
<span id="cb12-5"><a href="mag-catalogue.html#cb12-5" tabindex="-1"></a>genome_metadata<span class="sc">$</span>order <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>, genome_metadata<span class="sc">$</span>order)</span>
<span id="cb12-6"><a href="mag-catalogue.html#cb12-6" tabindex="-1"></a><span class="co"># order the genome metadata rows based on the order of the tree</span></span>
<span id="cb12-7"><a href="mag-catalogue.html#cb12-7" tabindex="-1"></a>genome_metadata <span class="ot">&lt;-</span> genome_metadata[<span class="fu">match</span>(genome_tree<span class="sc">$</span>tip.label, genome_metadata<span class="sc">$</span>genome), ]</span></code></pre></div>
</div>
<div id="factoring-of-taxonomic-levels" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Factoring of taxonomic levels<a href="mag-catalogue.html#factoring-of-taxonomic-levels" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create metadata summaries for phylum and order levels</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb13-1"><a href="mag-catalogue.html#cb13-1" tabindex="-1"></a>phylum_metadata <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="mag-catalogue.html#cb13-2" tabindex="-1"></a>  <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span>  </span>
<span id="cb13-3"><a href="mag-catalogue.html#cb13-3" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb13-4"><a href="mag-catalogue.html#cb13-4" tabindex="-1"></a>    <span class="co"># Count the microsamples</span></span>
<span id="cb13-5"><a href="mag-catalogue.html#cb13-5" tabindex="-1"></a>    <span class="st">`</span><span class="at">number_of_genera</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">n_distinct</span>(genus),  </span>
<span id="cb13-6"><a href="mag-catalogue.html#cb13-6" tabindex="-1"></a>    <span class="co"># Collect unique values separated by commas</span></span>
<span id="cb13-7"><a href="mag-catalogue.html#cb13-7" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(genus), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb13-8"><a href="mag-catalogue.html#cb13-8" tabindex="-1"></a>    <span class="at">phylum =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(phylum), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb13-9"><a href="mag-catalogue.html#cb13-9" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-10"><a href="mag-catalogue.html#cb13-10" tabindex="-1"></a>  <span class="fu">select</span>(phylum, number_of_genera, genus) <span class="sc">%&gt;%</span></span>
<span id="cb13-11"><a href="mag-catalogue.html#cb13-11" tabindex="-1"></a>  <span class="co"># Preserve original order</span></span>
<span id="cb13-12"><a href="mag-catalogue.html#cb13-12" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(phylum, <span class="fu">unique</span>(genome_metadata<span class="sc">$</span>phylum)))  </span>
<span id="cb13-13"><a href="mag-catalogue.html#cb13-13" tabindex="-1"></a></span>
<span id="cb13-14"><a href="mag-catalogue.html#cb13-14" tabindex="-1"></a>order_metadata <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb13-15"><a href="mag-catalogue.html#cb13-15" tabindex="-1"></a>  <span class="fu">group_by</span>(order) <span class="sc">%&gt;%</span>  </span>
<span id="cb13-16"><a href="mag-catalogue.html#cb13-16" tabindex="-1"></a>  <span class="fu">summarize</span>(</span>
<span id="cb13-17"><a href="mag-catalogue.html#cb13-17" tabindex="-1"></a>    <span class="co"># Count the microsamples</span></span>
<span id="cb13-18"><a href="mag-catalogue.html#cb13-18" tabindex="-1"></a>    <span class="st">`</span><span class="at">number_of_genera</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">n_distinct</span>(genus),  </span>
<span id="cb13-19"><a href="mag-catalogue.html#cb13-19" tabindex="-1"></a>    <span class="co"># Collect unique values separated by commas</span></span>
<span id="cb13-20"><a href="mag-catalogue.html#cb13-20" tabindex="-1"></a>    <span class="at">genus =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(genus), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb13-21"><a href="mag-catalogue.html#cb13-21" tabindex="-1"></a>    <span class="at">phylum =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(phylum), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb13-22"><a href="mag-catalogue.html#cb13-22" tabindex="-1"></a>    <span class="at">class =</span> <span class="fu">paste</span>(<span class="fu">unique</span>(class), <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>),</span>
<span id="cb13-23"><a href="mag-catalogue.html#cb13-23" tabindex="-1"></a>    <span class="co"># Ensure the dataframe is ungrouped</span></span>
<span id="cb13-24"><a href="mag-catalogue.html#cb13-24" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-25"><a href="mag-catalogue.html#cb13-25" tabindex="-1"></a>  <span class="fu">select</span>(phylum, class, order, number_of_genera, genus) <span class="sc">%&gt;%</span></span>
<span id="cb13-26"><a href="mag-catalogue.html#cb13-26" tabindex="-1"></a>  <span class="co"># Preserve original order</span></span>
<span id="cb13-27"><a href="mag-catalogue.html#cb13-27" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(order, <span class="fu">unique</span>(genome_metadata<span class="sc">$</span>order)))  </span>
<span id="cb13-28"><a href="mag-catalogue.html#cb13-28" tabindex="-1"></a></span>
<span id="cb13-29"><a href="mag-catalogue.html#cb13-29" tabindex="-1"></a>phylum_level_vector <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(phylum_metadata<span class="sc">$</span>phylum))</span>
<span id="cb13-30"><a href="mag-catalogue.html#cb13-30" tabindex="-1"></a></span>
<span id="cb13-31"><a href="mag-catalogue.html#cb13-31" tabindex="-1"></a>order_level_vector <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(order_metadata<span class="sc">$</span>order))</span></code></pre></div>
</div>
<div id="genome-mags-functional-annotations" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Genome (MAGs) functional annotations<a href="mag-catalogue.html#genome-mags-functional-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is the raw annotation table generated by DRAM, which is used to generate GIFT data using distillR.</p>
<p>Load functional annotations from DRAM</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb14-1"><a href="mag-catalogue.html#cb14-1" tabindex="-1"></a>genome_annotations <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/MAG_catalogue/dram.annotations.tsv.gz&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="mag-catalogue.html#cb14-2" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gene =</span> <span class="dv">1</span>, <span class="at">genome =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="mag-catalogue.html#cb14-3" tabindex="-1"></a>  <span class="fu">filter</span>(genome <span class="sc">%in%</span> selected_genomes)</span></code></pre></div>
<div id="distil-functional-annotations" class="section level3 hasAnchor" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Distil functional annotations<a href="mag-catalogue.html#distil-functional-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Raw functional annotations are distilled into genome-inferred functional traits to generate biologically more meaningful functional traits for downstream analyses.</p>
<p>Distill functional annotations into GIFT data</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb15-1"><a href="mag-catalogue.html#cb15-1" tabindex="-1"></a>genome_gifts <span class="ot">&lt;-</span> <span class="fu">distill</span>(genome_annotations, GIFT_db, <span class="at">genomecol =</span> <span class="dv">2</span>, <span class="at">annotcol =</span> <span class="fu">c</span>(<span class="dv">9</span>, <span class="dv">10</span>, <span class="dv">19</span>))</span></code></pre></div>
</div>
</div>
<div id="bacterial-phyla-color-scheme-data" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Bacterial phyla color scheme data<a href="mag-catalogue.html#bacterial-phyla-color-scheme-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="define-color-schemes-for-phyla-and-orders" class="section level3 hasAnchor" number="2.8.1">
<h3><span class="header-section-number">2.8.1</span> Define color schemes for phyla and orders<a href="mag-catalogue.html#define-color-schemes-for-phyla-and-orders" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb16-1"><a href="mag-catalogue.html#cb16-1" tabindex="-1"></a>phylum_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Actinomycetota =</span> <span class="st">&quot;#346254&quot;</span>,</span>
<span id="cb16-2"><a href="mag-catalogue.html#cb16-2" tabindex="-1"></a>                   <span class="at">Bacillota =</span> <span class="st">&quot;#4a6ab7&quot;</span>,</span>
<span id="cb16-3"><a href="mag-catalogue.html#cb16-3" tabindex="-1"></a>                   <span class="at">Bacillota_A =</span> <span class="st">&quot;#8c1c47&quot;</span>,</span>
<span id="cb16-4"><a href="mag-catalogue.html#cb16-4" tabindex="-1"></a>                   <span class="at">Bacillota_B =</span> <span class="st">&quot;#644ca3&quot;</span>,</span>
<span id="cb16-5"><a href="mag-catalogue.html#cb16-5" tabindex="-1"></a>                   <span class="at">Bacteroidota =</span> <span class="st">&quot;#9c8464&quot;</span>,</span>
<span id="cb16-6"><a href="mag-catalogue.html#cb16-6" tabindex="-1"></a>                   <span class="at">Pseudomonadota =</span> <span class="st">&quot;#c49d4b&quot;</span>,</span>
<span id="cb16-7"><a href="mag-catalogue.html#cb16-7" tabindex="-1"></a>                   <span class="at">Verrucomicrobiota =</span> <span class="st">&quot;#462410&quot;</span>,</span>
<span id="cb16-8"><a href="mag-catalogue.html#cb16-8" tabindex="-1"></a>                   <span class="at">Cyanobacteriota =</span> <span class="st">&quot;#b45f06&quot;</span>)</span>
<span id="cb16-9"><a href="mag-catalogue.html#cb16-9" tabindex="-1"></a></span>
<span id="cb16-10"><a href="mag-catalogue.html#cb16-10" tabindex="-1"></a>order_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Mycobacteriales   =</span> <span class="st">&quot;#79b1a3&quot;</span>,</span>
<span id="cb16-11"><a href="mag-catalogue.html#cb16-11" tabindex="-1"></a>                  <span class="at">Coriobacteriales =</span>    <span class="st">&quot;#498a77&quot;</span>,</span>
<span id="cb16-12"><a href="mag-catalogue.html#cb16-12" tabindex="-1"></a>                  </span>
<span id="cb16-13"><a href="mag-catalogue.html#cb16-13" tabindex="-1"></a>                  <span class="at">Acholeplasmatales      =</span> <span class="st">&quot;#cfe2f3&quot;</span>,</span>
<span id="cb16-14"><a href="mag-catalogue.html#cb16-14" tabindex="-1"></a>                  <span class="at">CAJFEE01 =</span>    <span class="st">&quot;#a0c5e8&quot;</span>,</span>
<span id="cb16-15"><a href="mag-catalogue.html#cb16-15" tabindex="-1"></a>                  <span class="at">Erysipelotrichales =</span>  <span class="st">&quot;#7acef4&quot;</span>,</span>
<span id="cb16-16"><a href="mag-catalogue.html#cb16-16" tabindex="-1"></a>                  <span class="at">Haloplasmatales        =</span> <span class="st">&quot;#70c2e5&quot;</span>,</span>
<span id="cb16-17"><a href="mag-catalogue.html#cb16-17" tabindex="-1"></a>                  <span class="at">Lactobacillales =</span> <span class="st">&quot;#6390fb&quot;</span>,</span>
<span id="cb16-18"><a href="mag-catalogue.html#cb16-18" tabindex="-1"></a>                  <span class="at">ML615J_28 =</span>   <span class="st">&quot;#70c9de&quot;</span>,</span>
<span id="cb16-19"><a href="mag-catalogue.html#cb16-19" tabindex="-1"></a>                  <span class="at">RF39 =</span>    <span class="st">&quot;#5c7fba&quot;</span>,</span>
<span id="cb16-20"><a href="mag-catalogue.html#cb16-20" tabindex="-1"></a>                  <span class="at">Staphylococcales        =</span> <span class="st">&quot;#70a8dc&quot;</span>,</span>
<span id="cb16-21"><a href="mag-catalogue.html#cb16-21" tabindex="-1"></a></span>
<span id="cb16-22"><a href="mag-catalogue.html#cb16-22" tabindex="-1"></a>                  <span class="at">Christensenellales =</span>  <span class="st">&quot;#f58262&quot;</span>,</span>
<span id="cb16-23"><a href="mag-catalogue.html#cb16-23" tabindex="-1"></a>                  <span class="at">Clostridiales =</span>   <span class="st">&quot;#cf4a82&quot;</span>,</span>
<span id="cb16-24"><a href="mag-catalogue.html#cb16-24" tabindex="-1"></a>                  <span class="at">Lachnospirales =</span>  <span class="st">&quot;#e15f7d&quot;</span>,</span>
<span id="cb16-25"><a href="mag-catalogue.html#cb16-25" tabindex="-1"></a>                  <span class="at">Monoglobales =</span> <span class="st">&quot;#e18299&quot;</span>,</span>
<span id="cb16-26"><a href="mag-catalogue.html#cb16-26" tabindex="-1"></a>                  <span class="at">Oscillospirales =</span> <span class="st">&quot;#ba2760&quot;</span>,</span>
<span id="cb16-27"><a href="mag-catalogue.html#cb16-27" tabindex="-1"></a>                  <span class="at">Peptostreptococcales =</span>    <span class="st">&quot;#e35d51&quot;</span>,</span>
<span id="cb16-28"><a href="mag-catalogue.html#cb16-28" tabindex="-1"></a>                  <span class="at">TANB77 =</span>  <span class="st">&quot;#ef3d26&quot;</span>,</span>
<span id="cb16-29"><a href="mag-catalogue.html#cb16-29" tabindex="-1"></a>                  <span class="at">UBA1381 =</span> <span class="st">&quot;#be5643&quot;</span>,</span>
<span id="cb16-30"><a href="mag-catalogue.html#cb16-30" tabindex="-1"></a>                  <span class="at">UBA1212      =</span> <span class="st">&quot;#cc0100&quot;</span>,</span>
<span id="cb16-31"><a href="mag-catalogue.html#cb16-31" tabindex="-1"></a>                  </span>
<span id="cb16-32"><a href="mag-catalogue.html#cb16-32" tabindex="-1"></a>                  <span class="at">Peptococcales       =</span> <span class="st">&quot;#d9d2e9&quot;</span>,</span>
<span id="cb16-33"><a href="mag-catalogue.html#cb16-33" tabindex="-1"></a>                  <span class="at">UBA4068        =</span> <span class="st">&quot;#b4a7d6&quot;</span>,</span>
<span id="cb16-34"><a href="mag-catalogue.html#cb16-34" tabindex="-1"></a></span>
<span id="cb16-35"><a href="mag-catalogue.html#cb16-35" tabindex="-1"></a>                  <span class="at">Bacteroidales =</span>   <span class="st">&quot;#e3c7a0&quot;</span>,</span>
<span id="cb16-36"><a href="mag-catalogue.html#cb16-36" tabindex="-1"></a>                  </span>
<span id="cb16-37"><a href="mag-catalogue.html#cb16-37" tabindex="-1"></a>                  <span class="at">Enterobacterales =</span>    <span class="st">&quot;#ffcc62&quot;</span>,</span>
<span id="cb16-38"><a href="mag-catalogue.html#cb16-38" tabindex="-1"></a>                  </span>
<span id="cb16-39"><a href="mag-catalogue.html#cb16-39" tabindex="-1"></a>                  <span class="at">Verrucomicrobiales =</span> <span class="st">&quot;#823f1d&quot;</span>,</span>
<span id="cb16-40"><a href="mag-catalogue.html#cb16-40" tabindex="-1"></a>                  </span>
<span id="cb16-41"><a href="mag-catalogue.html#cb16-41" tabindex="-1"></a>                  <span class="at">Gastranaerophilales     =</span> <span class="st">&quot;#bf9001&quot;</span></span>
<span id="cb16-42"><a href="mag-catalogue.html#cb16-42" tabindex="-1"></a>                  )</span></code></pre></div>
</div>
<div id="define-plotting-settings" class="section level3 hasAnchor" number="2.8.2">
<h3><span class="header-section-number">2.8.2</span> Define plotting settings<a href="mag-catalogue.html#define-plotting-settings" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create custom ggplot theme for consistent plotting</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb17-1"><a href="mag-catalogue.html#cb17-1" tabindex="-1"></a>custom_ggplot_theme <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb17-2"><a href="mag-catalogue.html#cb17-2" tabindex="-1"></a>  <span class="at">strip.text.y.left =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb17-3"><a href="mag-catalogue.html#cb17-3" tabindex="-1"></a>  <span class="at">strip.text.y.right =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb17-4"><a href="mag-catalogue.html#cb17-4" tabindex="-1"></a>  <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb17-5"><a href="mag-catalogue.html#cb17-5" tabindex="-1"></a>  <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb17-6"><a href="mag-catalogue.html#cb17-6" tabindex="-1"></a>  <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;#dde3e9&quot;</span>, <span class="at">color =</span> <span class="st">&quot;white&quot;</span>, <span class="at">size =</span> <span class="fl">0.8</span>), <span class="co"># Custom facet strip background</span></span>
<span id="cb17-7"><a href="mag-catalogue.html#cb17-7" tabindex="-1"></a>  <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>), <span class="co"># Custom facet text</span></span>
<span id="cb17-8"><a href="mag-catalogue.html#cb17-8" tabindex="-1"></a>  <span class="at">strip.placement =</span> <span class="st">&quot;outside&quot;</span>, <span class="co"># Place strip outside the panel grid</span></span>
<span id="cb17-9"><a href="mag-catalogue.html#cb17-9" tabindex="-1"></a>  <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;lines&quot;</span>), <span class="co"># Adjust space between panels</span></span>
<span id="cb17-10"><a href="mag-catalogue.html#cb17-10" tabindex="-1"></a>  <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">&quot;#dde3e9&quot;</span>), <span class="co"># Customize major grid lines</span></span>
<span id="cb17-11"><a href="mag-catalogue.html#cb17-11" tabindex="-1"></a>  <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Remove minor grid lines</span></span>
<span id="cb17-12"><a href="mag-catalogue.html#cb17-12" tabindex="-1"></a>  <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>), <span class="co"># Change panel background color</span></span>
<span id="cb17-13"><a href="mag-catalogue.html#cb17-13" tabindex="-1"></a>  <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>) <span class="co"># Adjust plot margins to ensure content fits</span></span>
<span id="cb17-14"><a href="mag-catalogue.html#cb17-14" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="define-treatment-color-schemes" class="section level3 hasAnchor" number="2.8.3">
<h3><span class="header-section-number">2.8.3</span> Define treatment color schemes<a href="mag-catalogue.html#define-treatment-color-schemes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb18-1"><a href="mag-catalogue.html#cb18-1" tabindex="-1"></a><span class="co"># Bright colors (opaque)</span></span>
<span id="cb18-2"><a href="mag-catalogue.html#cb18-2" tabindex="-1"></a>treatment_colours_bright <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb18-3"><a href="mag-catalogue.html#cb18-3" tabindex="-1"></a>  <span class="at">TM1 =</span> <span class="st">&quot;#4059AE&quot;</span>,</span>
<span id="cb18-4"><a href="mag-catalogue.html#cb18-4" tabindex="-1"></a>  <span class="at">TM2 =</span> <span class="st">&quot;#6A9AC3&quot;</span>,</span>
<span id="cb18-5"><a href="mag-catalogue.html#cb18-5" tabindex="-1"></a>  <span class="at">TM3 =</span> <span class="st">&quot;#97D8C4&quot;</span>,</span>
<span id="cb18-6"><a href="mag-catalogue.html#cb18-6" tabindex="-1"></a>  <span class="at">TM4 =</span> <span class="st">&quot;#F3B942&quot;</span>,</span>
<span id="cb18-7"><a href="mag-catalogue.html#cb18-7" tabindex="-1"></a>  <span class="at">TM0 =</span> <span class="st">&quot;grey&quot;</span></span>
<span id="cb18-8"><a href="mag-catalogue.html#cb18-8" tabindex="-1"></a>)</span>
<span id="cb18-9"><a href="mag-catalogue.html#cb18-9" tabindex="-1"></a></span>
<span id="cb18-10"><a href="mag-catalogue.html#cb18-10" tabindex="-1"></a><span class="co"># Pastel / transparent (for fill, with ~50% opacity)</span></span>
<span id="cb18-11"><a href="mag-catalogue.html#cb18-11" tabindex="-1"></a>treatment_colours_pastel <span class="ot">&lt;-</span> <span class="fu">paste0</span>(treatment_colours_bright, <span class="st">&quot;50&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="save-working-objects" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Save working objects<a href="mag-catalogue.html#save-working-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the last step, the objects that are needed for downstream analyses are stored in an R object.</p>
<p>Save MAG catalogue data and color schemes</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb19-1"><a href="mag-catalogue.html#cb19-1" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb19-2"><a href="mag-catalogue.html#cb19-2" tabindex="-1"></a>  genome_tree,</span>
<span id="cb19-3"><a href="mag-catalogue.html#cb19-3" tabindex="-1"></a>  genome_metadata,</span>
<span id="cb19-4"><a href="mag-catalogue.html#cb19-4" tabindex="-1"></a>  phylum_metadata,</span>
<span id="cb19-5"><a href="mag-catalogue.html#cb19-5" tabindex="-1"></a>  order_metadata,</span>
<span id="cb19-6"><a href="mag-catalogue.html#cb19-6" tabindex="-1"></a>  genome_gifts,</span>
<span id="cb19-7"><a href="mag-catalogue.html#cb19-7" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">&quot;data/MAG_catalogue/data.Rdata&quot;</span></span>
<span id="cb19-8"><a href="mag-catalogue.html#cb19-8" tabindex="-1"></a>)</span>
<span id="cb19-9"><a href="mag-catalogue.html#cb19-9" tabindex="-1"></a></span>
<span id="cb19-10"><a href="mag-catalogue.html#cb19-10" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb19-11"><a href="mag-catalogue.html#cb19-11" tabindex="-1"></a>  phylum_colors,</span>
<span id="cb19-12"><a href="mag-catalogue.html#cb19-12" tabindex="-1"></a>  order_colors,</span>
<span id="cb19-13"><a href="mag-catalogue.html#cb19-13" tabindex="-1"></a>  treatment_colours_bright,</span>
<span id="cb19-14"><a href="mag-catalogue.html#cb19-14" tabindex="-1"></a>  treatment_colours_pastel,</span>
<span id="cb19-15"><a href="mag-catalogue.html#cb19-15" tabindex="-1"></a>  custom_ggplot_theme,</span>
<span id="cb19-16"><a href="mag-catalogue.html#cb19-16" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">&quot;data/data_colors.Rdata&quot;</span></span>
<span id="cb19-17"><a href="mag-catalogue.html#cb19-17" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="validation" class="section level2 hasAnchor" number="2.10">
<h2><span class="header-section-number">2.10</span> Validation<a href="mag-catalogue.html#validation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Validate MAG catalogue data</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb20-1"><a href="mag-catalogue.html#cb20-1" tabindex="-1"></a><span class="co"># Check dimensions of genome_metadata</span></span>
<span id="cb20-2"><a href="mag-catalogue.html#cb20-2" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Genome metadata dimensions:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-3"><a href="mag-catalogue.html#cb20-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">dim</span>(genome_metadata))</span>
<span id="cb20-4"><a href="mag-catalogue.html#cb20-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Number of genomes:&quot;</span>, <span class="fu">nrow</span>(genome_metadata), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-5"><a href="mag-catalogue.html#cb20-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of columns:&quot;</span>, <span class="fu">ncol</span>(genome_metadata), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-6"><a href="mag-catalogue.html#cb20-6" tabindex="-1"></a></span>
<span id="cb20-7"><a href="mag-catalogue.html#cb20-7" tabindex="-1"></a><span class="co"># Check column names</span></span>
<span id="cb20-8"><a href="mag-catalogue.html#cb20-8" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Column names:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-9"><a href="mag-catalogue.html#cb20-9" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">colnames</span>(genome_metadata))</span>
<span id="cb20-10"><a href="mag-catalogue.html#cb20-10" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-11"><a href="mag-catalogue.html#cb20-11" tabindex="-1"></a></span>
<span id="cb20-12"><a href="mag-catalogue.html#cb20-12" tabindex="-1"></a><span class="co"># Display top 10 rows</span></span>
<span id="cb20-13"><a href="mag-catalogue.html#cb20-13" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Top 10 rows of genome_metadata:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-14"><a href="mag-catalogue.html#cb20-14" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(genome_metadata, <span class="dv">10</span>))</span>
<span id="cb20-15"><a href="mag-catalogue.html#cb20-15" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-16"><a href="mag-catalogue.html#cb20-16" tabindex="-1"></a></span>
<span id="cb20-17"><a href="mag-catalogue.html#cb20-17" tabindex="-1"></a><span class="co"># Count circularized genomes</span></span>
<span id="cb20-18"><a href="mag-catalogue.html#cb20-18" tabindex="-1"></a>circularized_count <span class="ot">&lt;-</span> <span class="fu">sum</span>(genome_quality<span class="sc">$</span>circularity <span class="sc">==</span> <span class="st">&quot;1&quot;</span>)</span>
<span id="cb20-19"><a href="mag-catalogue.html#cb20-19" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of circularized genomes:&quot;</span>, circularized_count, <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-20"><a href="mag-catalogue.html#cb20-20" tabindex="-1"></a></span>
<span id="cb20-21"><a href="mag-catalogue.html#cb20-21" tabindex="-1"></a><span class="co"># Count phyla and orders</span></span>
<span id="cb20-22"><a href="mag-catalogue.html#cb20-22" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of unique phyla:&quot;</span>, <span class="fu">length</span>(phylum_level_vector), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-23"><a href="mag-catalogue.html#cb20-23" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of unique orders:&quot;</span>, <span class="fu">length</span>(order_level_vector), <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-24"><a href="mag-catalogue.html#cb20-24" tabindex="-1"></a></span>
<span id="cb20-25"><a href="mag-catalogue.html#cb20-25" tabindex="-1"></a><span class="co"># Summary of phyla</span></span>
<span id="cb20-26"><a href="mag-catalogue.html#cb20-26" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Phyla summary:</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-27"><a href="mag-catalogue.html#cb20-27" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(genome_metadata<span class="sc">$</span>phylum))</span>
<span id="cb20-28"><a href="mag-catalogue.html#cb20-28" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-29"><a href="mag-catalogue.html#cb20-29" tabindex="-1"></a></span>
<span id="cb20-30"><a href="mag-catalogue.html#cb20-30" tabindex="-1"></a><span class="co"># Summary of orders</span></span>
<span id="cb20-31"><a href="mag-catalogue.html#cb20-31" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Number of genomes per order (top 10):</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb20-32"><a href="mag-catalogue.html#cb20-32" tabindex="-1"></a>order_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(genome_metadata<span class="sc">$</span>order)</span>
<span id="cb20-33"><a href="mag-catalogue.html#cb20-33" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">sort</span>(order_counts, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), <span class="dv">10</span>))</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="micro-sample-metadata.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
