<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>12 Sequencing Statistics | 3D’omics | Histomonas Turkeys</title>
  <meta name="description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="12 Sequencing Statistics | 3D’omics | Histomonas Turkeys" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  <meta name="github-repo" content="3d-omics/ANMI024_Histomonas_Turkeys_Macro_and_Micro" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="12 Sequencing Statistics | 3D’omics | Histomonas Turkeys" />
  
  <meta name="twitter:description" content="Data analysis code for 3D’omics Histomoans trial in Turkeys" />
  

<meta name="author" content="Amalia Bogri" />
<meta name="author" content="Jorge Langa" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2025-11-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="histomonas-presence.html"/>
<link rel="next" href="bacterial-load.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#environment"><i class="fa fa-check"></i><b>1.1.1</b> Environment</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#libraries"><i class="fa fa-check"></i><b>1.1.2</b> Libraries</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#helper-functions"><i class="fa fa-check"></i><b>1.2</b> Helper Functions</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#alpha-diversity-calculation"><i class="fa fa-check"></i><b>1.2.1</b> Alpha Diversity Calculation</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#remove-samples-or-taxa"><i class="fa fa-check"></i><b>1.2.2</b> Remove samples or taxa</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#clr_function"><i class="fa fa-check"></i><b>1.2.3</b> clr_function</a></li>
<li class="chapter" data-level="1.2.4" data-path="index.html"><a href="index.html#perform_pca"><i class="fa fa-check"></i><b>1.2.4</b> perform_pca</a></li>
<li class="chapter" data-level="1.2.5" data-path="index.html"><a href="index.html#plot_pca"><i class="fa fa-check"></i><b>1.2.5</b> plot_pca</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#notes"><i class="fa fa-check"></i><b>1.3</b> Notes:</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#installation-of-modules"><i class="fa fa-check"></i><b>1.3.1</b> Installation of modules</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#airtable-access"><i class="fa fa-check"></i><b>1.3.2</b> Airtable access:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>2</b> MAG Catalogue</a>
<ul>
<li class="chapter" data-level="2.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#dereplicated-genomes"><i class="fa fa-check"></i><b>2.1</b> Dereplicated genomes</a></li>
<li class="chapter" data-level="2.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-tree"><i class="fa fa-check"></i><b>2.2</b> Genome (MAGs) tree</a></li>
<li class="chapter" data-level="2.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-taxonomy"><i class="fa fa-check"></i><b>2.3</b> Genome (MAGs) taxonomy</a></li>
<li class="chapter" data-level="2.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-quality"><i class="fa fa-check"></i><b>2.4</b> Genome (MAGs) quality</a></li>
<li class="chapter" data-level="2.5" data-path="mag-catalogue.html"><a href="mag-catalogue.html#merged-genome-mags-taxonomy-quality"><i class="fa fa-check"></i><b>2.5</b> Merged Genome (MAGs) taxonomy &amp; quality</a></li>
<li class="chapter" data-level="2.6" data-path="mag-catalogue.html"><a href="mag-catalogue.html#factoring-of-taxonomic-levels"><i class="fa fa-check"></i><b>2.6</b> Factoring of taxonomic levels</a></li>
<li class="chapter" data-level="2.7" data-path="mag-catalogue.html"><a href="mag-catalogue.html#genome-mags-functional-annotations"><i class="fa fa-check"></i><b>2.7</b> Genome (MAGs) functional annotations</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#distil-functional-annotations"><i class="fa fa-check"></i><b>2.7.1</b> Distil functional annotations</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="mag-catalogue.html"><a href="mag-catalogue.html#bacterial-phyla-color-scheme-data"><i class="fa fa-check"></i><b>2.8</b> Bacterial phyla color scheme data</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-color-schemes-for-phyla-and-orders"><i class="fa fa-check"></i><b>2.8.1</b> Define color schemes for phyla and orders</a></li>
<li class="chapter" data-level="2.8.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-plotting-settings"><i class="fa fa-check"></i><b>2.8.2</b> Define plotting settings</a></li>
<li class="chapter" data-level="2.8.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#define-treatment-color-schemes"><i class="fa fa-check"></i><b>2.8.3</b> Define treatment color schemes</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="mag-catalogue.html"><a href="mag-catalogue.html#save-working-objects"><i class="fa fa-check"></i><b>2.9</b> Save working objects</a></li>
<li class="chapter" data-level="2.10" data-path="mag-catalogue.html"><a href="mag-catalogue.html#validation"><i class="fa fa-check"></i><b>2.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html"><i class="fa fa-check"></i><b>3</b> Micro-sample Metadata</a>
<ul>
<li class="chapter" data-level="3.1" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#load-and-process-micro-sample-metadata-from-airtable"><i class="fa fa-check"></i><b>3.1</b> Load and process micro-sample metadata from Airtable</a></li>
<li class="chapter" data-level="3.2" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#save-working-objects-1"><i class="fa fa-check"></i><b>3.2</b> Save working objects</a></li>
<li class="chapter" data-level="3.3" data-path="micro-sample-metadata.html"><a href="micro-sample-metadata.html#validation-1"><i class="fa fa-check"></i><b>3.3</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html"><i class="fa fa-check"></i><b>4</b> Micro-sample Counts</a>
<ul>
<li class="chapter" data-level="4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#load-required-data"><i class="fa fa-check"></i><b>4.1</b> Load required data</a></li>
<li class="chapter" data-level="4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#bacterial-reads-counts"><i class="fa fa-check"></i><b>4.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#genome-covered-bases"><i class="fa fa-check"></i><b>4.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="4.4" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#filter-and-normalise-data"><i class="fa fa-check"></i><b>4.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#coverage-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Coverage filtering</a></li>
<li class="chapter" data-level="4.4.2" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-genome-count-table"><i class="fa fa-check"></i><b>4.4.2</b> Generate genome count table</a></li>
<li class="chapter" data-level="4.4.3" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#generate-length-corrected-read-count-tables"><i class="fa fa-check"></i><b>4.4.3</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#alpha-diversity-calculations"><i class="fa fa-check"></i><b>4.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="4.6" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#save-working-objects-2"><i class="fa fa-check"></i><b>4.6</b> Save working objects</a></li>
<li class="chapter" data-level="4.7" data-path="micro-sample-counts.html"><a href="micro-sample-counts.html#validation-2"><i class="fa fa-check"></i><b>4.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>5</b> Micro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#load-required-data-1"><i class="fa fa-check"></i><b>5.1</b> Load required data</a></li>
<li class="chapter" data-level="5.2" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#fastp-sequencing-statistics"><i class="fa fa-check"></i><b>5.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="5.3" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas"><i class="fa fa-check"></i><b>5.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="5.4" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#bacterial-mapped-reads"><i class="fa fa-check"></i><b>5.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="5.5" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table"><i class="fa fa-check"></i><b>5.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="5.6" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#sanity-checks"><i class="fa fa-check"></i><b>5.6</b> Sanity checks</a></li>
<li class="chapter" data-level="5.7" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics"><i class="fa fa-check"></i><b>5.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="5.8" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#create-plot_data_stats"><i class="fa fa-check"></i><b>5.8</b> Create plot_data_stats</a></li>
<li class="chapter" data-level="5.9" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#save-working-objects-3"><i class="fa fa-check"></i><b>5.9</b> Save working objects</a></li>
<li class="chapter" data-level="5.10" data-path="micro-sample-sequencing-statistics.html"><a href="micro-sample-sequencing-statistics.html#validation-3"><i class="fa fa-check"></i><b>5.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html"><i class="fa fa-check"></i><b>6</b> Macro-sample Metadata</a>
<ul>
<li class="chapter" data-level="6.1" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#save-working-objects-4"><i class="fa fa-check"></i><b>6.1</b> Save working objects</a></li>
<li class="chapter" data-level="6.2" data-path="macro-sample-metadata.html"><a href="macro-sample-metadata.html#validation-4"><i class="fa fa-check"></i><b>6.2</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html"><i class="fa fa-check"></i><b>7</b> Macro-sample Counts</a>
<ul>
<li class="chapter" data-level="7.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#load-required-data-2"><i class="fa fa-check"></i><b>7.1</b> Load required data</a></li>
<li class="chapter" data-level="7.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#bacterial-reads-counts-1"><i class="fa fa-check"></i><b>7.2</b> Bacterial reads (counts)</a></li>
<li class="chapter" data-level="7.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#genome-covered-bases-1"><i class="fa fa-check"></i><b>7.3</b> Genome covered bases</a></li>
<li class="chapter" data-level="7.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#filter-and-normalise-data-1"><i class="fa fa-check"></i><b>7.4</b> Filter and normalise data</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-coverage-table"><i class="fa fa-check"></i><b>7.4.1</b> Generate coverage table</a></li>
<li class="chapter" data-level="7.4.2" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#coverage-filtering-1"><i class="fa fa-check"></i><b>7.4.2</b> Coverage filtering</a></li>
<li class="chapter" data-level="7.4.3" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-genome-count-table-1"><i class="fa fa-check"></i><b>7.4.3</b> Generate genome count table</a></li>
<li class="chapter" data-level="7.4.4" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#generate-length-corrected-read-count-tables-1"><i class="fa fa-check"></i><b>7.4.4</b> Generate length-corrected read count tables</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#alpha-diversity-calculations-1"><i class="fa fa-check"></i><b>7.5</b> Alpha diversity calculations</a></li>
<li class="chapter" data-level="7.6" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#save-working-objects-5"><i class="fa fa-check"></i><b>7.6</b> Save working objects</a></li>
<li class="chapter" data-level="7.7" data-path="macro-sample-counts.html"><a href="macro-sample-counts.html#validation-5"><i class="fa fa-check"></i><b>7.7</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html"><i class="fa fa-check"></i><b>8</b> Macro-sample Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#load-required-data-3"><i class="fa fa-check"></i><b>8.1</b> Load required data</a></li>
<li class="chapter" data-level="8.2" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#fastp-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.2</b> FastP sequencing statistics</a></li>
<li class="chapter" data-level="8.3" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#non-bacterial-mapped-reads-human-chicken-swine-turkey-histomonas-1"><i class="fa fa-check"></i><b>8.3</b> Non-bacterial mapped reads: human, chicken, swine, turkey, histomonas</a></li>
<li class="chapter" data-level="8.4" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#bacterial-mapped-reads-1"><i class="fa fa-check"></i><b>8.4</b> Bacterial mapped reads</a></li>
<li class="chapter" data-level="8.5" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#alternative-way-to-estimate-bacterial-read-counts-from-count-table-1"><i class="fa fa-check"></i><b>8.5</b> Alternative way to estimate bacterial read counts (from count table)</a></li>
<li class="chapter" data-level="8.6" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#sanity-checks-1"><i class="fa fa-check"></i><b>8.6</b> Sanity checks</a></li>
<li class="chapter" data-level="8.7" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#combine-relevant-sequencing-statistics-1"><i class="fa fa-check"></i><b>8.7</b> Combine relevant sequencing statistics</a></li>
<li class="chapter" data-level="8.8" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#create-plot_data_stats_macro"><i class="fa fa-check"></i><b>8.8</b> Create plot_data_stats_macro</a></li>
<li class="chapter" data-level="8.9" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#save-working-objects-6"><i class="fa fa-check"></i><b>8.9</b> Save working objects</a></li>
<li class="chapter" data-level="8.10" data-path="macro-sample-sequencing-statistics.html"><a href="macro-sample-sequencing-statistics.html#validation-6"><i class="fa fa-check"></i><b>8.10</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html"><i class="fa fa-check"></i><b>9</b> Metabolomics Data Preparation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-metabolomics-data"><i class="fa fa-check"></i><b>9.1</b> Load metabolomics data</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#load-abundance-values-and-metabolites-metadata"><i class="fa fa-check"></i><b>9.1.1</b> Load abundance values and metabolites metadata</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#classify-samples-by-type"><i class="fa fa-check"></i><b>9.2</b> Classify samples by type</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#identify-sample-columns-and-classify-by-type"><i class="fa fa-check"></i><b>9.2.1</b> Identify sample columns and classify by type</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#create-dataframes-for-each-sample-type"><i class="fa fa-check"></i><b>9.3</b> Create dataframes for each sample type</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hn-histomonas-chicken-host-tissue"><i class="fa fa-check"></i><b>9.3.1</b> HN: Histomonas Chicken, host tissue</a></li>
<li class="chapter" data-level="9.3.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#hm-histomonas-chicken-digesta"><i class="fa fa-check"></i><b>9.3.2</b> HM: Histomonas Chicken, digesta</a></li>
<li class="chapter" data-level="9.3.3" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mn-histomonas-turkey-host-tissue"><i class="fa fa-check"></i><b>9.3.3</b> MN: Histomonas Turkey, host tissue</a></li>
<li class="chapter" data-level="9.3.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#mm-histomonas-turkey-digesta"><i class="fa fa-check"></i><b>9.3.4</b> MM: Histomonas Turkey, digesta</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-processed-data"><i class="fa fa-check"></i><b>9.4</b> Save processed data</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#save-metabolomics-dataframes"><i class="fa fa-check"></i><b>9.4.1</b> Save metabolomics dataframes</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validation-7"><i class="fa fa-check"></i><b>9.5</b> Validation</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#validate-metabolomics-data"><i class="fa fa-check"></i><b>9.5.1</b> Validate metabolomics data</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#notes-on-metabolites-metadata"><i class="fa fa-check"></i><b>9.6</b> Notes on metabolites metadata</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#available-metadata-fields"><i class="fa fa-check"></i><b>9.6.1</b> Available metadata fields:</a></li>
<li class="chapter" data-level="9.6.2" data-path="metabolomics-data-preparation.html"><a href="metabolomics-data-preparation.html#suggested-uses"><i class="fa fa-check"></i><b>9.6.2</b> Suggested uses:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html"><i class="fa fa-check"></i><b>10</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-quality"><i class="fa fa-check"></i><b>10.1</b> Genome quality</a></li>
<li class="chapter" data-level="10.2" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours"><i class="fa fa-check"></i><b>10.2</b> MAG catalogue phyla and order colours</a></li>
<li class="chapter" data-level="10.3" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#mag-catalogue-phyla-and-order-colours-and-genera-number"><i class="fa fa-check"></i><b>10.3</b> MAG catalogue phyla and order colours, and genera number</a></li>
<li class="chapter" data-level="10.4" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot"><i class="fa fa-check"></i><b>10.4</b> Genome phylogeny plot</a></li>
<li class="chapter" data-level="10.5" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#genome-phylogeny-plot-without-legend"><i class="fa fa-check"></i><b>10.5</b> Genome phylogeny plot without legend</a></li>
<li class="chapter" data-level="10.6" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-overview"><i class="fa fa-check"></i><b>10.6</b> Functional overview</a></li>
<li class="chapter" data-level="10.7" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-distances"><i class="fa fa-check"></i><b>10.7</b> Functional distances</a></li>
<li class="chapter" data-level="10.8" data-path="mag-catalogue-1.html"><a href="mag-catalogue-1.html#functional-ordination"><i class="fa fa-check"></i><b>10.8</b> Functional ordination</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="histomonas-presence.html"><a href="histomonas-presence.html"><i class="fa fa-check"></i><b>11</b> Histomonas Presence</a>
<ul>
<li class="chapter" data-level="11.1" data-path="histomonas-presence.html"><a href="histomonas-presence.html#macro"><i class="fa fa-check"></i><b>11.1</b> Macro</a></li>
<li class="chapter" data-level="11.2" data-path="histomonas-presence.html"><a href="histomonas-presence.html#micro"><i class="fa fa-check"></i><b>11.2</b> Micro</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html"><i class="fa fa-check"></i><b>12</b> Sequencing Statistics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#prepare-tidy-tables-for-plotting"><i class="fa fa-check"></i><b>12.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="12.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#macro-samples"><i class="fa fa-check"></i><b>12.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases-reads"><i class="fa fa-check"></i><b>12.2.1</b> Number of bases &amp; reads</a></li>
<li class="chapter" data-level="12.2.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped"><i class="fa fa-check"></i><b>12.2.2</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.2.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#composition-of-reads-vs.-references"><i class="fa fa-check"></i><b>12.2.3</b> Composition of reads vs. references</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#micro-samples"><i class="fa fa-check"></i><b>12.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-bases"><i class="fa fa-check"></i><b>12.3.1</b> Number of bases</a></li>
<li class="chapter" data-level="12.3.2" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#number-of-reads-after-trimming"><i class="fa fa-check"></i><b>12.3.2</b> Number of reads after trimming</a></li>
<li class="chapter" data-level="12.3.3" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped-1"><i class="fa fa-check"></i><b>12.3.3</b> % of reads mapped to references, removed by trimming, unmapped</a></li>
<li class="chapter" data-level="12.3.4" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts"><i class="fa fa-check"></i><b>12.3.4</b> Collection attempts</a></li>
<li class="chapter" data-level="12.3.5" data-path="sequencing-statistics.html"><a href="sequencing-statistics.html#collection-attempts-vs-filtering-retention"><i class="fa fa-check"></i><b>12.3.5</b> Collection attempts vs filtering retention</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="bacterial-load.html"><a href="bacterial-load.html"><i class="fa fa-check"></i><b>13</b> Bacterial Load</a>
<ul>
<li class="chapter" data-level="13.1" data-path="bacterial-load.html"><a href="bacterial-load.html#macro-samples-1"><i class="fa fa-check"></i><b>13.1</b> Macro-samples</a></li>
<li class="chapter" data-level="13.2" data-path="bacterial-load.html"><a href="bacterial-load.html#micro-samples-1"><i class="fa fa-check"></i><b>13.2</b> Micro-samples</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>14</b> Alpha Diversity</a>
<ul>
<li class="chapter" data-level="14.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#prepare-tidy-tables-for-plotting-1"><i class="fa fa-check"></i><b>14.1</b> Prepare tidy tables for plotting</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-tidy-tables"><i class="fa fa-check"></i><b>14.1.1</b> Alpha diversity tidy tables</a></li>
<li class="chapter" data-level="14.1.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#genome-counts-tidy-tables"><i class="fa fa-check"></i><b>14.1.2</b> Genome counts tidy tables</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#macro-samples-2"><i class="fa fa-check"></i><b>14.2</b> Macro-samples</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div"><i class="fa fa-check"></i><b>14.2.1</b> Alpha-div</a></li>
<li class="chapter" data-level="14.2.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size"><i class="fa fa-check"></i><b>14.2.2</b> Alpha-div vs genome size</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#micro-samples-2"><i class="fa fa-check"></i><b>14.3</b> Micro-samples</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-diversity-1"><i class="fa fa-check"></i><b>14.3.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="14.3.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#alpha-div-vs-genome-size-1"><i class="fa fa-check"></i><b>14.3.2</b> Alpha-div vs genome size</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html"><i class="fa fa-check"></i><b>15</b> Beta Diversity - Macro Samples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#prepare-tidy-tables-for-plotting-2"><i class="fa fa-check"></i><b>15.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="15.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-level"><i class="fa fa-check"></i><b>15.2</b> Species level</a></li>
<li class="chapter" data-level="15.3" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#genus-level"><i class="fa fa-check"></i><b>15.3</b> Genus level</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#permanova"><i class="fa fa-check"></i><b>15.3.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="15.3.2" data-path="beta-diversity---macro-samples.html"><a href="beta-diversity---macro-samples.html#species-representative-genomes-overlap"><i class="fa fa-check"></i><b>15.3.2</b> Species representative genomes overlap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html"><i class="fa fa-check"></i><b>16</b> Beta Diversity - Micro Samples</a>
<ul>
<li class="chapter" data-level="16.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#prepare-tidy-tables-for-plotting-3"><i class="fa fa-check"></i><b>16.1</b> Prepare tidy tables for plotting</a></li>
<li class="chapter" data-level="16.2" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#species-level-1"><i class="fa fa-check"></i><b>16.2</b> Species level</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="beta-diversity---micro-samples.html"><a href="beta-diversity---micro-samples.html#permanova-1"><i class="fa fa-check"></i><b>16.2.1</b> PERMANOVA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html"><i class="fa fa-check"></i><b>17</b> Beta Diversity - Macro vs Micro Comparison</a>
<ul>
<li class="chapter" data-level="17.1" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#genomes-overlap"><i class="fa fa-check"></i><b>17.1</b> Genomes overlap</a></li>
<li class="chapter" data-level="17.2" data-path="beta-diversity---macro-vs-micro-comparison.html"><a href="beta-diversity---macro-vs-micro-comparison.html#micro-vs-macro-genus-level"><i class="fa fa-check"></i><b>17.2</b> Micro vs Macro Genus level</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html"><i class="fa fa-check"></i><b>18</b> Spatial Analysis - Micro Samples</a>
<ul>
<li class="chapter" data-level="18.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#functions"><i class="fa fa-check"></i><b>18.1</b> Functions</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#spatial_cryosections"><i class="fa fa-check"></i><b>18.1.1</b> spatial_cryosections</a></li>
<li class="chapter" data-level="18.1.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#additive-diversity-partitioning"><i class="fa fa-check"></i><b>18.1.2</b> Additive diversity partitioning</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#load-data"><i class="fa fa-check"></i><b>18.2</b> Load data</a></li>
<li class="chapter" data-level="18.3" data-path="spatial-analysis---micro-samples.html"><a href="spatial-analysis---micro-samples.html#extended-rlq-analysis"><i class="fa fa-check"></i><b>18.3</b> Extended RLQ analysis</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html"><i class="fa fa-check"></i><b>19</b> Export Data for MicrobeTag</a>
<ul>
<li class="chapter" data-level="19.1" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#load-required-data-4"><i class="fa fa-check"></i><b>19.1</b> Load required data</a></li>
<li class="chapter" data-level="19.2" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-sample-metadata-macro-samples"><i class="fa fa-check"></i><b>19.2</b> Export sample metadata (macro samples)</a></li>
<li class="chapter" data-level="19.3" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-metabolites-data-mm-samples"><i class="fa fa-check"></i><b>19.3</b> Export metabolites data (MM samples)</a></li>
<li class="chapter" data-level="19.4" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-metadata"><i class="fa fa-check"></i><b>19.4</b> Export genome metadata</a></li>
<li class="chapter" data-level="19.5" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#export-genome-counts-macro-samples-filtered"><i class="fa fa-check"></i><b>19.5</b> Export genome counts (macro samples, filtered)</a></li>
<li class="chapter" data-level="19.6" data-path="export-data-for-microbetag.html"><a href="export-data-for-microbetag.html#summary"><i class="fa fa-check"></i><b>19.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="package-versions.html"><a href="package-versions.html"><i class="fa fa-check"></i><b>20</b> Package Versions</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">3D’omics | Histomonas Turkeys</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sequencing-statistics" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">12</span> Sequencing Statistics<a href="sequencing-statistics.html#sequencing-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb128"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb128-1"><a href="sequencing-statistics.html#cb128-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/macro/sample_metadata.Rdata&quot;</span>)</span>
<span id="cb128-2"><a href="sequencing-statistics.html#cb128-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/macro/counts.Rdata&quot;</span>)</span>
<span id="cb128-3"><a href="sequencing-statistics.html#cb128-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/macro/seq_stats.Rdata&quot;</span>)</span>
<span id="cb128-4"><a href="sequencing-statistics.html#cb128-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/micro/sample_metadata.Rdata&quot;</span>)</span>
<span id="cb128-5"><a href="sequencing-statistics.html#cb128-5" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/micro/counts.Rdata&quot;</span>)</span>
<span id="cb128-6"><a href="sequencing-statistics.html#cb128-6" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/micro/seq_stats.Rdata&quot;</span>)</span>
<span id="cb128-7"><a href="sequencing-statistics.html#cb128-7" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/MAG_catalogue/data.Rdata&quot;</span>)</span>
<span id="cb128-8"><a href="sequencing-statistics.html#cb128-8" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data_colors.Rdata&quot;</span>)</span></code></pre></div>
<div id="prepare-tidy-tables-for-plotting" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> Prepare tidy tables for plotting<a href="sequencing-statistics.html#prepare-tidy-tables-for-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Create tidy table for sequencing stats before &amp; after trimming (micro)</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb129-1"><a href="sequencing-statistics.html#cb129-1" tabindex="-1"></a>tidy_plot_data_stats_before_after <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb129-2"><a href="sequencing-statistics.html#cb129-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb129-3"><a href="sequencing-statistics.html#cb129-3" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb129-4"><a href="sequencing-statistics.html#cb129-4" tabindex="-1"></a>    total_sequences_before_trim, total_sequences_after_trim,</span>
<span id="cb129-5"><a href="sequencing-statistics.html#cb129-5" tabindex="-1"></a>    percent_gc_before_trim, percent_gc_after_trim,</span>
<span id="cb129-6"><a href="sequencing-statistics.html#cb129-6" tabindex="-1"></a>    total_bases_before_trim, total_bases_after_trim),</span>
<span id="cb129-7"><a href="sequencing-statistics.html#cb129-7" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">&quot;Metric&quot;</span>, <span class="st">&quot;Condition&quot;</span>), <span class="co"># Split column names</span></span>
<span id="cb129-8"><a href="sequencing-statistics.html#cb129-8" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">&quot;(.*)_(before|after)_trim&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb129-9"><a href="sequencing-statistics.html#cb129-9" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Metric, <span class="at">values_from =</span> value) <span class="sc">%&gt;%</span></span>
<span id="cb129-10"><a href="sequencing-statistics.html#cb129-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Condition =</span> <span class="fu">factor</span>(Condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;before&quot;</span>, <span class="st">&quot;after&quot;</span>)))</span></code></pre></div>
<p>Create tidy table for sequencing stats before &amp; after trimming (macro)</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb130-1"><a href="sequencing-statistics.html#cb130-1" tabindex="-1"></a>tidy_plot_data_stats_before_after_macro <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb130-2"><a href="sequencing-statistics.html#cb130-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb130-3"><a href="sequencing-statistics.html#cb130-3" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb130-4"><a href="sequencing-statistics.html#cb130-4" tabindex="-1"></a>    total_sequences_before_trim, total_sequences_after_trim,</span>
<span id="cb130-5"><a href="sequencing-statistics.html#cb130-5" tabindex="-1"></a>    percent_gc_before_trim, percent_gc_after_trim,</span>
<span id="cb130-6"><a href="sequencing-statistics.html#cb130-6" tabindex="-1"></a>    total_bases_before_trim, total_bases_after_trim),</span>
<span id="cb130-7"><a href="sequencing-statistics.html#cb130-7" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">&quot;Metric&quot;</span>, <span class="st">&quot;Condition&quot;</span>), <span class="co"># Split column names</span></span>
<span id="cb130-8"><a href="sequencing-statistics.html#cb130-8" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">&quot;(.*)_(before|after)_trim&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb130-9"><a href="sequencing-statistics.html#cb130-9" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Metric, <span class="at">values_from =</span> value) <span class="sc">%&gt;%</span></span>
<span id="cb130-10"><a href="sequencing-statistics.html#cb130-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Condition =</span> <span class="fu">factor</span>(Condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;before&quot;</span>, <span class="st">&quot;after&quot;</span>)))</span></code></pre></div>
<p>Create tidy table for read counts (micro)</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb131-1"><a href="sequencing-statistics.html#cb131-1" tabindex="-1"></a>category_counts <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb131-2"><a href="sequencing-statistics.html#cb131-2" tabindex="-1"></a>  <span class="at">category =</span> <span class="fu">c</span>(</span>
<span id="cb131-3"><a href="sequencing-statistics.html#cb131-3" tabindex="-1"></a>    <span class="st">&quot;bacteria_total_mapped&quot;</span>, <span class="st">&quot;turkey_total_mapped&quot;</span>, <span class="st">&quot;histomonas_total_mapped&quot;</span>, <span class="st">&quot;chicken_total_mapped&quot;</span>,</span>
<span id="cb131-4"><a href="sequencing-statistics.html#cb131-4" tabindex="-1"></a>    <span class="st">&quot;human_total_mapped&quot;</span>, <span class="st">&quot;swine_total_mapped&quot;</span>, <span class="st">&quot;unmapped&quot;</span>, <span class="st">&quot;removed_sequences_after_trim&quot;</span>),</span>
<span id="cb131-5"><a href="sequencing-statistics.html#cb131-5" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>,  <span class="st">&quot;Swine&quot;</span>,  <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>),</span>
<span id="cb131-6"><a href="sequencing-statistics.html#cb131-6" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">c</span>( <span class="st">&quot;#B92D65&quot;</span>, <span class="st">&quot;#79b1a3&quot;</span>, <span class="st">&quot;#e35d51&quot;</span>, <span class="st">&quot;#E8BD50&quot;</span>, <span class="st">&quot;#1092b4&quot;</span>, <span class="st">&quot;#3baf6f&quot;</span>, <span class="st">&quot;#8d98ae&quot;</span>, <span class="st">&quot;#2b2d42&quot;</span>))</span>
<span id="cb131-7"><a href="sequencing-statistics.html#cb131-7" tabindex="-1"></a></span>
<span id="cb131-8"><a href="sequencing-statistics.html#cb131-8" tabindex="-1"></a>tidy_plot_data_stats_counts <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb131-9"><a href="sequencing-statistics.html#cb131-9" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb131-10"><a href="sequencing-statistics.html#cb131-10" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb131-11"><a href="sequencing-statistics.html#cb131-11" tabindex="-1"></a>      chicken_total_mapped, human_total_mapped,</span>
<span id="cb131-12"><a href="sequencing-statistics.html#cb131-12" tabindex="-1"></a>      swine_total_mapped, bacteria_total_mapped, turkey_total_mapped, histomonas_total_mapped,</span>
<span id="cb131-13"><a href="sequencing-statistics.html#cb131-13" tabindex="-1"></a>      unmapped, removed_sequences_after_trim),</span>
<span id="cb131-14"><a href="sequencing-statistics.html#cb131-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;mapping_status&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb131-15"><a href="sequencing-statistics.html#cb131-15" tabindex="-1"></a>  <span class="fu">left_join</span>(category_counts, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;mapping_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;category&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb131-16"><a href="sequencing-statistics.html#cb131-16" tabindex="-1"></a>  <span class="co"># order the groups in desired order</span></span>
<span id="cb131-17"><a href="sequencing-statistics.html#cb131-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label,</span>
<span id="cb131-18"><a href="sequencing-statistics.html#cb131-18" tabindex="-1"></a>         <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Swine&quot;</span>, <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>)))</span>
<span id="cb131-19"><a href="sequencing-statistics.html#cb131-19" tabindex="-1"></a></span>
<span id="cb131-20"><a href="sequencing-statistics.html#cb131-20" tabindex="-1"></a><span class="co"># set labels &amp; colours as colour palette</span></span>
<span id="cb131-21"><a href="sequencing-statistics.html#cb131-21" tabindex="-1"></a>color_palette_tidy_plot_data_stats_counts <span class="ot">&lt;-</span> <span class="fu">setNames</span>(category_counts<span class="sc">$</span>color, category_counts<span class="sc">$</span>label)</span></code></pre></div>
<p>Create tidy table for read counts (macro)</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb132-1"><a href="sequencing-statistics.html#cb132-1" tabindex="-1"></a>category_counts <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb132-2"><a href="sequencing-statistics.html#cb132-2" tabindex="-1"></a>  <span class="at">category =</span> <span class="fu">c</span>(</span>
<span id="cb132-3"><a href="sequencing-statistics.html#cb132-3" tabindex="-1"></a>    <span class="st">&quot;bacteria_total_mapped&quot;</span>, <span class="st">&quot;turkey_total_mapped&quot;</span>, <span class="st">&quot;histomonas_total_mapped&quot;</span>,<span class="st">&quot;chicken_total_mapped&quot;</span>,  </span>
<span id="cb132-4"><a href="sequencing-statistics.html#cb132-4" tabindex="-1"></a>    <span class="st">&quot;human_total_mapped&quot;</span>, <span class="st">&quot;swine_total_mapped&quot;</span>, <span class="st">&quot;unmapped&quot;</span>, <span class="st">&quot;removed_sequences_after_trim&quot;</span>),</span>
<span id="cb132-5"><a href="sequencing-statistics.html#cb132-5" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>,  <span class="st">&quot;Swine&quot;</span>,  <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>),</span>
<span id="cb132-6"><a href="sequencing-statistics.html#cb132-6" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">c</span>( <span class="st">&quot;#B92D65&quot;</span>, <span class="st">&quot;#79b1a3&quot;</span>, <span class="st">&quot;#e35d51&quot;</span>, <span class="st">&quot;#E8BD50&quot;</span>, <span class="st">&quot;#1092b4&quot;</span>, <span class="st">&quot;#3baf6f&quot;</span>, <span class="st">&quot;#8d98ae&quot;</span>, <span class="st">&quot;#2b2d42&quot;</span>))</span>
<span id="cb132-7"><a href="sequencing-statistics.html#cb132-7" tabindex="-1"></a></span>
<span id="cb132-8"><a href="sequencing-statistics.html#cb132-8" tabindex="-1"></a>tidy_plot_data_stats_counts_macro <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb132-9"><a href="sequencing-statistics.html#cb132-9" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb132-10"><a href="sequencing-statistics.html#cb132-10" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb132-11"><a href="sequencing-statistics.html#cb132-11" tabindex="-1"></a>      chicken_total_mapped, human_total_mapped,</span>
<span id="cb132-12"><a href="sequencing-statistics.html#cb132-12" tabindex="-1"></a>      swine_total_mapped, bacteria_total_mapped, turkey_total_mapped, histomonas_total_mapped,</span>
<span id="cb132-13"><a href="sequencing-statistics.html#cb132-13" tabindex="-1"></a>      unmapped, removed_sequences_after_trim),</span>
<span id="cb132-14"><a href="sequencing-statistics.html#cb132-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;mapping_status&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-15"><a href="sequencing-statistics.html#cb132-15" tabindex="-1"></a>  <span class="fu">left_join</span>(category_counts, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;mapping_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;category&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb132-16"><a href="sequencing-statistics.html#cb132-16" tabindex="-1"></a>  <span class="co"># order the groups in desired order</span></span>
<span id="cb132-17"><a href="sequencing-statistics.html#cb132-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label,</span>
<span id="cb132-18"><a href="sequencing-statistics.html#cb132-18" tabindex="-1"></a>         <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Swine&quot;</span>, <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>)))</span>
<span id="cb132-19"><a href="sequencing-statistics.html#cb132-19" tabindex="-1"></a></span>
<span id="cb132-20"><a href="sequencing-statistics.html#cb132-20" tabindex="-1"></a><span class="co"># set labels &amp; colours as colour palette</span></span>
<span id="cb132-21"><a href="sequencing-statistics.html#cb132-21" tabindex="-1"></a>color_palette_tidy_plot_data_stats_counts <span class="ot">&lt;-</span> <span class="fu">setNames</span>(category_counts<span class="sc">$</span>color, category_counts<span class="sc">$</span>label)</span></code></pre></div>
<p>Create tidy table for read percentages (micro)</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb133-1"><a href="sequencing-statistics.html#cb133-1" tabindex="-1"></a>category_prc <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb133-2"><a href="sequencing-statistics.html#cb133-2" tabindex="-1"></a>  <span class="at">category =</span> <span class="fu">c</span>(</span>
<span id="cb133-3"><a href="sequencing-statistics.html#cb133-3" tabindex="-1"></a>    <span class="st">&quot;bacteria_percentage&quot;</span>, <span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;histomonas_percentage&quot;</span>,  <span class="st">&quot;chicken_percentage&quot;</span>,  </span>
<span id="cb133-4"><a href="sequencing-statistics.html#cb133-4" tabindex="-1"></a>    <span class="st">&quot;human_percentage&quot;</span>, <span class="st">&quot;swine_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>),</span>
<span id="cb133-5"><a href="sequencing-statistics.html#cb133-5" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>,  <span class="st">&quot;Swine&quot;</span>,  <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>),</span>
<span id="cb133-6"><a href="sequencing-statistics.html#cb133-6" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">c</span>( <span class="st">&quot;#B92D65&quot;</span>, <span class="st">&quot;#79b1a3&quot;</span>, <span class="st">&quot;#e35d51&quot;</span>, <span class="st">&quot;#E8BD50&quot;</span>, <span class="st">&quot;#1092b4&quot;</span>, <span class="st">&quot;#3baf6f&quot;</span>, <span class="st">&quot;#8d98ae&quot;</span>, <span class="st">&quot;#2b2d42&quot;</span>))</span>
<span id="cb133-7"><a href="sequencing-statistics.html#cb133-7" tabindex="-1"></a></span>
<span id="cb133-8"><a href="sequencing-statistics.html#cb133-8" tabindex="-1"></a>tidy_plot_data_stats_counts_prc <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb133-9"><a href="sequencing-statistics.html#cb133-9" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb133-10"><a href="sequencing-statistics.html#cb133-10" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb133-11"><a href="sequencing-statistics.html#cb133-11" tabindex="-1"></a>      chicken_percentage, human_percentage,</span>
<span id="cb133-12"><a href="sequencing-statistics.html#cb133-12" tabindex="-1"></a>      swine_percentage, bacteria_percentage, turkey_percentage, histomonas_percentage,</span>
<span id="cb133-13"><a href="sequencing-statistics.html#cb133-13" tabindex="-1"></a>      unmapped_percentage, trimmed_reads_percentage),</span>
<span id="cb133-14"><a href="sequencing-statistics.html#cb133-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;mapping_status&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb133-15"><a href="sequencing-statistics.html#cb133-15" tabindex="-1"></a>  <span class="fu">left_join</span>(category_prc, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;mapping_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;category&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb133-16"><a href="sequencing-statistics.html#cb133-16" tabindex="-1"></a>  <span class="co"># order the groups in desired order</span></span>
<span id="cb133-17"><a href="sequencing-statistics.html#cb133-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label,</span>
<span id="cb133-18"><a href="sequencing-statistics.html#cb133-18" tabindex="-1"></a>         <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>,  <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Swine&quot;</span>, <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>)))</span>
<span id="cb133-19"><a href="sequencing-statistics.html#cb133-19" tabindex="-1"></a></span>
<span id="cb133-20"><a href="sequencing-statistics.html#cb133-20" tabindex="-1"></a><span class="co"># set labels &amp; colours as colour palette</span></span>
<span id="cb133-21"><a href="sequencing-statistics.html#cb133-21" tabindex="-1"></a>color_palette_tidy_plot_data_stats_counts <span class="ot">&lt;-</span> <span class="fu">setNames</span>(category_counts<span class="sc">$</span>color, category_counts<span class="sc">$</span>label)</span></code></pre></div>
<p>Create tidy table for read percentages (macro)</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb134-1"><a href="sequencing-statistics.html#cb134-1" tabindex="-1"></a>category_prc <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb134-2"><a href="sequencing-statistics.html#cb134-2" tabindex="-1"></a>  <span class="at">category =</span> <span class="fu">c</span>(</span>
<span id="cb134-3"><a href="sequencing-statistics.html#cb134-3" tabindex="-1"></a>    <span class="st">&quot;bacteria_percentage&quot;</span>, <span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;histomonas_percentage&quot;</span>,  <span class="st">&quot;chicken_percentage&quot;</span>,  </span>
<span id="cb134-4"><a href="sequencing-statistics.html#cb134-4" tabindex="-1"></a>    <span class="st">&quot;human_percentage&quot;</span>, <span class="st">&quot;swine_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>),</span>
<span id="cb134-5"><a href="sequencing-statistics.html#cb134-5" tabindex="-1"></a>  <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>, <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>,  <span class="st">&quot;Swine&quot;</span>,  <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>),</span>
<span id="cb134-6"><a href="sequencing-statistics.html#cb134-6" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">c</span>( <span class="st">&quot;#B92D65&quot;</span>, <span class="st">&quot;#79b1a3&quot;</span>, <span class="st">&quot;#e35d51&quot;</span>, <span class="st">&quot;#E8BD50&quot;</span>, <span class="st">&quot;#1092b4&quot;</span>, <span class="st">&quot;#3baf6f&quot;</span>, <span class="st">&quot;#8d98ae&quot;</span>, <span class="st">&quot;#2b2d42&quot;</span>))</span>
<span id="cb134-7"><a href="sequencing-statistics.html#cb134-7" tabindex="-1"></a></span>
<span id="cb134-8"><a href="sequencing-statistics.html#cb134-8" tabindex="-1"></a>tidy_plot_data_stats_counts_prc_macro <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb134-9"><a href="sequencing-statistics.html#cb134-9" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb134-10"><a href="sequencing-statistics.html#cb134-10" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb134-11"><a href="sequencing-statistics.html#cb134-11" tabindex="-1"></a>      chicken_percentage, human_percentage,</span>
<span id="cb134-12"><a href="sequencing-statistics.html#cb134-12" tabindex="-1"></a>      swine_percentage, bacteria_percentage, turkey_percentage, histomonas_percentage,</span>
<span id="cb134-13"><a href="sequencing-statistics.html#cb134-13" tabindex="-1"></a>      unmapped_percentage, trimmed_reads_percentage),</span>
<span id="cb134-14"><a href="sequencing-statistics.html#cb134-14" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">&quot;mapping_status&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb134-15"><a href="sequencing-statistics.html#cb134-15" tabindex="-1"></a>  <span class="fu">left_join</span>(category_prc, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;mapping_status&quot;</span> <span class="ot">=</span> <span class="st">&quot;category&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb134-16"><a href="sequencing-statistics.html#cb134-16" tabindex="-1"></a>  <span class="co"># order the groups in desired order</span></span>
<span id="cb134-17"><a href="sequencing-statistics.html#cb134-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label,</span>
<span id="cb134-18"><a href="sequencing-statistics.html#cb134-18" tabindex="-1"></a>         <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Bacteria&quot;</span>,  <span class="st">&quot;Turkey&quot;</span>, <span class="st">&quot;Histomonas&quot;</span>, <span class="st">&quot;Chicken&quot;</span>, <span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Swine&quot;</span>, <span class="st">&quot;Unmapped&quot;</span>, <span class="st">&quot;Removed&quot;</span>)))</span>
<span id="cb134-19"><a href="sequencing-statistics.html#cb134-19" tabindex="-1"></a></span>
<span id="cb134-20"><a href="sequencing-statistics.html#cb134-20" tabindex="-1"></a><span class="co"># set labels &amp; colours as colour palette</span></span>
<span id="cb134-21"><a href="sequencing-statistics.html#cb134-21" tabindex="-1"></a>color_palette_tidy_plot_data_stats_counts <span class="ot">&lt;-</span> <span class="fu">setNames</span>(category_counts<span class="sc">$</span>color, category_counts<span class="sc">$</span>label)</span></code></pre></div>
<p>Create tidy tables for genome counts (micro)</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb135-1"><a href="sequencing-statistics.html#cb135-1" tabindex="-1"></a>tidy_plot_genome_counts <span class="ot">&lt;-</span>  genome_counts <span class="sc">%&gt;%</span></span>
<span id="cb135-2"><a href="sequencing-statistics.html#cb135-2" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;microsample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb135-3"><a href="sequencing-statistics.html#cb135-3" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> </span>
<span id="cb135-4"><a href="sequencing-statistics.html#cb135-4" tabindex="-1"></a>    <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span> </span>
<span id="cb135-5"><a href="sequencing-statistics.html#cb135-5" tabindex="-1"></a>    <span class="fu">left_join</span>(final_combined_stats, <span class="at">by =</span> <span class="fu">join_by</span>(microsample <span class="sc">==</span> microsample)) <span class="sc">%&gt;%</span></span>
<span id="cb135-6"><a href="sequencing-statistics.html#cb135-6" tabindex="-1"></a>    <span class="co"># Flatten section if it&#39;s a list</span></span>
<span id="cb135-7"><a href="sequencing-statistics.html#cb135-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">section =</span> <span class="fu">unlist</span>(section)) <span class="sc">%&gt;%</span> </span>
<span id="cb135-8"><a href="sequencing-statistics.html#cb135-8" tabindex="-1"></a>    <span class="co"># Filter out rows with count &lt;= 0 </span></span>
<span id="cb135-9"><a href="sequencing-statistics.html#cb135-9" tabindex="-1"></a>    <span class="co"># (that are redundant after pivot_longer, but important for b div)</span></span>
<span id="cb135-10"><a href="sequencing-statistics.html#cb135-10" tabindex="-1"></a>    <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb135-11"><a href="sequencing-statistics.html#cb135-11" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> phylum_level_vector),</span>
<span id="cb135-12"><a href="sequencing-statistics.html#cb135-12" tabindex="-1"></a>           <span class="at">order =</span> <span class="fu">factor</span>(order, <span class="at">levels =</span> order_level_vector))</span></code></pre></div>
</div>
<div id="macro-samples" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Macro-samples<a href="sequencing-statistics.html#macro-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="number-of-bases-reads" class="section level3 hasAnchor" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Number of bases &amp; reads<a href="sequencing-statistics.html#number-of-bases-reads" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb136"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb136-1"><a href="sequencing-statistics.html#cb136-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_bases_before_trim&quot;</span></span>
<span id="cb136-2"><a href="sequencing-statistics.html#cb136-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb136-3"><a href="sequencing-statistics.html#cb136-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb136-4"><a href="sequencing-statistics.html#cb136-4" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb136-5"><a href="sequencing-statistics.html#cb136-5" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb136-6"><a href="sequencing-statistics.html#cb136-6" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb136-7"><a href="sequencing-statistics.html#cb136-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb136-8"><a href="sequencing-statistics.html#cb136-8" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb136-9"><a href="sequencing-statistics.html#cb136-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb136-10"><a href="sequencing-statistics.html#cb136-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb136-11"><a href="sequencing-statistics.html#cb136-11" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="9%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">S3B014</td>
<td align="right">6596700</td>
<td align="right">0</td>
<td align="right">6596700</td>
<td align="right">6596700</td>
<td align="left">6.6e+06</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="right">5511612750</td>
<td align="right">1499163075</td>
<td align="right">4918069350</td>
<td align="right">6417232425</td>
<td align="left">5.5e+09</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb137"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb137-1"><a href="sequencing-statistics.html#cb137-1" tabindex="-1"></a>barplot_bases <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb137-2"><a href="sequencing-statistics.html#cb137-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type_simple, <span class="at">y =</span> total_bases_before_trim, <span class="at">color =</span> bacteria_percentage, <span class="at">shape=</span>filter_status)) <span class="sc">+</span></span>
<span id="cb137-3"><a href="sequencing-statistics.html#cb137-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sample type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of bases before trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb137-4"><a href="sequencing-statistics.html#cb137-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb137-5"><a href="sequencing-statistics.html#cb137-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb137-6"><a href="sequencing-statistics.html#cb137-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb137-7"><a href="sequencing-statistics.html#cb137-7" tabindex="-1"></a>      <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb137-8"><a href="sequencing-statistics.html#cb137-8" tabindex="-1"></a>  <span class="fu">facet_nested</span>(. <span class="sc">~</span> batch, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb137-9"><a href="sequencing-statistics.html#cb137-9" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb137-10"><a href="sequencing-statistics.html#cb137-10" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb137-11"><a href="sequencing-statistics.html#cb137-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb137-12"><a href="sequencing-statistics.html#cb137-12" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="dv">10</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb137-13"><a href="sequencing-statistics.html#cb137-13" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb137-14"><a href="sequencing-statistics.html#cb137-14" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb137-15"><a href="sequencing-statistics.html#cb137-15" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> type_simple, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb137-16"><a href="sequencing-statistics.html#cb137-16" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb137-17"><a href="sequencing-statistics.html#cb137-17" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>,</span>
<span id="cb137-18"><a href="sequencing-statistics.html#cb137-18" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) </span>
<span id="cb137-19"><a href="sequencing-statistics.html#cb137-19" tabindex="-1"></a></span>
<span id="cb137-20"><a href="sequencing-statistics.html#cb137-20" tabindex="-1"></a><span class="co"># barplot_bases</span></span>
<span id="cb137-21"><a href="sequencing-statistics.html#cb137-21" tabindex="-1"></a></span>
<span id="cb137-22"><a href="sequencing-statistics.html#cb137-22" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_sequences_after_trim&quot;</span></span>
<span id="cb137-23"><a href="sequencing-statistics.html#cb137-23" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb137-24"><a href="sequencing-statistics.html#cb137-24" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb137-25"><a href="sequencing-statistics.html#cb137-25" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb137-26"><a href="sequencing-statistics.html#cb137-26" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb137-27"><a href="sequencing-statistics.html#cb137-27" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb137-28"><a href="sequencing-statistics.html#cb137-28" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb137-29"><a href="sequencing-statistics.html#cb137-29" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb137-30"><a href="sequencing-statistics.html#cb137-30" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb137-31"><a href="sequencing-statistics.html#cb137-31" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb137-32"><a href="sequencing-statistics.html#cb137-32" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">S3B014</td>
<td align="right">28</td>
<td align="right">0</td>
<td align="right">28</td>
<td align="right">28</td>
<td align="left">2.8e+01</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="right">35792651</td>
<td align="right">9020001</td>
<td align="right">31968296</td>
<td align="right">40988296</td>
<td align="left">3.6e+07</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb138"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb138-1"><a href="sequencing-statistics.html#cb138-1" tabindex="-1"></a>barplot_reads_after <span class="ot">&lt;-</span> plot_data_stats_macro <span class="sc">%&gt;%</span></span>
<span id="cb138-2"><a href="sequencing-statistics.html#cb138-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type_simple, <span class="at">y =</span> total_sequences_after_trim, <span class="at">color =</span> bacteria_percentage, <span class="at">shape=</span>filter_status)) <span class="sc">+</span></span>
<span id="cb138-3"><a href="sequencing-statistics.html#cb138-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Sample type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of reads after trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb138-4"><a href="sequencing-statistics.html#cb138-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb138-5"><a href="sequencing-statistics.html#cb138-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb138-6"><a href="sequencing-statistics.html#cb138-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb138-7"><a href="sequencing-statistics.html#cb138-7" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb138-8"><a href="sequencing-statistics.html#cb138-8" tabindex="-1"></a>  <span class="fu">facet_nested</span>(. <span class="sc">~</span> batch, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb138-9"><a href="sequencing-statistics.html#cb138-9" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb138-10"><a href="sequencing-statistics.html#cb138-10" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb138-11"><a href="sequencing-statistics.html#cb138-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb138-12"><a href="sequencing-statistics.html#cb138-12" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.6</span>)) <span class="sc">+</span></span>
<span id="cb138-13"><a href="sequencing-statistics.html#cb138-13" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb138-14"><a href="sequencing-statistics.html#cb138-14" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb138-15"><a href="sequencing-statistics.html#cb138-15" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> type_simple, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb138-16"><a href="sequencing-statistics.html#cb138-16" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb138-17"><a href="sequencing-statistics.html#cb138-17" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>,</span>
<span id="cb138-18"><a href="sequencing-statistics.html#cb138-18" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb138-19"><a href="sequencing-statistics.html#cb138-19" tabindex="-1"></a></span>
<span id="cb138-20"><a href="sequencing-statistics.html#cb138-20" tabindex="-1"></a><span class="co"># barplot_trimmed_reads + barplot_reads_after +</span></span>
<span id="cb138-21"><a href="sequencing-statistics.html#cb138-21" tabindex="-1"></a><span class="co">#   plot_annotation(tag_levels = &#39;a&#39;) +</span></span>
<span id="cb138-22"><a href="sequencing-statistics.html#cb138-22" tabindex="-1"></a><span class="co">#   plot_layout(guides = &quot;collect&quot;) &amp;</span></span>
<span id="cb138-23"><a href="sequencing-statistics.html#cb138-23" tabindex="-1"></a><span class="co">#   theme(legend.position = &quot;bottom&quot;, legend.box = &quot;vertical&quot;)</span></span>
<span id="cb138-24"><a href="sequencing-statistics.html#cb138-24" tabindex="-1"></a></span>
<span id="cb138-25"><a href="sequencing-statistics.html#cb138-25" tabindex="-1"></a></span>
<span id="cb138-26"><a href="sequencing-statistics.html#cb138-26" tabindex="-1"></a>barplot_bases <span class="sc">+</span> barplot_reads_after <span class="sc">+</span></span>
<span id="cb138-27"><a href="sequencing-statistics.html#cb138-27" tabindex="-1"></a>    <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">+</span></span>
<span id="cb138-28"><a href="sequencing-statistics.html#cb138-28" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>) <span class="sc">&amp;</span></span>
<span id="cb138-29"><a href="sequencing-statistics.html#cb138-29" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_macro_bases_reads-1.png" width="960" /></p>
</div>
<div id="of-reads-mapped-to-references-removed-by-trimming-unmapped" class="section level3 hasAnchor" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> % of reads mapped to references, removed by trimming, unmapped<a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb139"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb139-1"><a href="sequencing-statistics.html#cb139-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;counts&quot;</span></span>
<span id="cb139-2"><a href="sequencing-statistics.html#cb139-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc_macro <span class="sc">%&gt;%</span></span>
<span id="cb139-3"><a href="sequencing-statistics.html#cb139-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb139-4"><a href="sequencing-statistics.html#cb139-4" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>,<span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-5"><a href="sequencing-statistics.html#cb139-5" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch, label) <span class="sc">%&gt;%</span> <span class="co">#treatment_expl, age, </span></span>
<span id="cb139-6"><a href="sequencing-statistics.html#cb139-6" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb139-7"><a href="sequencing-statistics.html#cb139-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb139-8"><a href="sequencing-statistics.html#cb139-8" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb139-9"><a href="sequencing-statistics.html#cb139-9" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb139-10"><a href="sequencing-statistics.html#cb139-10" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb139-11"><a href="sequencing-statistics.html#cb139-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb139-12"><a href="sequencing-statistics.html#cb139-12" tabindex="-1"></a>    <span class="at">label_txt =</span> <span class="fu">paste0</span>(</span>
<span id="cb139-13"><a href="sequencing-statistics.html#cb139-13" tabindex="-1"></a>      <span class="st">&quot;&quot;</span>, <span class="fu">round</span>(median, <span class="dv">1</span>),<span class="st">&quot;%&quot;</span>,</span>
<span id="cb139-14"><a href="sequencing-statistics.html#cb139-14" tabindex="-1"></a>      <span class="st">&quot; [&quot;</span>, <span class="fu">round</span>(Q1, <span class="dv">1</span>), <span class="st">&quot;–&quot;</span>, <span class="fu">round</span>(Q3, <span class="dv">1</span>), <span class="st">&quot;]&quot;</span>))</span>
<span id="cb139-15"><a href="sequencing-statistics.html#cb139-15" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="9%" />
<col width="11%" />
<col width="13%" />
<col width="13%" />
<col width="13%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="left">label</th>
<th align="right">Q1</th>
<th align="right">median</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="left">Bacteria</td>
<td align="right">11.837008</td>
<td align="right">47.666029</td>
<td align="right">80.284200</td>
<td align="left">47.7% [11.8–80.3]</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="left">Turkey</td>
<td align="right">6.416658</td>
<td align="right">47.463445</td>
<td align="right">85.672307</td>
<td align="left">47.5% [6.4–85.7]</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="left">Unmapped</td>
<td align="right">3.149501</td>
<td align="right">6.687567</td>
<td align="right">11.451008</td>
<td align="left">6.7% [3.1–11.5]</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">S3B014</td>
<td align="left">Removed</td>
<td align="right">2.404141</td>
<td align="right">2.754639</td>
<td align="right">3.467268</td>
<td align="left">2.8% [2.4–3.5]</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb140"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb140-1"><a href="sequencing-statistics.html#cb140-1" tabindex="-1"></a>boxplot_reads_prc_1 <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc_macro <span class="sc">%&gt;%</span></span>
<span id="cb140-2"><a href="sequencing-statistics.html#cb140-2" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb140-3"><a href="sequencing-statistics.html#cb140-3" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>, <span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb140-4"><a href="sequencing-statistics.html#cb140-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> counts, <span class="at">color =</span> label)) <span class="sc">+</span></span>
<span id="cb140-5"><a href="sequencing-statistics.html#cb140-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb140-6"><a href="sequencing-statistics.html#cb140-6" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb140-7"><a href="sequencing-statistics.html#cb140-7" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> color_palette_tidy_plot_data_stats_counts) <span class="sc">+</span></span>
<span id="cb140-8"><a href="sequencing-statistics.html#cb140-8" tabindex="-1"></a>  <span class="fu">facet_nested</span>(. <span class="sc">~</span> label, <span class="at">scales =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb140-9"><a href="sequencing-statistics.html#cb140-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,<span class="at">y =</span> <span class="st">&quot;Reads %&quot;</span>) <span class="sc">+</span></span>
<span id="cb140-10"><a href="sequencing-statistics.html#cb140-10" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span>  <span class="co"># Remove legend</span></span>
<span id="cb140-11"><a href="sequencing-statistics.html#cb140-11" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb140-12"><a href="sequencing-statistics.html#cb140-12" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb140-13"><a href="sequencing-statistics.html#cb140-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb140-14"><a href="sequencing-statistics.html#cb140-14" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span> </span>
<span id="cb140-15"><a href="sequencing-statistics.html#cb140-15" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.15</span>))) <span class="sc">+</span></span>
<span id="cb140-16"><a href="sequencing-statistics.html#cb140-16" tabindex="-1"></a>    <span class="co"># median + IQR text (placed a bit above the median line)</span></span>
<span id="cb140-17"><a href="sequencing-statistics.html#cb140-17" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb140-18"><a href="sequencing-statistics.html#cb140-18" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb140-19"><a href="sequencing-statistics.html#cb140-19" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb140-20"><a href="sequencing-statistics.html#cb140-20" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb140-21"><a href="sequencing-statistics.html#cb140-21" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">1.5</span>,</span>
<span id="cb140-22"><a href="sequencing-statistics.html#cb140-22" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb140-23"><a href="sequencing-statistics.html#cb140-23" tabindex="-1"></a></span>
<span id="cb140-24"><a href="sequencing-statistics.html#cb140-24" tabindex="-1"></a><span class="co"># boxplot_reads_prc_1</span></span></code></pre></div>
</div>
<div id="composition-of-reads-vs.-references" class="section level3 hasAnchor" number="12.2.3">
<h3><span class="header-section-number">12.2.3</span> Composition of reads vs. references<a href="sequencing-statistics.html#composition-of-reads-vs.-references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb141"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb141-1"><a href="sequencing-statistics.html#cb141-1" tabindex="-1"></a>barplot_1 <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_macro <span class="sc">%&gt;%</span></span>
<span id="cb141-2"><a href="sequencing-statistics.html#cb141-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> counts, <span class="at">y =</span> microsample, <span class="at">fill =</span> label)) <span class="sc">+</span></span>
<span id="cb141-3"><a href="sequencing-statistics.html#cb141-3" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">colour=</span><span class="st">&quot;white&quot;</span>, <span class="at">linewidth=</span><span class="fl">0.1</span>, <span class="at">position =</span> <span class="st">&quot;fill&quot;</span>) <span class="sc">+</span> <span class="co">#plot stacked bars with white borders</span></span>
<span id="cb141-4"><a href="sequencing-statistics.html#cb141-4" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_palette_tidy_plot_data_stats_counts) <span class="sc">+</span></span>
<span id="cb141-5"><a href="sequencing-statistics.html#cb141-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Counts&quot;</span>,<span class="at">y =</span> <span class="st">&quot;Microsample&quot;</span>,<span class="at">fill =</span> <span class="st">&quot;Read type&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-6"><a href="sequencing-statistics.html#cb141-6" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple <span class="sc">+</span> batch <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span> <span class="co"># treatment + day + animal + </span></span>
<span id="cb141-7"><a href="sequencing-statistics.html#cb141-7" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb141-8"><a href="sequencing-statistics.html#cb141-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb141-9"><a href="sequencing-statistics.html#cb141-9" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb141-10"><a href="sequencing-statistics.html#cb141-10" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-11"><a href="sequencing-statistics.html#cb141-11" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Ratio of reads&quot;</span>)</span>
<span id="cb141-12"><a href="sequencing-statistics.html#cb141-12" tabindex="-1"></a><span class="co"># barplot_1</span></span>
<span id="cb141-13"><a href="sequencing-statistics.html#cb141-13" tabindex="-1"></a></span>
<span id="cb141-14"><a href="sequencing-statistics.html#cb141-14" tabindex="-1"></a>barplot_2 <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_macro <span class="sc">%&gt;%</span></span>
<span id="cb141-15"><a href="sequencing-statistics.html#cb141-15" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> counts, <span class="at">y =</span> microsample, <span class="at">fill =</span> label)) <span class="sc">+</span></span>
<span id="cb141-16"><a href="sequencing-statistics.html#cb141-16" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">colour=</span><span class="st">&quot;white&quot;</span>, <span class="at">linewidth=</span><span class="fl">0.1</span>) <span class="sc">+</span> <span class="co">#plot stacked bars with white borders</span></span>
<span id="cb141-17"><a href="sequencing-statistics.html#cb141-17" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_palette_tidy_plot_data_stats_counts) <span class="sc">+</span></span>
<span id="cb141-18"><a href="sequencing-statistics.html#cb141-18" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Counts&quot;</span>,<span class="at">y =</span> <span class="st">&quot;Samples&quot;</span>,<span class="at">fill =</span> <span class="st">&quot;Read type&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-19"><a href="sequencing-statistics.html#cb141-19" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple  <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-20"><a href="sequencing-statistics.html#cb141-20" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb141-21"><a href="sequencing-statistics.html#cb141-21" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb141-22"><a href="sequencing-statistics.html#cb141-22" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb141-23"><a href="sequencing-statistics.html#cb141-23" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb141-24"><a href="sequencing-statistics.html#cb141-24" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) </span>
<span id="cb141-25"><a href="sequencing-statistics.html#cb141-25" tabindex="-1"></a><span class="co"># barplot_2</span></span></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb142-1"><a href="sequencing-statistics.html#cb142-1" tabindex="-1"></a>(boxplot_reads_prc_1 <span class="sc">|</span>barplot_2  )    <span class="sc">+</span></span>
<span id="cb142-2"><a href="sequencing-statistics.html#cb142-2" tabindex="-1"></a>  <span class="fu">plot_layout</span>(</span>
<span id="cb142-3"><a href="sequencing-statistics.html#cb142-3" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span>,        <span class="co"># single column</span></span>
<span id="cb142-4"><a href="sequencing-statistics.html#cb142-4" tabindex="-1"></a>    <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>),   <span class="co"># assign relative heights</span></span>
<span id="cb142-5"><a href="sequencing-statistics.html#cb142-5" tabindex="-1"></a>    <span class="at">guides =</span> <span class="st">&quot;collect&quot;</span></span>
<span id="cb142-6"><a href="sequencing-statistics.html#cb142-6" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb142-7"><a href="sequencing-statistics.html#cb142-7" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">&amp;</span></span>
<span id="cb142-8"><a href="sequencing-statistics.html#cb142-8" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_macro_combo-1.png" width="960" /></p>
</div>
</div>
<div id="micro-samples" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Micro-samples<a href="sequencing-statistics.html#micro-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="number-of-bases" class="section level3 hasAnchor" number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> Number of bases<a href="sequencing-statistics.html#number-of-bases" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb143"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb143-1"><a href="sequencing-statistics.html#cb143-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_bases_before_trim&quot;</span></span>
<span id="cb143-2"><a href="sequencing-statistics.html#cb143-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb143-3"><a href="sequencing-statistics.html#cb143-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, section) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb143-4"><a href="sequencing-statistics.html#cb143-4" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb143-5"><a href="sequencing-statistics.html#cb143-5" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb143-6"><a href="sequencing-statistics.html#cb143-6" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb143-7"><a href="sequencing-statistics.html#cb143-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb143-8"><a href="sequencing-statistics.html#cb143-8" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb143-9"><a href="sequencing-statistics.html#cb143-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb143-10"><a href="sequencing-statistics.html#cb143-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb143-11"><a href="sequencing-statistics.html#cb143-11" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="16%" />
<col width="13%" />
<col width="13%" />
<col width="13%" />
<col width="13%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">section</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">Caecum right</td>
<td align="right">37071450</td>
<td align="right">147571200</td>
<td align="right">11465475</td>
<td align="right">159036675</td>
<td align="left">3.7e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="right">4127211450</td>
<td align="right">4200413625</td>
<td align="right">2236636050</td>
<td align="right">6437049675</td>
<td align="left">4.1e+09</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb144"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb144-1"><a href="sequencing-statistics.html#cb144-1" tabindex="-1"></a>barplot_bases_all <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb144-2"><a href="sequencing-statistics.html#cb144-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> total_bases_before_trim)) <span class="sc">+</span></span>
<span id="cb144-3"><a href="sequencing-statistics.html#cb144-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of bases before trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb144-4"><a href="sequencing-statistics.html#cb144-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb144-5"><a href="sequencing-statistics.html#cb144-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb144-6"><a href="sequencing-statistics.html#cb144-6" tabindex="-1"></a></span>
<span id="cb144-7"><a href="sequencing-statistics.html#cb144-7" tabindex="-1"></a>  <span class="co"># single boxplot per batch</span></span>
<span id="cb144-8"><a href="sequencing-statistics.html#cb144-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb144-9"><a href="sequencing-statistics.html#cb144-9" tabindex="-1"></a></span>
<span id="cb144-10"><a href="sequencing-statistics.html#cb144-10" tabindex="-1"></a>  <span class="co"># keep color &amp; shape for jitter points only</span></span>
<span id="cb144-11"><a href="sequencing-statistics.html#cb144-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> bacteria_percentage, <span class="at">shape =</span> filter_status),</span>
<span id="cb144-12"><a href="sequencing-statistics.html#cb144-12" tabindex="-1"></a>              <span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb144-13"><a href="sequencing-statistics.html#cb144-13" tabindex="-1"></a></span>
<span id="cb144-14"><a href="sequencing-statistics.html#cb144-14" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb144-15"><a href="sequencing-statistics.html#cb144-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb144-16"><a href="sequencing-statistics.html#cb144-16" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb144-17"><a href="sequencing-statistics.html#cb144-17" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb144-18"><a href="sequencing-statistics.html#cb144-18" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="dv">10</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>),</span>
<span id="cb144-19"><a href="sequencing-statistics.html#cb144-19" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove x-axis title</span></span>
<span id="cb144-20"><a href="sequencing-statistics.html#cb144-20" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),   <span class="co"># remove tick labels #axis.text.x = element_text(angle = 90, hjust = 1)</span></span>
<span id="cb144-21"><a href="sequencing-statistics.html#cb144-21" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove tick marks</span></span>
<span id="cb144-22"><a href="sequencing-statistics.html#cb144-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb144-23"><a href="sequencing-statistics.html#cb144-23" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb144-24"><a href="sequencing-statistics.html#cb144-24" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb144-25"><a href="sequencing-statistics.html#cb144-25" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb144-26"><a href="sequencing-statistics.html#cb144-26" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb144-27"><a href="sequencing-statistics.html#cb144-27" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb144-28"><a href="sequencing-statistics.html#cb144-28" tabindex="-1"></a></span>
<span id="cb144-29"><a href="sequencing-statistics.html#cb144-29" tabindex="-1"></a></span>
<span id="cb144-30"><a href="sequencing-statistics.html#cb144-30" tabindex="-1"></a> <span class="co"># barplot_bases_all</span></span></code></pre></div>
<div class="sourceCode" id="cb145"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb145-1"><a href="sequencing-statistics.html#cb145-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_bases_before_trim&quot;</span></span>
<span id="cb145-2"><a href="sequencing-statistics.html#cb145-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb145-3"><a href="sequencing-statistics.html#cb145-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb145-4"><a href="sequencing-statistics.html#cb145-4" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb145-5"><a href="sequencing-statistics.html#cb145-5" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb145-6"><a href="sequencing-statistics.html#cb145-6" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb145-7"><a href="sequencing-statistics.html#cb145-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb145-8"><a href="sequencing-statistics.html#cb145-8" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-9"><a href="sequencing-statistics.html#cb145-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb145-10"><a href="sequencing-statistics.html#cb145-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb145-11"><a href="sequencing-statistics.html#cb145-11" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table>
<colgroup>
<col width="15%" />
<col width="11%" />
<col width="14%" />
<col width="14%" />
<col width="14%" />
<col width="15%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0042</td>
<td align="right">381594000</td>
<td align="right">670141875</td>
<td align="right">76284600</td>
<td align="right">746426475</td>
<td align="left">3.8e+08</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0043</td>
<td align="right">308657850</td>
<td align="right">1501134900</td>
<td align="right">74798250</td>
<td align="right">1575933150</td>
<td align="left">3.1e+08</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0044</td>
<td align="right">313270950</td>
<td align="right">1033517175</td>
<td align="right">149868450</td>
<td align="right">1183385625</td>
<td align="left">3.1e+08</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0045</td>
<td align="right">593576250</td>
<td align="right">352137000</td>
<td align="right">319580400</td>
<td align="right">671717400</td>
<td align="left">5.9e+08</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0046</td>
<td align="right">15834300</td>
<td align="right">30303375</td>
<td align="right">6954150</td>
<td align="right">37257525</td>
<td align="left">1.6e+07</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0047</td>
<td align="right">17806500</td>
<td align="right">13547250</td>
<td align="right">8759100</td>
<td align="right">22306350</td>
<td align="left">1.8e+07</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0048</td>
<td align="right">64911300</td>
<td align="right">89521875</td>
<td align="right">12463500</td>
<td align="right">101985375</td>
<td align="left">6.5e+07</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0049</td>
<td align="right">36247200</td>
<td align="right">88652625</td>
<td align="right">20577750</td>
<td align="right">109230375</td>
<td align="left">3.6e+07</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0050</td>
<td align="right">29742000</td>
<td align="right">41094225</td>
<td align="right">12299175</td>
<td align="right">53393400</td>
<td align="left">3.0e+07</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0051</td>
<td align="right">19828800</td>
<td align="right">25715625</td>
<td align="right">11875950</td>
<td align="right">37591575</td>
<td align="left">2.0e+07</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0052</td>
<td align="right">11830200</td>
<td align="right">18216525</td>
<td align="right">2158200</td>
<td align="right">20374725</td>
<td align="left">1.2e+07</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0053</td>
<td align="right">28588950</td>
<td align="right">109663650</td>
<td align="right">7541775</td>
<td align="right">117205425</td>
<td align="left">2.9e+07</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0054</td>
<td align="right">13662450</td>
<td align="right">15208875</td>
<td align="right">8577675</td>
<td align="right">23786550</td>
<td align="left">1.4e+07</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0055</td>
<td align="right">14385600</td>
<td align="right">21515325</td>
<td align="right">6401850</td>
<td align="right">27917175</td>
<td align="left">1.4e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="right">4981095750</td>
<td align="right">1444314375</td>
<td align="right">4247286150</td>
<td align="right">5691600525</td>
<td align="left">5.0e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="right">4875332700</td>
<td align="right">1379680050</td>
<td align="right">4143547875</td>
<td align="right">5523227925</td>
<td align="left">4.9e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="right">4187239200</td>
<td align="right">1095840975</td>
<td align="right">3664518150</td>
<td align="right">4760359125</td>
<td align="left">4.2e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="right">2820497550</td>
<td align="right">754390725</td>
<td align="right">2456315775</td>
<td align="right">3210706500</td>
<td align="left">2.8e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="right">3801025050</td>
<td align="right">6926109975</td>
<td align="right">1345935750</td>
<td align="right">8272045725</td>
<td align="left">3.8e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="right">4781791650</td>
<td align="right">8701430250</td>
<td align="right">1523287050</td>
<td align="right">10224717300</td>
<td align="left">4.8e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="right">3194397450</td>
<td align="right">4334175900</td>
<td align="right">1442898375</td>
<td align="right">5777074275</td>
<td align="left">3.2e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="right">2603536650</td>
<td align="right">4829702550</td>
<td align="right">904386675</td>
<td align="right">5734089225</td>
<td align="left">2.6e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="right">4860384600</td>
<td align="right">4451667900</td>
<td align="right">3329187675</td>
<td align="right">7780855575</td>
<td align="left">4.9e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="right">5171776050</td>
<td align="right">3118518375</td>
<td align="right">3798897525</td>
<td align="right">6917415900</td>
<td align="left">5.2e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="right">1813507650</td>
<td align="right">8666631525</td>
<td align="right">421784625</td>
<td align="right">9088416150</td>
<td align="left">1.8e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="right">1641974700</td>
<td align="right">8794150350</td>
<td align="right">674247675</td>
<td align="right">9468398025</td>
<td align="left">1.6e+09</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="right">4084768050</td>
<td align="right">4030189125</td>
<td align="right">2531654325</td>
<td align="right">6561843450</td>
<td align="left">4.1e+09</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="right">3484194300</td>
<td align="right">3949275675</td>
<td align="right">2372665575</td>
<td align="right">6321941250</td>
<td align="left">3.5e+09</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb146"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb146-1"><a href="sequencing-statistics.html#cb146-1" tabindex="-1"></a>barplot_bases_batch <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb146-2"><a href="sequencing-statistics.html#cb146-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> total_bases_before_trim)) <span class="sc">+</span></span>
<span id="cb146-3"><a href="sequencing-statistics.html#cb146-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Batch&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of bases before trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb146-4"><a href="sequencing-statistics.html#cb146-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb146-5"><a href="sequencing-statistics.html#cb146-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb146-6"><a href="sequencing-statistics.html#cb146-6" tabindex="-1"></a></span>
<span id="cb146-7"><a href="sequencing-statistics.html#cb146-7" tabindex="-1"></a>  <span class="co"># single boxplot per batch</span></span>
<span id="cb146-8"><a href="sequencing-statistics.html#cb146-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">group =</span> batch), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb146-9"><a href="sequencing-statistics.html#cb146-9" tabindex="-1"></a></span>
<span id="cb146-10"><a href="sequencing-statistics.html#cb146-10" tabindex="-1"></a>  <span class="co"># keep color &amp; shape for jitter points only</span></span>
<span id="cb146-11"><a href="sequencing-statistics.html#cb146-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> bacteria_percentage, <span class="at">shape =</span> filter_status),</span>
<span id="cb146-12"><a href="sequencing-statistics.html#cb146-12" tabindex="-1"></a>              <span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb146-13"><a href="sequencing-statistics.html#cb146-13" tabindex="-1"></a></span>
<span id="cb146-14"><a href="sequencing-statistics.html#cb146-14" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb146-15"><a href="sequencing-statistics.html#cb146-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb146-16"><a href="sequencing-statistics.html#cb146-16" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb146-17"><a href="sequencing-statistics.html#cb146-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb146-18"><a href="sequencing-statistics.html#cb146-18" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb146-19"><a href="sequencing-statistics.html#cb146-19" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb146-20"><a href="sequencing-statistics.html#cb146-20" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb146-21"><a href="sequencing-statistics.html#cb146-21" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb146-22"><a href="sequencing-statistics.html#cb146-22" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb146-23"><a href="sequencing-statistics.html#cb146-23" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb146-24"><a href="sequencing-statistics.html#cb146-24" tabindex="-1"></a></span>
<span id="cb146-25"><a href="sequencing-statistics.html#cb146-25" tabindex="-1"></a></span>
<span id="cb146-26"><a href="sequencing-statistics.html#cb146-26" tabindex="-1"></a> <span class="co"># barplot_bases_batch</span></span></code></pre></div>
<div class="sourceCode" id="cb147"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb147-1"><a href="sequencing-statistics.html#cb147-1" tabindex="-1"></a>(barplot_bases_all <span class="sc">+</span> barplot_bases_batch) <span class="sc">+</span></span>
<span id="cb147-2"><a href="sequencing-statistics.html#cb147-2" tabindex="-1"></a> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">+</span></span>
<span id="cb147-3"><a href="sequencing-statistics.html#cb147-3" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>,</span>
<span id="cb147-4"><a href="sequencing-statistics.html#cb147-4" tabindex="-1"></a>              <span class="at">ncol =</span> <span class="dv">2</span>,        <span class="co"># single column</span></span>
<span id="cb147-5"><a href="sequencing-statistics.html#cb147-5" tabindex="-1"></a>              <span class="at">width =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">8</span>)) <span class="sc">&amp;</span></span>
<span id="cb147-6"><a href="sequencing-statistics.html#cb147-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_micro_bases_combo_fixed-1.png" width="960" /></p>
</div>
<div id="number-of-reads-after-trimming" class="section level3 hasAnchor" number="12.3.2">
<h3><span class="header-section-number">12.3.2</span> Number of reads after trimming<a href="sequencing-statistics.html#number-of-reads-after-trimming" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb148"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb148-1"><a href="sequencing-statistics.html#cb148-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_sequences_after_trim&quot;</span></span>
<span id="cb148-2"><a href="sequencing-statistics.html#cb148-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb148-3"><a href="sequencing-statistics.html#cb148-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, section) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb148-4"><a href="sequencing-statistics.html#cb148-4" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb148-5"><a href="sequencing-statistics.html#cb148-5" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb148-6"><a href="sequencing-statistics.html#cb148-6" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb148-7"><a href="sequencing-statistics.html#cb148-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb148-8"><a href="sequencing-statistics.html#cb148-8" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb148-9"><a href="sequencing-statistics.html#cb148-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb148-10"><a href="sequencing-statistics.html#cb148-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb148-11"><a href="sequencing-statistics.html#cb148-11" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="17%" />
<col width="11%" />
<col width="14%" />
<col width="14%" />
<col width="11%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">section</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">Caecum right</td>
<td align="right">228788</td>
<td align="right">975349.5</td>
<td align="right">65023.5</td>
<td align="right">1040373</td>
<td align="left">2.3e+05</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="right">27009021</td>
<td align="right">27553216.5</td>
<td align="right">14600111.5</td>
<td align="right">42153328</td>
<td align="left">2.7e+07</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb149"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb149-1"><a href="sequencing-statistics.html#cb149-1" tabindex="-1"></a>barplot_trimmed_reads_all <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb149-2"><a href="sequencing-statistics.html#cb149-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> total_sequences_after_trim)) <span class="sc">+</span></span>
<span id="cb149-3"><a href="sequencing-statistics.html#cb149-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Type&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of reads after trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-4"><a href="sequencing-statistics.html#cb149-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-5"><a href="sequencing-statistics.html#cb149-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb149-6"><a href="sequencing-statistics.html#cb149-6" tabindex="-1"></a></span>
<span id="cb149-7"><a href="sequencing-statistics.html#cb149-7" tabindex="-1"></a>  <span class="co"># single boxplot per batch</span></span>
<span id="cb149-8"><a href="sequencing-statistics.html#cb149-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb149-9"><a href="sequencing-statistics.html#cb149-9" tabindex="-1"></a></span>
<span id="cb149-10"><a href="sequencing-statistics.html#cb149-10" tabindex="-1"></a>  <span class="co"># keep color &amp; shape for jitter points only</span></span>
<span id="cb149-11"><a href="sequencing-statistics.html#cb149-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> bacteria_percentage, <span class="at">shape =</span> filter_status),</span>
<span id="cb149-12"><a href="sequencing-statistics.html#cb149-12" tabindex="-1"></a>              <span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb149-13"><a href="sequencing-statistics.html#cb149-13" tabindex="-1"></a></span>
<span id="cb149-14"><a href="sequencing-statistics.html#cb149-14" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-15"><a href="sequencing-statistics.html#cb149-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb149-16"><a href="sequencing-statistics.html#cb149-16" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb149-17"><a href="sequencing-statistics.html#cb149-17" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb149-18"><a href="sequencing-statistics.html#cb149-18" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="dv">10</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>),</span>
<span id="cb149-19"><a href="sequencing-statistics.html#cb149-19" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove x-axis title</span></span>
<span id="cb149-20"><a href="sequencing-statistics.html#cb149-20" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),   <span class="co"># remove tick labels #axis.text.x = element_text(angle = 90, hjust = 1)</span></span>
<span id="cb149-21"><a href="sequencing-statistics.html#cb149-21" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove tick marks</span></span>
<span id="cb149-22"><a href="sequencing-statistics.html#cb149-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb149-23"><a href="sequencing-statistics.html#cb149-23" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb149-24"><a href="sequencing-statistics.html#cb149-24" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb149-25"><a href="sequencing-statistics.html#cb149-25" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb149-26"><a href="sequencing-statistics.html#cb149-26" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb149-27"><a href="sequencing-statistics.html#cb149-27" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb149-28"><a href="sequencing-statistics.html#cb149-28" tabindex="-1"></a></span>
<span id="cb149-29"><a href="sequencing-statistics.html#cb149-29" tabindex="-1"></a></span>
<span id="cb149-30"><a href="sequencing-statistics.html#cb149-30" tabindex="-1"></a> <span class="co"># barplot_trimmed_reads_all</span></span></code></pre></div>
<div class="sourceCode" id="cb150"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb150-1"><a href="sequencing-statistics.html#cb150-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_sequences_after_trim&quot;</span></span>
<span id="cb150-2"><a href="sequencing-statistics.html#cb150-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb150-3"><a href="sequencing-statistics.html#cb150-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch) <span class="sc">%&gt;%</span>  <span class="co"># replace with your grouping var if any</span></span>
<span id="cb150-4"><a href="sequencing-statistics.html#cb150-4" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb150-5"><a href="sequencing-statistics.html#cb150-5" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb150-6"><a href="sequencing-statistics.html#cb150-6" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb150-7"><a href="sequencing-statistics.html#cb150-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb150-8"><a href="sequencing-statistics.html#cb150-8" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-9"><a href="sequencing-statistics.html#cb150-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb150-10"><a href="sequencing-statistics.html#cb150-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb150-11"><a href="sequencing-statistics.html#cb150-11" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="12%" />
<col width="12%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="right">median</th>
<th align="right">IQR</th>
<th align="right">Q1</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0042</td>
<td align="right">2459842</td>
<td align="right">4376797.5</td>
<td align="right">484543.5</td>
<td align="right">4861341.0</td>
<td align="left">2.5e+06</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0043</td>
<td align="right">1995247</td>
<td align="right">9765967.5</td>
<td align="right">473412.5</td>
<td align="right">10239380.0</td>
<td align="left">2.0e+06</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0044</td>
<td align="right">2031699</td>
<td align="right">6757173.5</td>
<td align="right">962332.0</td>
<td align="right">7719505.5</td>
<td align="left">2.0e+06</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0045</td>
<td align="right">3858248</td>
<td align="right">2286869.0</td>
<td align="right">2070613.5</td>
<td align="right">4357482.5</td>
<td align="left">3.9e+06</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0046</td>
<td align="right">97302</td>
<td align="right">194909.0</td>
<td align="right">41086.5</td>
<td align="right">235995.5</td>
<td align="left">9.7e+04</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0047</td>
<td align="right">112590</td>
<td align="right">94468.0</td>
<td align="right">48824.0</td>
<td align="right">143292.0</td>
<td align="left">1.1e+05</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0048</td>
<td align="right">422369</td>
<td align="right">589077.5</td>
<td align="right">77794.0</td>
<td align="right">666871.5</td>
<td align="left">4.2e+05</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0049</td>
<td align="right">233202</td>
<td align="right">584384.0</td>
<td align="right">127094.0</td>
<td align="right">711478.0</td>
<td align="left">2.3e+05</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0050</td>
<td align="right">183764</td>
<td align="right">263284.5</td>
<td align="right">73870.5</td>
<td align="right">337155.0</td>
<td align="left">1.8e+05</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0051</td>
<td align="right">126293</td>
<td align="right">170428.5</td>
<td align="right">74213.0</td>
<td align="right">244641.5</td>
<td align="left">1.3e+05</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0052</td>
<td align="right">71049</td>
<td align="right">117197.5</td>
<td align="right">11686.5</td>
<td align="right">128884.0</td>
<td align="left">7.1e+04</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0053</td>
<td align="right">125765</td>
<td align="right">726604.5</td>
<td align="right">38541.5</td>
<td align="right">765146.0</td>
<td align="left">1.3e+05</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">MSEB0054</td>
<td align="right">81420</td>
<td align="right">96777.0</td>
<td align="right">50288.5</td>
<td align="right">147065.5</td>
<td align="left">8.1e+04</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">MSEB0055</td>
<td align="right">51828</td>
<td align="right">123490.0</td>
<td align="right">17762.0</td>
<td align="right">141252.0</td>
<td align="left">5.2e+04</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="right">32672262</td>
<td align="right">9469309.5</td>
<td align="right">27653623.5</td>
<td align="right">37122933.0</td>
<td align="left">3.3e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="right">31845581</td>
<td align="right">9116962.5</td>
<td align="right">27053912.5</td>
<td align="right">36170875.0</td>
<td align="left">3.2e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="right">27419607</td>
<td align="right">7164158.0</td>
<td align="right">24065772.5</td>
<td align="right">31229930.5</td>
<td align="left">2.7e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="right">18358025</td>
<td align="right">4841444.0</td>
<td align="right">16017658.5</td>
<td align="right">20859102.5</td>
<td align="left">1.8e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="right">24882457</td>
<td align="right">45345945.0</td>
<td align="right">8747816.5</td>
<td align="right">54093761.5</td>
<td align="left">2.5e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="right">31398133</td>
<td align="right">56331608.5</td>
<td align="right">9964526.0</td>
<td align="right">66296134.5</td>
<td align="left">3.1e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="right">20889594</td>
<td align="right">28266741.5</td>
<td align="right">9443605.0</td>
<td align="right">37710346.5</td>
<td align="left">2.1e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="right">17060753</td>
<td align="right">31821673.0</td>
<td align="right">5901804.0</td>
<td align="right">37723477.0</td>
<td align="left">1.7e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="right">31730068</td>
<td align="right">29040813.5</td>
<td align="right">21703138.5</td>
<td align="right">50743952.0</td>
<td align="left">3.2e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="right">33817176</td>
<td align="right">20378751.5</td>
<td align="right">24883836.5</td>
<td align="right">45262588.0</td>
<td align="left">3.4e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="right">11879344</td>
<td align="right">56917064.5</td>
<td align="right">2760204.0</td>
<td align="right">59677268.5</td>
<td align="left">1.2e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="right">10752621</td>
<td align="right">57914814.0</td>
<td align="right">4433817.0</td>
<td align="right">62348631.0</td>
<td align="left">1.1e+07</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="right">26803237</td>
<td align="right">26544213.0</td>
<td align="right">16659330.5</td>
<td align="right">43203543.5</td>
<td align="left">2.7e+07</td>
</tr>
<tr class="even">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="right">22755967</td>
<td align="right">25748179.5</td>
<td align="right">15532349.0</td>
<td align="right">41280528.5</td>
<td align="left">2.3e+07</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb151"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb151-1"><a href="sequencing-statistics.html#cb151-1" tabindex="-1"></a>barplot_reads_batch <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb151-2"><a href="sequencing-statistics.html#cb151-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> total_sequences_after_trim)) <span class="sc">+</span></span>
<span id="cb151-3"><a href="sequencing-statistics.html#cb151-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Batch&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of reads after trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb151-4"><a href="sequencing-statistics.html#cb151-4" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#c90076&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#3598bf&quot;</span>, <span class="at">name =</span> <span class="st">&quot;bacteria_percentage&quot;</span>) <span class="sc">+</span></span>
<span id="cb151-5"><a href="sequencing-statistics.html#cb151-5" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Retained by filtering&quot;</span> <span class="ot">=</span> <span class="dv">16</span>, <span class="st">&quot;Excluded from filtering&quot;</span> <span class="ot">=</span> <span class="dv">17</span>)) <span class="sc">+</span></span>
<span id="cb151-6"><a href="sequencing-statistics.html#cb151-6" tabindex="-1"></a></span>
<span id="cb151-7"><a href="sequencing-statistics.html#cb151-7" tabindex="-1"></a>  <span class="co"># single boxplot per batch</span></span>
<span id="cb151-8"><a href="sequencing-statistics.html#cb151-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">group =</span> batch), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb151-9"><a href="sequencing-statistics.html#cb151-9" tabindex="-1"></a></span>
<span id="cb151-10"><a href="sequencing-statistics.html#cb151-10" tabindex="-1"></a>  <span class="co"># keep color &amp; shape for jitter points only</span></span>
<span id="cb151-11"><a href="sequencing-statistics.html#cb151-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> bacteria_percentage, <span class="at">shape =</span> filter_status),</span>
<span id="cb151-12"><a href="sequencing-statistics.html#cb151-12" tabindex="-1"></a>              <span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb151-13"><a href="sequencing-statistics.html#cb151-13" tabindex="-1"></a></span>
<span id="cb151-14"><a href="sequencing-statistics.html#cb151-14" tabindex="-1"></a>  <span class="fu">facet_nested</span>(type_simple <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>) <span class="sc">+</span></span>
<span id="cb151-15"><a href="sequencing-statistics.html#cb151-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb151-16"><a href="sequencing-statistics.html#cb151-16" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb151-17"><a href="sequencing-statistics.html#cb151-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb151-18"><a href="sequencing-statistics.html#cb151-18" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb151-19"><a href="sequencing-statistics.html#cb151-19" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb151-20"><a href="sequencing-statistics.html#cb151-20" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb151-21"><a href="sequencing-statistics.html#cb151-21" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb151-22"><a href="sequencing-statistics.html#cb151-22" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb151-23"><a href="sequencing-statistics.html#cb151-23" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">7</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb151-24"><a href="sequencing-statistics.html#cb151-24" tabindex="-1"></a></span>
<span id="cb151-25"><a href="sequencing-statistics.html#cb151-25" tabindex="-1"></a></span>
<span id="cb151-26"><a href="sequencing-statistics.html#cb151-26" tabindex="-1"></a> <span class="co"># barplot_reads_batch</span></span></code></pre></div>
<div class="sourceCode" id="cb152"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb152-1"><a href="sequencing-statistics.html#cb152-1" tabindex="-1"></a>(barplot_trimmed_reads_all <span class="sc">+</span> barplot_reads_batch) <span class="sc">+</span></span>
<span id="cb152-2"><a href="sequencing-statistics.html#cb152-2" tabindex="-1"></a> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">+</span></span>
<span id="cb152-3"><a href="sequencing-statistics.html#cb152-3" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>,</span>
<span id="cb152-4"><a href="sequencing-statistics.html#cb152-4" tabindex="-1"></a>              <span class="at">ncol =</span> <span class="dv">2</span>,        <span class="co"># single column</span></span>
<span id="cb152-5"><a href="sequencing-statistics.html#cb152-5" tabindex="-1"></a>              <span class="at">width =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">8</span>)) <span class="sc">&amp;</span></span>
<span id="cb152-6"><a href="sequencing-statistics.html#cb152-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_micro_reads_combo-1.png" width="960" /></p>
</div>
<div id="of-reads-mapped-to-references-removed-by-trimming-unmapped-1" class="section level3 hasAnchor" number="12.3.3">
<h3><span class="header-section-number">12.3.3</span> % of reads mapped to references, removed by trimming, unmapped<a href="sequencing-statistics.html#of-reads-mapped-to-references-removed-by-trimming-unmapped-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb153"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb153-1"><a href="sequencing-statistics.html#cb153-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;counts&quot;</span></span>
<span id="cb153-2"><a href="sequencing-statistics.html#cb153-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc <span class="sc">%&gt;%</span></span>
<span id="cb153-3"><a href="sequencing-statistics.html#cb153-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb153-4"><a href="sequencing-statistics.html#cb153-4" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>,<span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb153-5"><a href="sequencing-statistics.html#cb153-5" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, batch, label) <span class="sc">%&gt;%</span> <span class="co">#treatment_expl, age, </span></span>
<span id="cb153-6"><a href="sequencing-statistics.html#cb153-6" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb153-7"><a href="sequencing-statistics.html#cb153-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb153-8"><a href="sequencing-statistics.html#cb153-8" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb153-9"><a href="sequencing-statistics.html#cb153-9" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb153-10"><a href="sequencing-statistics.html#cb153-10" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb153-11"><a href="sequencing-statistics.html#cb153-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb153-12"><a href="sequencing-statistics.html#cb153-12" tabindex="-1"></a>    <span class="at">label_txt =</span> <span class="fu">paste0</span>(</span>
<span id="cb153-13"><a href="sequencing-statistics.html#cb153-13" tabindex="-1"></a>      <span class="st">&quot;&quot;</span>, <span class="fu">round</span>(median, <span class="dv">1</span>),<span class="st">&quot;%&quot;</span>,</span>
<span id="cb153-14"><a href="sequencing-statistics.html#cb153-14" tabindex="-1"></a>      <span class="st">&quot;</span><span class="sc">\n</span><span class="st">[&quot;</span>, <span class="fu">round</span>(Q1, <span class="dv">1</span>), <span class="st">&quot;–&quot;</span>, <span class="fu">round</span>(Q3, <span class="dv">1</span>), <span class="st">&quot;]&quot;</span>))</span>
<span id="cb153-15"><a href="sequencing-statistics.html#cb153-15" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="11%" />
<col width="11%" />
<col width="13%" />
<col width="13%" />
<col width="13%" />
<col width="21%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">batch</th>
<th align="left">label</th>
<th align="right">Q1</th>
<th align="right">median</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="left">Bacteria</td>
<td align="right">77.1293571</td>
<td align="right">83.2961642</td>
<td align="right">84.5142164</td>
<td align="left">83.3%</td>
</tr>
<tr class="even">
<td align="left">[77.1–84.5]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="left">Turkey</td>
<td align="right">0.0163631</td>
<td align="right">0.0775952</td>
<td align="right">0.6249221</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–0.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="left">Unmapped</td>
<td align="right">14.8680114</td>
<td align="right">15.5989254</td>
<td align="right">16.6786750</td>
<td align="left">15.6%</td>
</tr>
<tr class="even">
<td align="left">[14.9–16.7]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0042</td>
<td align="left">Removed</td>
<td align="right">1.8511632</td>
<td align="right">2.0813832</td>
<td align="right">2.2809274</td>
<td align="left">2.1%</td>
</tr>
<tr class="even">
<td align="left">[1.9–2.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="left">Bacteria</td>
<td align="right">74.7726283</td>
<td align="right">83.6791796</td>
<td align="right">84.5846879</td>
<td align="left">83.7%</td>
</tr>
<tr class="even">
<td align="left">[74.8–84.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="left">Turkey</td>
<td align="right">0.0182617</td>
<td align="right">0.0603251</td>
<td align="right">0.6514980</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–0.7]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="left">Unmapped</td>
<td align="right">14.9959424</td>
<td align="right">15.5211378</td>
<td align="right">17.6947215</td>
<td align="left">15.5%</td>
</tr>
<tr class="even">
<td align="left">[15–17.7]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0043</td>
<td align="left">Removed</td>
<td align="right">1.7752821</td>
<td align="right">1.8905281</td>
<td align="right">2.0392039</td>
<td align="left">1.9%</td>
</tr>
<tr class="even">
<td align="left">[1.8–2]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="left">Bacteria</td>
<td align="right">83.5210896</td>
<td align="right">84.9906296</td>
<td align="right">86.5133880</td>
<td align="left">85%</td>
</tr>
<tr class="even">
<td align="left">[83.5–86.5]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="left">Turkey</td>
<td align="right">0.0080653</td>
<td align="right">0.0274581</td>
<td align="right">0.0989144</td>
<td align="left">0%</td>
</tr>
<tr class="even">
<td align="left">[0–0.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="left">Unmapped</td>
<td align="right">13.2241987</td>
<td align="right">14.3693125</td>
<td align="right">15.4378358</td>
<td align="left">14.4%</td>
</tr>
<tr class="even">
<td align="left">[13.2–15.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0044</td>
<td align="left">Removed</td>
<td align="right">1.5822186</td>
<td align="right">1.7263551</td>
<td align="right">1.9232553</td>
<td align="left">1.7%</td>
</tr>
<tr class="even">
<td align="left">[1.6–1.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="left">Bacteria</td>
<td align="right">82.3508961</td>
<td align="right">84.6193457</td>
<td align="right">85.9756554</td>
<td align="left">84.6%</td>
</tr>
<tr class="even">
<td align="left">[82.4–86]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="left">Turkey</td>
<td align="right">0.0105255</td>
<td align="right">0.0311007</td>
<td align="right">0.0730272</td>
<td align="left">0%</td>
</tr>
<tr class="even">
<td align="left">[0–0.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="left">Unmapped</td>
<td align="right">13.4687550</td>
<td align="right">14.5327848</td>
<td align="right">16.3119726</td>
<td align="left">14.5%</td>
</tr>
<tr class="even">
<td align="left">[13.5–16.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0045</td>
<td align="left">Removed</td>
<td align="right">2.0254291</td>
<td align="right">2.1240288</td>
<td align="right">2.2712657</td>
<td align="left">2.1%</td>
</tr>
<tr class="even">
<td align="left">[2–2.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="left">Bacteria</td>
<td align="right">74.2299750</td>
<td align="right">84.6805031</td>
<td align="right">85.7276876</td>
<td align="left">84.7%</td>
</tr>
<tr class="even">
<td align="left">[74.2–85.7]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="left">Turkey</td>
<td align="right">0.0396768</td>
<td align="right">0.0668880</td>
<td align="right">3.8427631</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–3.8]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="left">Unmapped</td>
<td align="right">12.9973831</td>
<td align="right">14.1442125</td>
<td align="right">14.7835713</td>
<td align="left">14.1%</td>
</tr>
<tr class="even">
<td align="left">[13–14.8]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0046</td>
<td align="left">Removed</td>
<td align="right">1.8415862</td>
<td align="right">2.0623751</td>
<td align="right">2.4170526</td>
<td align="left">2.1%</td>
</tr>
<tr class="even">
<td align="left">[1.8–2.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="left">Bacteria</td>
<td align="right">77.1587525</td>
<td align="right">85.3044997</td>
<td align="right">86.0318982</td>
<td align="left">85.3%</td>
</tr>
<tr class="even">
<td align="left">[77.2–86]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="left">Turkey</td>
<td align="right">0.0204822</td>
<td align="right">0.0510766</td>
<td align="right">1.8557002</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–1.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="left">Unmapped</td>
<td align="right">13.4511761</td>
<td align="right">13.9991858</td>
<td align="right">14.6097289</td>
<td align="left">14%</td>
</tr>
<tr class="even">
<td align="left">[13.5–14.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0047</td>
<td align="left">Removed</td>
<td align="right">1.4889355</td>
<td align="right">1.6335126</td>
<td align="right">2.0240630</td>
<td align="left">1.6%</td>
</tr>
<tr class="even">
<td align="left">[1.5–2]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="left">Bacteria</td>
<td align="right">44.3119121</td>
<td align="right">79.5957342</td>
<td align="right">82.7621054</td>
<td align="left">79.6%</td>
</tr>
<tr class="even">
<td align="left">[44.3–82.8]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="left">Turkey</td>
<td align="right">0.1161374</td>
<td align="right">0.4561889</td>
<td align="right">8.7927736</td>
<td align="left">0.5%</td>
</tr>
<tr class="even">
<td align="left">[0.1–8.8]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="left">Unmapped</td>
<td align="right">15.3743041</td>
<td align="right">16.4974759</td>
<td align="right">19.2672206</td>
<td align="left">16.5%</td>
</tr>
<tr class="even">
<td align="left">[15.4–19.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0048</td>
<td align="left">Removed</td>
<td align="right">1.3131258</td>
<td align="right">1.6093493</td>
<td align="right">1.9238509</td>
<td align="left">1.6%</td>
</tr>
<tr class="even">
<td align="left">[1.3–1.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="left">Bacteria</td>
<td align="right">35.5715021</td>
<td align="right">74.9750661</td>
<td align="right">83.6844949</td>
<td align="left">75%</td>
</tr>
<tr class="even">
<td align="left">[35.6–83.7]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="left">Turkey</td>
<td align="right">0.2431261</td>
<td align="right">1.0566258</td>
<td align="right">13.8758066</td>
<td align="left">1.1%</td>
</tr>
<tr class="even">
<td align="left">[0.2–13.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="left">Unmapped</td>
<td align="right">13.8179520</td>
<td align="right">15.4889199</td>
<td align="right">16.9442993</td>
<td align="left">15.5%</td>
</tr>
<tr class="even">
<td align="left">[13.8–16.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0049</td>
<td align="left">Removed</td>
<td align="right">1.5424711</td>
<td align="right">1.7097963</td>
<td align="right">2.1480213</td>
<td align="left">1.7%</td>
</tr>
<tr class="even">
<td align="left">[1.5–2.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="left">Bacteria</td>
<td align="right">72.2037405</td>
<td align="right">77.3274856</td>
<td align="right">84.0068925</td>
<td align="left">77.3%</td>
</tr>
<tr class="even">
<td align="left">[72.2–84]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="left">Turkey</td>
<td align="right">0.0260250</td>
<td align="right">0.1860261</td>
<td align="right">0.9077713</td>
<td align="left">0.2%</td>
</tr>
<tr class="even">
<td align="left">[0–0.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="left">Unmapped</td>
<td align="right">15.4932886</td>
<td align="right">17.0659670</td>
<td align="right">25.7773520</td>
<td align="left">17.1%</td>
</tr>
<tr class="even">
<td align="left">[15.5–25.8]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0050</td>
<td align="left">Removed</td>
<td align="right">1.9281271</td>
<td align="right">2.1525063</td>
<td align="right">2.3540014</td>
<td align="left">2.2%</td>
</tr>
<tr class="even">
<td align="left">[1.9–2.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="left">Bacteria</td>
<td align="right">64.1898671</td>
<td align="right">73.5891383</td>
<td align="right">83.5954659</td>
<td align="left">73.6%</td>
</tr>
<tr class="even">
<td align="left">[64.2–83.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="left">Turkey</td>
<td align="right">0.1082772</td>
<td align="right">1.5973252</td>
<td align="right">13.5046668</td>
<td align="left">1.6%</td>
</tr>
<tr class="even">
<td align="left">[0.1–13.5]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="left">Unmapped</td>
<td align="right">14.9626748</td>
<td align="right">15.9121371</td>
<td align="right">21.9471895</td>
<td align="left">15.9%</td>
</tr>
<tr class="even">
<td align="left">[15–21.9]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0051</td>
<td align="left">Removed</td>
<td align="right">1.6643115</td>
<td align="right">1.7528041</td>
<td align="right">2.0521254</td>
<td align="left">1.8%</td>
</tr>
<tr class="even">
<td align="left">[1.7–2.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="left">Bacteria</td>
<td align="right">75.4714397</td>
<td align="right">80.3653056</td>
<td align="right">81.4467296</td>
<td align="left">80.4%</td>
</tr>
<tr class="even">
<td align="left">[75.5–81.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="left">Turkey</td>
<td align="right">0.0157989</td>
<td align="right">0.0659510</td>
<td align="right">1.4378359</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–1.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="left">Unmapped</td>
<td align="right">14.0850042</td>
<td align="right">18.4452630</td>
<td align="right">19.2824423</td>
<td align="left">18.4%</td>
</tr>
<tr class="even">
<td align="left">[14.1–19.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0052</td>
<td align="left">Removed</td>
<td align="right">1.4784958</td>
<td align="right">1.7128655</td>
<td align="right">1.9542425</td>
<td align="left">1.7%</td>
</tr>
<tr class="even">
<td align="left">[1.5–2]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="left">Bacteria</td>
<td align="right">75.2843254</td>
<td align="right">80.6497342</td>
<td align="right">82.1695767</td>
<td align="left">80.6%</td>
</tr>
<tr class="even">
<td align="left">[75.3–82.2]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="left">Turkey</td>
<td align="right">0.0077423</td>
<td align="right">0.0697835</td>
<td align="right">1.1262749</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–1.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="left">Unmapped</td>
<td align="right">14.6916246</td>
<td align="right">18.3726727</td>
<td align="right">19.4179476</td>
<td align="left">18.4%</td>
</tr>
<tr class="even">
<td align="left">[14.7–19.4]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0053</td>
<td align="left">Removed</td>
<td align="right">1.2567151</td>
<td align="right">1.4179257</td>
<td align="right">1.5856805</td>
<td align="left">1.4%</td>
</tr>
<tr class="even">
<td align="left">[1.3–1.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="left">Bacteria</td>
<td align="right">80.9539823</td>
<td align="right">81.7838361</td>
<td align="right">83.5732925</td>
<td align="left">81.8%</td>
</tr>
<tr class="even">
<td align="left">[81–83.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="left">Turkey</td>
<td align="right">0.0257158</td>
<td align="right">0.0393262</td>
<td align="right">0.0742836</td>
<td align="left">0%</td>
</tr>
<tr class="even">
<td align="left">[0–0.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="left">Unmapped</td>
<td align="right">15.3496390</td>
<td align="right">17.8094652</td>
<td align="right">18.5849505</td>
<td align="left">17.8%</td>
</tr>
<tr class="even">
<td align="left">[15.3–18.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0054</td>
<td align="left">Removed</td>
<td align="right">1.1057230</td>
<td align="right">1.3786884</td>
<td align="right">1.6344811</td>
<td align="left">1.4%</td>
</tr>
<tr class="even">
<td align="left">[1.1–1.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="left">Bacteria</td>
<td align="right">80.4113285</td>
<td align="right">81.4877636</td>
<td align="right">84.2331777</td>
<td align="left">81.5%</td>
</tr>
<tr class="even">
<td align="left">[80.4–84.2]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="left">Turkey</td>
<td align="right">0.0060460</td>
<td align="right">0.0190133</td>
<td align="right">0.0901665</td>
<td align="left">0%</td>
</tr>
<tr class="even">
<td align="left">[0–0.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="left">Unmapped</td>
<td align="right">15.3220475</td>
<td align="right">18.2083489</td>
<td align="right">19.0241478</td>
<td align="left">18.2%</td>
</tr>
<tr class="even">
<td align="left">[15.3–19]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">MSEB0055</td>
<td align="left">Removed</td>
<td align="right">1.7621776</td>
<td align="right">2.0057958</td>
<td align="right">2.3095357</td>
<td align="left">2%</td>
</tr>
<tr class="even">
<td align="left">[1.8–2.3]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb154"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb154-1"><a href="sequencing-statistics.html#cb154-1" tabindex="-1"></a>boxplot_reads_prc_1_batch <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc <span class="sc">%&gt;%</span></span>
<span id="cb154-2"><a href="sequencing-statistics.html#cb154-2" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb154-3"><a href="sequencing-statistics.html#cb154-3" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>, <span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb154-4"><a href="sequencing-statistics.html#cb154-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> counts, <span class="at">color =</span> label)) <span class="sc">+</span></span>
<span id="cb154-5"><a href="sequencing-statistics.html#cb154-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb154-6"><a href="sequencing-statistics.html#cb154-6" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.05</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb154-7"><a href="sequencing-statistics.html#cb154-7" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> color_palette_tidy_plot_data_stats_counts) <span class="sc">+</span></span>
<span id="cb154-8"><a href="sequencing-statistics.html#cb154-8" tabindex="-1"></a>  <span class="fu">facet_nested</span>(label <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb154-9"><a href="sequencing-statistics.html#cb154-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,<span class="at">y =</span> <span class="st">&quot;Reads %&quot;</span>) <span class="sc">+</span></span>
<span id="cb154-10"><a href="sequencing-statistics.html#cb154-10" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span>  <span class="co"># Remove legend</span></span>
<span id="cb154-11"><a href="sequencing-statistics.html#cb154-11" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb154-12"><a href="sequencing-statistics.html#cb154-12" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb154-13"><a href="sequencing-statistics.html#cb154-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb154-14"><a href="sequencing-statistics.html#cb154-14" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>),</span>
<span id="cb154-15"><a href="sequencing-statistics.html#cb154-15" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb154-16"><a href="sequencing-statistics.html#cb154-16" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.20</span>))) <span class="sc">+</span></span>
<span id="cb154-17"><a href="sequencing-statistics.html#cb154-17" tabindex="-1"></a>    <span class="co"># median + IQR text (placed a bit above the median line)</span></span>
<span id="cb154-18"><a href="sequencing-statistics.html#cb154-18" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb154-19"><a href="sequencing-statistics.html#cb154-19" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb154-20"><a href="sequencing-statistics.html#cb154-20" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb154-21"><a href="sequencing-statistics.html#cb154-21" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb154-22"><a href="sequencing-statistics.html#cb154-22" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.8</span>,</span>
<span id="cb154-23"><a href="sequencing-statistics.html#cb154-23" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb154-24"><a href="sequencing-statistics.html#cb154-24" tabindex="-1"></a></span>
<span id="cb154-25"><a href="sequencing-statistics.html#cb154-25" tabindex="-1"></a>boxplot_reads_prc_1_batch</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_micro_reads_prc_batch-1.png" width="768" /></p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb155-1"><a href="sequencing-statistics.html#cb155-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;counts&quot;</span></span>
<span id="cb155-2"><a href="sequencing-statistics.html#cb155-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc <span class="sc">%&gt;%</span></span>
<span id="cb155-3"><a href="sequencing-statistics.html#cb155-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb155-4"><a href="sequencing-statistics.html#cb155-4" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>,<span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb155-5"><a href="sequencing-statistics.html#cb155-5" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, section, label) <span class="sc">%&gt;%</span> <span class="co">#treatment_expl, age, </span></span>
<span id="cb155-6"><a href="sequencing-statistics.html#cb155-6" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb155-7"><a href="sequencing-statistics.html#cb155-7" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb155-8"><a href="sequencing-statistics.html#cb155-8" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb155-9"><a href="sequencing-statistics.html#cb155-9" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb155-10"><a href="sequencing-statistics.html#cb155-10" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb155-11"><a href="sequencing-statistics.html#cb155-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb155-12"><a href="sequencing-statistics.html#cb155-12" tabindex="-1"></a>    <span class="at">label_txt =</span> <span class="fu">paste0</span>(</span>
<span id="cb155-13"><a href="sequencing-statistics.html#cb155-13" tabindex="-1"></a>      <span class="st">&quot;&quot;</span>, <span class="fu">round</span>(median, <span class="dv">1</span>),<span class="st">&quot;%&quot;</span>,</span>
<span id="cb155-14"><a href="sequencing-statistics.html#cb155-14" tabindex="-1"></a>      <span class="st">&quot;</span><span class="sc">\n</span><span class="st">[&quot;</span>, <span class="fu">round</span>(Q1, <span class="dv">1</span>), <span class="st">&quot;–&quot;</span>, <span class="fu">round</span>(Q3, <span class="dv">1</span>), <span class="st">&quot;]&quot;</span>))</span>
<span id="cb155-15"><a href="sequencing-statistics.html#cb155-15" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(summary_stats)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="15%" />
<col width="10%" />
<col width="13%" />
<col width="13%" />
<col width="12%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">type_simple</th>
<th align="left">section</th>
<th align="left">label</th>
<th align="right">Q1</th>
<th align="right">median</th>
<th align="right">Q3</th>
<th align="left">label_txt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="left">Bacteria</td>
<td align="right">74.3035568</td>
<td align="right">81.8784869</td>
<td align="right">84.566429</td>
<td align="left">81.9%</td>
</tr>
<tr class="even">
<td align="left">[74.3–84.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="left">Turkey</td>
<td align="right">0.0220467</td>
<td align="right">0.0822649</td>
<td align="right">1.003598</td>
<td align="left">0.1%</td>
</tr>
<tr class="even">
<td align="left">[0–1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="left">Unmapped</td>
<td align="right">14.1923635</td>
<td align="right">15.5623896</td>
<td align="right">18.611575</td>
<td align="left">15.6%</td>
</tr>
<tr class="even">
<td align="left">[14.2–18.6]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left">Caecum right</td>
<td align="left">Removed</td>
<td align="right">1.5771308</td>
<td align="right">1.8370861</td>
<td align="right">2.141348</td>
<td align="left">1.8%</td>
</tr>
<tr class="even">
<td align="left">[1.6–2.1]</td>
<td align="left"></td>
<td align="left"></td>
<td align="right"></td>
<td align="right"></td>
<td align="right"></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb156"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb156-1"><a href="sequencing-statistics.html#cb156-1" tabindex="-1"></a>boxplot_reads_prc_1_all <span class="ot">&lt;-</span> tidy_plot_data_stats_counts_prc <span class="sc">%&gt;%</span></span>
<span id="cb156-2"><a href="sequencing-statistics.html#cb156-2" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb156-3"><a href="sequencing-statistics.html#cb156-3" tabindex="-1"></a>  <span class="fu">filter</span>(mapping_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_percentage&quot;</span>, <span class="st">&quot;turkey_percentage&quot;</span>, <span class="st">&quot;unmapped_percentage&quot;</span>, <span class="st">&quot;trimmed_reads_percentage&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb156-4"><a href="sequencing-statistics.html#cb156-4" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> counts, <span class="at">color =</span> label)) <span class="sc">+</span></span>
<span id="cb156-5"><a href="sequencing-statistics.html#cb156-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb156-6"><a href="sequencing-statistics.html#cb156-6" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb156-7"><a href="sequencing-statistics.html#cb156-7" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> color_palette_tidy_plot_data_stats_counts) <span class="sc">+</span></span>
<span id="cb156-8"><a href="sequencing-statistics.html#cb156-8" tabindex="-1"></a>  <span class="fu">facet_nested</span>(label <span class="sc">~</span> ., <span class="at">scales =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">space =</span> <span class="st">&quot;fixed&quot;</span>, <span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb156-9"><a href="sequencing-statistics.html#cb156-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,<span class="at">y =</span> <span class="st">&quot;Reads %&quot;</span>) <span class="sc">+</span></span>
<span id="cb156-10"><a href="sequencing-statistics.html#cb156-10" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span>  <span class="co"># Remove legend</span></span>
<span id="cb156-11"><a href="sequencing-statistics.html#cb156-11" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb156-12"><a href="sequencing-statistics.html#cb156-12" tabindex="-1"></a>  custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb156-13"><a href="sequencing-statistics.html#cb156-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb156-14"><a href="sequencing-statistics.html#cb156-14" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="fl">0.2</span>, <span class="dv">10</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span>),</span>
<span id="cb156-15"><a href="sequencing-statistics.html#cb156-15" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove x-axis title</span></span>
<span id="cb156-16"><a href="sequencing-statistics.html#cb156-16" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),   <span class="co"># remove tick labels #axis.text.x = element_text(angle = 90, hjust = 1)</span></span>
<span id="cb156-17"><a href="sequencing-statistics.html#cb156-17" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),  <span class="co"># remove tick marks</span></span>
<span id="cb156-18"><a href="sequencing-statistics.html#cb156-18" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb156-19"><a href="sequencing-statistics.html#cb156-19" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.21</span>))) <span class="sc">+</span></span>
<span id="cb156-20"><a href="sequencing-statistics.html#cb156-20" tabindex="-1"></a>    <span class="co"># median + IQR text (placed a bit above the median line)</span></span>
<span id="cb156-21"><a href="sequencing-statistics.html#cb156-21" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb156-22"><a href="sequencing-statistics.html#cb156-22" tabindex="-1"></a>    <span class="at">data =</span> summary_stats,</span>
<span id="cb156-23"><a href="sequencing-statistics.html#cb156-23" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> section, <span class="at">y =</span> median, <span class="at">label =</span> label_txt),</span>
<span id="cb156-24"><a href="sequencing-statistics.html#cb156-24" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb156-25"><a href="sequencing-statistics.html#cb156-25" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.3</span>,</span>
<span id="cb156-26"><a href="sequencing-statistics.html#cb156-26" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb156-27"><a href="sequencing-statistics.html#cb156-27" tabindex="-1"></a></span>
<span id="cb156-28"><a href="sequencing-statistics.html#cb156-28" tabindex="-1"></a>boxplot_reads_prc_1_all</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_micro_reads_prc_section-1.png" width="768" /></p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb157-1"><a href="sequencing-statistics.html#cb157-1" tabindex="-1"></a>(boxplot_reads_prc_1_all <span class="sc">+</span> boxplot_reads_prc_1_batch) <span class="sc">+</span></span>
<span id="cb157-2"><a href="sequencing-statistics.html#cb157-2" tabindex="-1"></a> <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">+</span></span>
<span id="cb157-3"><a href="sequencing-statistics.html#cb157-3" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>,</span>
<span id="cb157-4"><a href="sequencing-statistics.html#cb157-4" tabindex="-1"></a>              <span class="at">ncol =</span> <span class="dv">2</span>,        <span class="co"># single column</span></span>
<span id="cb157-5"><a href="sequencing-statistics.html#cb157-5" tabindex="-1"></a>              <span class="at">width =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">8</span>)) <span class="sc">&amp;</span></span>
<span id="cb157-6"><a href="sequencing-statistics.html#cb157-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_micro_reads_prc_combo-1.png" width="960" /></p>
</div>
<div id="collection-attempts" class="section level3 hasAnchor" number="12.3.4">
<h3><span class="header-section-number">12.3.4</span> Collection attempts<a href="sequencing-statistics.html#collection-attempts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb158"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb158-1"><a href="sequencing-statistics.html#cb158-1" tabindex="-1"></a>collection_attempts_plot_simple <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb158-2"><a href="sequencing-statistics.html#cb158-2" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb158-3"><a href="sequencing-statistics.html#cb158-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb158-4"><a href="sequencing-statistics.html#cb158-4" tabindex="-1"></a>  <span class="co"># filter(batch == &quot;MSEB0048&quot;)</span></span>
<span id="cb158-5"><a href="sequencing-statistics.html#cb158-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb158-6"><a href="sequencing-statistics.html#cb158-6" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb158-7"><a href="sequencing-statistics.html#cb158-7" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb158-8"><a href="sequencing-statistics.html#cb158-8" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span></span>
<span id="cb158-9"><a href="sequencing-statistics.html#cb158-9" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb158-10"><a href="sequencing-statistics.html#cb158-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">factor</span>(collection_attempts_text, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Fail&quot;</span>, <span class="st">&quot;Two&quot;</span>, <span class="st">&quot;One&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb158-11"><a href="sequencing-statistics.html#cb158-11" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch,  <span class="at">fill =</span> collection_attempts_text)) <span class="sc">+</span></span>
<span id="cb158-12"><a href="sequencing-statistics.html#cb158-12" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">&quot;# of collection attempts&quot;</span>) <span class="sc">+</span></span>
<span id="cb158-13"><a href="sequencing-statistics.html#cb158-13" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span> <span class="co">#position = &quot;fill&quot;</span></span>
<span id="cb158-14"><a href="sequencing-statistics.html#cb158-14" tabindex="-1"></a>   <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb158-15"><a href="sequencing-statistics.html#cb158-15" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">label =</span> ..count.., </span>
<span id="cb158-16"><a href="sequencing-statistics.html#cb158-16" tabindex="-1"></a>                <span class="at">group =</span> collection_attempts_text),</span>
<span id="cb158-17"><a href="sequencing-statistics.html#cb158-17" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>), </span>
<span id="cb158-18"><a href="sequencing-statistics.html#cb158-18" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span>   <span class="co"># text colour; adjust for contrast</span></span>
<span id="cb158-19"><a href="sequencing-statistics.html#cb158-19" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#685d79&quot;</span>,<span class="st">&quot;#d9727f&quot;</span>,<span class="st">&quot;#fcbb6d&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb158-20"><a href="sequencing-statistics.html#cb158-20" tabindex="-1"></a>    <span class="fu">facet_nested</span>(.  <span class="sc">~</span> type_simple, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>,<span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb158-21"><a href="sequencing-statistics.html#cb158-21" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb158-22"><a href="sequencing-statistics.html#cb158-22" tabindex="-1"></a>    custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb158-23"><a href="sequencing-statistics.html#cb158-23" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb158-24"><a href="sequencing-statistics.html#cb158-24" tabindex="-1"></a>          <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb158-25"><a href="sequencing-statistics.html#cb158-25" tabindex="-1"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb158-26"><a href="sequencing-statistics.html#cb158-26" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)) </span>
<span id="cb158-27"><a href="sequencing-statistics.html#cb158-27" tabindex="-1"></a></span>
<span id="cb158-28"><a href="sequencing-statistics.html#cb158-28" tabindex="-1"></a>collection_attempts_plot_simple</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_collection_attempts-1.png" width="672" /></p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb159-1"><a href="sequencing-statistics.html#cb159-1" tabindex="-1"></a>filter_status_plot_simple <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb159-2"><a href="sequencing-statistics.html#cb159-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch,  <span class="at">fill =</span> filter_status)) <span class="sc">+</span></span>
<span id="cb159-3"><a href="sequencing-statistics.html#cb159-3" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">&quot;# of collection attempts&quot;</span>) <span class="sc">+</span></span>
<span id="cb159-4"><a href="sequencing-statistics.html#cb159-4" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span> <span class="co">#position = &quot;fill&quot;</span></span>
<span id="cb159-5"><a href="sequencing-statistics.html#cb159-5" tabindex="-1"></a>   <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb159-6"><a href="sequencing-statistics.html#cb159-6" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">label =</span> ..count.., </span>
<span id="cb159-7"><a href="sequencing-statistics.html#cb159-7" tabindex="-1"></a>                <span class="at">group =</span> filter_status),</span>
<span id="cb159-8"><a href="sequencing-statistics.html#cb159-8" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>), </span>
<span id="cb159-9"><a href="sequencing-statistics.html#cb159-9" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span>   <span class="co"># text colour; adjust for contrast</span></span>
<span id="cb159-10"><a href="sequencing-statistics.html#cb159-10" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#685d79&quot;</span>,<span class="st">&quot;#d9727f&quot;</span>,<span class="st">&quot;#fcbb6d&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb159-11"><a href="sequencing-statistics.html#cb159-11" tabindex="-1"></a>    <span class="fu">facet_nested</span>(.  <span class="sc">~</span> type_simple, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>,<span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb159-12"><a href="sequencing-statistics.html#cb159-12" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb159-13"><a href="sequencing-statistics.html#cb159-13" tabindex="-1"></a>    custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb159-14"><a href="sequencing-statistics.html#cb159-14" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb159-15"><a href="sequencing-statistics.html#cb159-15" tabindex="-1"></a>          <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb159-16"><a href="sequencing-statistics.html#cb159-16" tabindex="-1"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb159-17"><a href="sequencing-statistics.html#cb159-17" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)) </span>
<span id="cb159-18"><a href="sequencing-statistics.html#cb159-18" tabindex="-1"></a></span>
<span id="cb159-19"><a href="sequencing-statistics.html#cb159-19" tabindex="-1"></a>filter_status_plot_simple <span class="sc">&amp;</span></span>
<span id="cb159-20"><a href="sequencing-statistics.html#cb159-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_filter_status-1.png" width="672" /></p>
</div>
<div id="collection-attempts-vs-filtering-retention" class="section level3 hasAnchor" number="12.3.5">
<h3><span class="header-section-number">12.3.5</span> Collection attempts vs filtering retention<a href="sequencing-statistics.html#collection-attempts-vs-filtering-retention" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb160"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb160-1"><a href="sequencing-statistics.html#cb160-1" tabindex="-1"></a>collection_attempts_plot <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb160-2"><a href="sequencing-statistics.html#cb160-2" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb160-3"><a href="sequencing-statistics.html#cb160-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb160-4"><a href="sequencing-statistics.html#cb160-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb160-5"><a href="sequencing-statistics.html#cb160-5" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb160-6"><a href="sequencing-statistics.html#cb160-6" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb160-7"><a href="sequencing-statistics.html#cb160-7" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span></span>
<span id="cb160-8"><a href="sequencing-statistics.html#cb160-8" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb160-9"><a href="sequencing-statistics.html#cb160-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">factor</span>(collection_attempts_text, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Fail&quot;</span>, <span class="st">&quot;Two&quot;</span>, <span class="st">&quot;One&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb160-10"><a href="sequencing-statistics.html#cb160-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch,  <span class="at">fill =</span> collection_attempts_text)) <span class="sc">+</span></span>
<span id="cb160-11"><a href="sequencing-statistics.html#cb160-11" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">&quot;# of collection attempts&quot;</span>) <span class="sc">+</span></span>
<span id="cb160-12"><a href="sequencing-statistics.html#cb160-12" tabindex="-1"></a>    <span class="fu">geom_bar</span>() <span class="sc">+</span> <span class="co">#position = &quot;fill&quot;</span></span>
<span id="cb160-13"><a href="sequencing-statistics.html#cb160-13" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#685d79&quot;</span>,<span class="st">&quot;#d9727f&quot;</span>,<span class="st">&quot;#fcbb6d&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb160-14"><a href="sequencing-statistics.html#cb160-14" tabindex="-1"></a>    <span class="fu">facet_nested</span>(.  <span class="sc">~</span> filter_status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>,<span class="at">switch =</span> <span class="st">&quot;y&quot;</span>) <span class="sc">+</span></span>
<span id="cb160-15"><a href="sequencing-statistics.html#cb160-15" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb160-16"><a href="sequencing-statistics.html#cb160-16" tabindex="-1"></a>    custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb160-17"><a href="sequencing-statistics.html#cb160-17" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-18"><a href="sequencing-statistics.html#cb160-18" tabindex="-1"></a>          <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb160-19"><a href="sequencing-statistics.html#cb160-19" tabindex="-1"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb160-20"><a href="sequencing-statistics.html#cb160-20" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)) </span>
<span id="cb160-21"><a href="sequencing-statistics.html#cb160-21" tabindex="-1"></a></span>
<span id="cb160-22"><a href="sequencing-statistics.html#cb160-22" tabindex="-1"></a>barplot_collection_reads_after <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb160-23"><a href="sequencing-statistics.html#cb160-23" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb160-24"><a href="sequencing-statistics.html#cb160-24" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb160-25"><a href="sequencing-statistics.html#cb160-25" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb160-26"><a href="sequencing-statistics.html#cb160-26" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb160-27"><a href="sequencing-statistics.html#cb160-27" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb160-28"><a href="sequencing-statistics.html#cb160-28" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span></span>
<span id="cb160-29"><a href="sequencing-statistics.html#cb160-29" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb160-30"><a href="sequencing-statistics.html#cb160-30" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> batch, <span class="at">y =</span> total_sequences_after_trim)) <span class="sc">+</span></span>
<span id="cb160-31"><a href="sequencing-statistics.html#cb160-31" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Batch&quot;</span>, <span class="at">y =</span> <span class="st">&quot;# of reads after trimming&quot;</span>) <span class="sc">+</span></span>
<span id="cb160-32"><a href="sequencing-statistics.html#cb160-32" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb160-33"><a href="sequencing-statistics.html#cb160-33" tabindex="-1"></a>        <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> collection_attempts_text),</span>
<span id="cb160-34"><a href="sequencing-statistics.html#cb160-34" tabindex="-1"></a>                    <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size=</span><span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb160-35"><a href="sequencing-statistics.html#cb160-35" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#685d79&quot;</span>,<span class="st">&quot;#d9727f&quot;</span>,<span class="st">&quot;#fcbb6d&quot;</span>)) <span class="sc">+</span>    </span>
<span id="cb160-36"><a href="sequencing-statistics.html#cb160-36" tabindex="-1"></a>    <span class="fu">facet_nested</span>(. <span class="sc">~</span> filter_status, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb160-37"><a href="sequencing-statistics.html#cb160-37" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb160-38"><a href="sequencing-statistics.html#cb160-38" tabindex="-1"></a>    custom_ggplot_theme <span class="sc">+</span></span>
<span id="cb160-39"><a href="sequencing-statistics.html#cb160-39" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span>  <span class="co"># Remove legend</span></span>
<span id="cb160-40"><a href="sequencing-statistics.html#cb160-40" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb160-41"><a href="sequencing-statistics.html#cb160-41" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-42"><a href="sequencing-statistics.html#cb160-42" tabindex="-1"></a>          <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb160-43"><a href="sequencing-statistics.html#cb160-43" tabindex="-1"></a>          <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb160-44"><a href="sequencing-statistics.html#cb160-44" tabindex="-1"></a>          <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb160-45"><a href="sequencing-statistics.html#cb160-45" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>))   </span>
<span id="cb160-46"><a href="sequencing-statistics.html#cb160-46" tabindex="-1"></a>  </span>
<span id="cb160-47"><a href="sequencing-statistics.html#cb160-47" tabindex="-1"></a>  </span>
<span id="cb160-48"><a href="sequencing-statistics.html#cb160-48" tabindex="-1"></a><span class="co"># collection_attempts_plot / barplot_collection_reads_after +</span></span>
<span id="cb160-49"><a href="sequencing-statistics.html#cb160-49" tabindex="-1"></a><span class="co">#   plot_annotation(tag_levels = &#39;a&#39;) +</span></span>
<span id="cb160-50"><a href="sequencing-statistics.html#cb160-50" tabindex="-1"></a><span class="co">#   plot_layout(guides = &quot;collect&quot;) &amp;</span></span>
<span id="cb160-51"><a href="sequencing-statistics.html#cb160-51" tabindex="-1"></a><span class="co">#   theme(legend.position = &quot;bottom&quot;, legend.box = &quot;vertical&quot;) </span></span>
<span id="cb160-52"><a href="sequencing-statistics.html#cb160-52" tabindex="-1"></a><span class="co">#     # ggtitle(&quot;*Microsample M305574 removed (2 attempts)&quot;)</span></span>
<span id="cb160-53"><a href="sequencing-statistics.html#cb160-53" tabindex="-1"></a></span>
<span id="cb160-54"><a href="sequencing-statistics.html#cb160-54" tabindex="-1"></a></span>
<span id="cb160-55"><a href="sequencing-statistics.html#cb160-55" tabindex="-1"></a>scale_fonts <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb160-56"><a href="sequencing-statistics.html#cb160-56" tabindex="-1"></a></span>
<span id="cb160-57"><a href="sequencing-statistics.html#cb160-57" tabindex="-1"></a>plot_attempts <span class="ot">&lt;-</span> collection_attempts_plot <span class="sc">/</span> barplot_collection_reads_after <span class="sc">+</span></span>
<span id="cb160-58"><a href="sequencing-statistics.html#cb160-58" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">+</span></span>
<span id="cb160-59"><a href="sequencing-statistics.html#cb160-59" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>) <span class="sc">&amp;</span></span>
<span id="cb160-60"><a href="sequencing-statistics.html#cb160-60" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span>) </span>
<span id="cb160-61"><a href="sequencing-statistics.html#cb160-61" tabindex="-1"></a>    <span class="co"># ggtitle(&quot;*Microsample M305574 removed (2 attempts)&quot;)</span></span>
<span id="cb160-62"><a href="sequencing-statistics.html#cb160-62" tabindex="-1"></a></span>
<span id="cb160-63"><a href="sequencing-statistics.html#cb160-63" tabindex="-1"></a></span>
<span id="cb160-64"><a href="sequencing-statistics.html#cb160-64" tabindex="-1"></a></span>
<span id="cb160-65"><a href="sequencing-statistics.html#cb160-65" tabindex="-1"></a>plot_attempts <span class="sc">+</span></span>
<span id="cb160-66"><a href="sequencing-statistics.html#cb160-66" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb160-67"><a href="sequencing-statistics.html#cb160-67" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(</span>
<span id="cb160-68"><a href="sequencing-statistics.html#cb160-68" tabindex="-1"></a>  <span class="at">axis.text  =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts)),   <span class="co"># axis tick labels</span></span>
<span id="cb160-69"><a href="sequencing-statistics.html#cb160-69" tabindex="-1"></a>  <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts)),   <span class="co"># axis titles</span></span>
<span id="cb160-70"><a href="sequencing-statistics.html#cb160-70" tabindex="-1"></a>  <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts)),   <span class="co"># facet labels</span></span>
<span id="cb160-71"><a href="sequencing-statistics.html#cb160-71" tabindex="-1"></a>  <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts)),   <span class="co"># plot titles</span></span>
<span id="cb160-72"><a href="sequencing-statistics.html#cb160-72" tabindex="-1"></a>  <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts)),  <span class="co"># legend text</span></span>
<span id="cb160-73"><a href="sequencing-statistics.html#cb160-73" tabindex="-1"></a>  <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(scale_fonts))  <span class="co"># legend title</span></span>
<span id="cb160-74"><a href="sequencing-statistics.html#cb160-74" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_collection_vs_filter-1.png" width="672" /></p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb161-1"><a href="sequencing-statistics.html#cb161-1" tabindex="-1"></a>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb161-2"><a href="sequencing-statistics.html#cb161-2" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb161-3"><a href="sequencing-statistics.html#cb161-3" tabindex="-1"></a>         type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb161-4"><a href="sequencing-statistics.html#cb161-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb161-5"><a href="sequencing-statistics.html#cb161-5" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb161-6"><a href="sequencing-statistics.html#cb161-6" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb161-7"><a href="sequencing-statistics.html#cb161-7" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span>,</span>
<span id="cb161-8"><a href="sequencing-statistics.html#cb161-8" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;Other&quot;</span></span>
<span id="cb161-9"><a href="sequencing-statistics.html#cb161-9" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb161-10"><a href="sequencing-statistics.html#cb161-10" tabindex="-1"></a>  <span class="fu">count</span>(filter_status, collection_attempts_text) <span class="sc">%&gt;%</span></span>
<span id="cb161-11"><a href="sequencing-statistics.html#cb161-11" tabindex="-1"></a>  <span class="fu">group_by</span>(filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb161-12"><a href="sequencing-statistics.html#cb161-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> <span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb161-13"><a href="sequencing-statistics.html#cb161-13" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code></pre></div>
<pre class="script-output"><code># A tibble: 6 × 4
  filter_status           collection_attempts_text     n percent
  &lt;chr&gt;                   &lt;chr&gt;                    &lt;int&gt;   &lt;dbl&gt;
1 Excluded from filtering Fail                        29   59.2 
2 Excluded from filtering One                         17   34.7 
3 Excluded from filtering Two                          3    6.12
4 Retained by filtering   Fail                       142   12.6 
5 Retained by filtering   One                        743   66.0 
6 Retained by filtering   Two                        241   21.4 </code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb163-1"><a href="sequencing-statistics.html#cb163-1" tabindex="-1"></a>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb163-2"><a href="sequencing-statistics.html#cb163-2" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb163-3"><a href="sequencing-statistics.html#cb163-3" tabindex="-1"></a>         type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb163-4"><a href="sequencing-statistics.html#cb163-4" tabindex="-1"></a>  <span class="fu">count</span>(filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb163-5"><a href="sequencing-statistics.html#cb163-5" tabindex="-1"></a>  <span class="co"># group_by(filter_status) %&gt;%</span></span>
<span id="cb163-6"><a href="sequencing-statistics.html#cb163-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> <span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb163-7"><a href="sequencing-statistics.html#cb163-7" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code></pre></div>
<pre class="script-output"><code># A tibble: 2 × 3
  filter_status               n percent
  &lt;chr&gt;                   &lt;int&gt;   &lt;dbl&gt;
1 Excluded from filtering    49    4.17
2 Retained by filtering    1126   95.8 </code></pre>
<div class="sourceCode" id="cb165"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb165-1"><a href="sequencing-statistics.html#cb165-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_sequences_after_trim&quot;</span></span>
<span id="cb165-2"><a href="sequencing-statistics.html#cb165-2" tabindex="-1"></a>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb165-3"><a href="sequencing-statistics.html#cb165-3" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb165-4"><a href="sequencing-statistics.html#cb165-4" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb165-5"><a href="sequencing-statistics.html#cb165-5" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Excluded from filtering&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb165-6"><a href="sequencing-statistics.html#cb165-6" tabindex="-1"></a>  <span class="fu">select</span>(microsample, filter_status, collection_attempts, collection, total_sequences_after_trim) <span class="sc">%&gt;%</span></span>
<span id="cb165-7"><a href="sequencing-statistics.html#cb165-7" tabindex="-1"></a>  <span class="fu">group_by</span>(filter_status) <span class="sc">%&gt;%</span>  </span>
<span id="cb165-8"><a href="sequencing-statistics.html#cb165-8" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb165-9"><a href="sequencing-statistics.html#cb165-9" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb165-10"><a href="sequencing-statistics.html#cb165-10" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb165-11"><a href="sequencing-statistics.html#cb165-11" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb165-12"><a href="sequencing-statistics.html#cb165-12" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb165-13"><a href="sequencing-statistics.html#cb165-13" tabindex="-1"></a>  )</span></code></pre></div>
<pre class="script-output"><code># A tibble: 1 × 5
  filter_status           median     IQR     Q1      Q3
  &lt;chr&gt;                    &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
1 Excluded from filtering 404160 1432942 265862 1698804</code></pre>
<div class="sourceCode" id="cb167"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb167-1"><a href="sequencing-statistics.html#cb167-1" tabindex="-1"></a>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb167-2"><a href="sequencing-statistics.html#cb167-2" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb167-3"><a href="sequencing-statistics.html#cb167-3" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb167-4"><a href="sequencing-statistics.html#cb167-4" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb167-5"><a href="sequencing-statistics.html#cb167-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb167-6"><a href="sequencing-statistics.html#cb167-6" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb167-7"><a href="sequencing-statistics.html#cb167-7" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb167-8"><a href="sequencing-statistics.html#cb167-8" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb167-9"><a href="sequencing-statistics.html#cb167-9" tabindex="-1"></a>  <span class="fu">select</span>(microsample, filter_status, collection_attempts_text, total_sequences_after_trim) <span class="sc">%&gt;%</span></span>
<span id="cb167-10"><a href="sequencing-statistics.html#cb167-10" tabindex="-1"></a>  <span class="fu">count</span>(collection_attempts_text) <span class="sc">%&gt;%</span></span>
<span id="cb167-11"><a href="sequencing-statistics.html#cb167-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> <span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n))</span></code></pre></div>
<pre class="script-output"><code># A tibble: 3 × 3
  collection_attempts_text     n percent
  &lt;chr&gt;                    &lt;int&gt;   &lt;dbl&gt;
1 Fail                       142    12.6
2 One                        743    66.0
3 Two                        241    21.4</code></pre>
<div class="sourceCode" id="cb169"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb169-1"><a href="sequencing-statistics.html#cb169-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> <span class="st">&quot;total_sequences_after_trim&quot;</span></span>
<span id="cb169-2"><a href="sequencing-statistics.html#cb169-2" tabindex="-1"></a>plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb169-3"><a href="sequencing-statistics.html#cb169-3" tabindex="-1"></a>  <span class="fu">filter</span>(collection_attempts <span class="sc">&gt;</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb169-4"><a href="sequencing-statistics.html#cb169-4" tabindex="-1"></a>  <span class="fu">filter</span>(type_simple <span class="sc">==</span> <span class="st">&quot;P&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb169-5"><a href="sequencing-statistics.html#cb169-5" tabindex="-1"></a>  <span class="fu">filter</span>(filter_status <span class="sc">==</span> <span class="st">&quot;Retained by filtering&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb169-6"><a href="sequencing-statistics.html#cb169-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_attempts_text =</span> <span class="fu">case_when</span>(</span>
<span id="cb169-7"><a href="sequencing-statistics.html#cb169-7" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Fail&quot;</span>,</span>
<span id="cb169-8"><a href="sequencing-statistics.html#cb169-8" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;One&quot;</span>,</span>
<span id="cb169-9"><a href="sequencing-statistics.html#cb169-9" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Two&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb169-10"><a href="sequencing-statistics.html#cb169-10" tabindex="-1"></a>  <span class="fu">select</span>(microsample, filter_status, collection_attempts_text, total_sequences_after_trim) <span class="sc">%&gt;%</span></span>
<span id="cb169-11"><a href="sequencing-statistics.html#cb169-11" tabindex="-1"></a>  <span class="fu">group_by</span>(collection_attempts_text) <span class="sc">%&gt;%</span>  </span>
<span id="cb169-12"><a href="sequencing-statistics.html#cb169-12" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb169-13"><a href="sequencing-statistics.html#cb169-13" tabindex="-1"></a>    <span class="at">median =</span> <span class="fu">median</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb169-14"><a href="sequencing-statistics.html#cb169-14" tabindex="-1"></a>    <span class="at">IQR    =</span> <span class="fu">IQR</span>(.data[[var]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb169-15"><a href="sequencing-statistics.html#cb169-15" tabindex="-1"></a>    <span class="at">Q1     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb169-16"><a href="sequencing-statistics.html#cb169-16" tabindex="-1"></a>    <span class="at">Q3     =</span> <span class="fu">quantile</span>(.data[[var]], <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb169-17"><a href="sequencing-statistics.html#cb169-17" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb169-18"><a href="sequencing-statistics.html#cb169-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label_txt =</span> <span class="fu">paste0</span>(<span class="st">&quot;&quot;</span>, <span class="fu">format</span>(median, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">2</span>)))</span></code></pre></div>
<pre class="script-output"><code># A tibble: 3 × 6
  collection_attempts_text   median      IQR       Q1       Q3 label_txt
  &lt;chr&gt;                       &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;    
1 Fail                     22489991 25385741  5015120 30400861 2.2e+07  
2 One                      28437666 27578028 17203632 44781660 2.8e+07  
3 Two                      29158200 29214266 17931182 47145448 2.9e+07  </code></pre>
<div class="sourceCode" id="cb171"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb171-1"><a href="sequencing-statistics.html#cb171-1" tabindex="-1"></a><span class="co"># Select data &amp; pivot to wide df</span></span>
<span id="cb171-2"><a href="sequencing-statistics.html#cb171-2" tabindex="-1"></a>neg_vs_pos_wide <span class="ot">&lt;-</span> tidy_plot_genome_counts <span class="sc">%&gt;%</span></span>
<span id="cb171-3"><a href="sequencing-statistics.html#cb171-3" tabindex="-1"></a>  <span class="co"># arrange samples: first negatives, then positives (for ALDEx2).</span></span>
<span id="cb171-4"><a href="sequencing-statistics.html#cb171-4" tabindex="-1"></a>  <span class="fu">arrange</span>(type_simple) <span class="sc">%&gt;%</span> </span>
<span id="cb171-5"><a href="sequencing-statistics.html#cb171-5" tabindex="-1"></a>  <span class="fu">select</span>(genome, microsample, count) <span class="sc">%&gt;%</span></span>
<span id="cb171-6"><a href="sequencing-statistics.html#cb171-6" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb171-7"><a href="sequencing-statistics.html#cb171-7" tabindex="-1"></a>    <span class="at">names_from =</span> genome,   </span>
<span id="cb171-8"><a href="sequencing-statistics.html#cb171-8" tabindex="-1"></a>    <span class="at">values_from =</span> count,   </span>
<span id="cb171-9"><a href="sequencing-statistics.html#cb171-9" tabindex="-1"></a>    <span class="at">values_fill =</span> <span class="dv">0</span></span>
<span id="cb171-10"><a href="sequencing-statistics.html#cb171-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb171-11"><a href="sequencing-statistics.html#cb171-11" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">any</span>(. <span class="sc">!=</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb171-12"><a href="sequencing-statistics.html#cb171-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span>microsample, <span class="sc">~</span> . <span class="sc">/</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric))) <span class="sc">*</span> <span class="dv">100</span> )) <span class="sc">%&gt;%</span></span>
<span id="cb171-13"><a href="sequencing-statistics.html#cb171-13" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;microsample&quot;</span>)</span>
<span id="cb171-14"><a href="sequencing-statistics.html#cb171-14" tabindex="-1"></a></span>
<span id="cb171-15"><a href="sequencing-statistics.html#cb171-15" tabindex="-1"></a><span class="co"># Remove taxa in few samples, and samples with few taxa</span></span>
<span id="cb171-16"><a href="sequencing-statistics.html#cb171-16" tabindex="-1"></a>neg_vs_pos_wide <span class="ot">&lt;-</span> <span class="fu">remove_samples_or_taxa</span>(<span class="at">df =</span> neg_vs_pos_wide,</span>
<span id="cb171-17"><a href="sequencing-statistics.html#cb171-17" tabindex="-1"></a>                                     <span class="at">min_samples_per_taxon =</span> <span class="dv">10</span>,</span>
<span id="cb171-18"><a href="sequencing-statistics.html#cb171-18" tabindex="-1"></a>                                     <span class="at">min_taxa_per_sample =</span> <span class="dv">10</span>)</span></code></pre></div>
<pre class="script-output"><code>Initial df: Rows (samples): 1344 , Columns (taxa): 526 
Removed: Rows (samples): 0 , Columns (taxa): 0 
Resulting df: Rows (samples): 1344 , Columns (taxa): 526 </code></pre>
<div class="sourceCode" id="cb173"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb173-1"><a href="sequencing-statistics.html#cb173-1" tabindex="-1"></a><span class="co"># Remove taxa with less than 0.1% rel. abun. in all samples</span></span>
<span id="cb173-2"><a href="sequencing-statistics.html#cb173-2" tabindex="-1"></a><span class="co"># neg_vs_pos_wide &lt;- neg_vs_pos_wide %&gt;% select(where(~ any(. &gt;= 0.1)))</span></span></code></pre></div>
<div class="sourceCode" id="cb174"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb174-1"><a href="sequencing-statistics.html#cb174-1" tabindex="-1"></a><span class="co"># Perform PCA &amp; make PCA plot</span></span>
<span id="cb174-2"><a href="sequencing-statistics.html#cb174-2" tabindex="-1"></a>neg_pos_PCA <span class="ot">&lt;-</span> <span class="fu">perform_pca</span>(neg_vs_pos_wide, <span class="at">z_delete =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre class="script-output"><code>[1] &quot;Zeros found&quot;</code></pre>
<pre class="script-output"><code>No. adjusted imputations:  70993 
Rows (samples) removed after zero replacement: 27 
Columns (taxa) removed after zero replacement: 0 </code></pre>
<div class="sourceCode" id="cb177"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb177-1"><a href="sequencing-statistics.html#cb177-1" tabindex="-1"></a>plot_data_stats_temp <span class="ot">&lt;-</span> plot_data_stats <span class="sc">%&gt;%</span></span>
<span id="cb177-2"><a href="sequencing-statistics.html#cb177-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">collection_type_simple =</span> <span class="fu">case_when</span> (</span>
<span id="cb177-3"><a href="sequencing-statistics.html#cb177-3" tabindex="-1"></a>    type_simple <span class="sc">==</span> <span class="st">&quot;N&quot;</span> <span class="sc">~</span> <span class="st">&quot;Negative Sample&quot;</span>,</span>
<span id="cb177-4"><a href="sequencing-statistics.html#cb177-4" tabindex="-1"></a>    collection <span class="sc">==</span> <span class="st">&quot;Fail&quot;</span> <span class="sc">~</span> <span class="st">&quot;Microsample: Failed Collection&quot;</span>,</span>
<span id="cb177-5"><a href="sequencing-statistics.html#cb177-5" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">&quot;Microsample: One Collection&quot;</span>,</span>
<span id="cb177-6"><a href="sequencing-statistics.html#cb177-6" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;Microsample: One Collection&quot;</span>,</span>
<span id="cb177-7"><a href="sequencing-statistics.html#cb177-7" tabindex="-1"></a>    collection_attempts <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">&quot;Microsample: Two Collections&quot;</span>))</span>
<span id="cb177-8"><a href="sequencing-statistics.html#cb177-8" tabindex="-1"></a></span>
<span id="cb177-9"><a href="sequencing-statistics.html#cb177-9" tabindex="-1"></a></span>
<span id="cb177-10"><a href="sequencing-statistics.html#cb177-10" tabindex="-1"></a>neg_pos_pca_plot <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(neg_pos_PCA<span class="sc">$</span>pca_result,</span>
<span id="cb177-11"><a href="sequencing-statistics.html#cb177-11" tabindex="-1"></a>  <span class="at">samples_color_metadata =</span> <span class="st">&quot;collection_type_simple&quot;</span>,</span>
<span id="cb177-12"><a href="sequencing-statistics.html#cb177-12" tabindex="-1"></a>  <span class="at">samples_shape_metadata =</span> <span class="st">&quot;filter_status&quot;</span>,</span>
<span id="cb177-13"><a href="sequencing-statistics.html#cb177-13" tabindex="-1"></a>  <span class="at">samples_color_value =</span> <span class="fu">c</span>(<span class="st">&quot;#685d79&quot;</span>,<span class="st">&quot;#fcbb6d&quot;</span>, <span class="st">&quot;#d9727f&quot;</span>, <span class="st">&quot;#3cb2bd&quot;</span>),</span>
<span id="cb177-14"><a href="sequencing-statistics.html#cb177-14" tabindex="-1"></a>  <span class="at">loadings_color_metadata =</span> <span class="st">&quot;order&quot;</span>,</span>
<span id="cb177-15"><a href="sequencing-statistics.html#cb177-15" tabindex="-1"></a>  <span class="at">loadings_color_value =</span> order_colors,</span>
<span id="cb177-16"><a href="sequencing-statistics.html#cb177-16" tabindex="-1"></a>  <span class="at">loadings_taxon_level =</span> <span class="st">&quot;species&quot;</span>,</span>
<span id="cb177-17"><a href="sequencing-statistics.html#cb177-17" tabindex="-1"></a>  <span class="at">sample_metadata =</span> plot_data_stats_temp,</span>
<span id="cb177-18"><a href="sequencing-statistics.html#cb177-18" tabindex="-1"></a>  <span class="at">genome_metadata =</span> genome_metadata,</span>
<span id="cb177-19"><a href="sequencing-statistics.html#cb177-19" tabindex="-1"></a>  order_colors,</span>
<span id="cb177-20"><a href="sequencing-statistics.html#cb177-20" tabindex="-1"></a>  custom_ggplot_theme,</span>
<span id="cb177-21"><a href="sequencing-statistics.html#cb177-21" tabindex="-1"></a>  <span class="at">loadings_number =</span> <span class="dv">20</span> <span class="co"># nrow(neg_vs_pos_ALDEx2_significant)</span></span>
<span id="cb177-22"><a href="sequencing-statistics.html#cb177-22" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb177-23"><a href="sequencing-statistics.html#cb177-23" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb177-24"><a href="sequencing-statistics.html#cb177-24" tabindex="-1"></a></span>
<span id="cb177-25"><a href="sequencing-statistics.html#cb177-25" tabindex="-1"></a>neg_pos_pca_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/plot_seq_stats_pca-1.png" width="960" /></p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb178-1"><a href="sequencing-statistics.html#cb178-1" tabindex="-1"></a><span class="co"># Calculate Euclidean distance on CLR data</span></span>
<span id="cb178-2"><a href="sequencing-statistics.html#cb178-2" tabindex="-1"></a>neg_pos_dist_matrix <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(neg_pos_PCA<span class="sc">$</span>df_clr_dist, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb178-3"><a href="sequencing-statistics.html#cb178-3" tabindex="-1"></a></span>
<span id="cb178-4"><a href="sequencing-statistics.html#cb178-4" tabindex="-1"></a><span class="co"># Make sure that filtered df and metadata have the same number &amp; order of rows</span></span>
<span id="cb178-5"><a href="sequencing-statistics.html#cb178-5" tabindex="-1"></a><span class="co"># Get the exact order of rownames from the PCA data</span></span>
<span id="cb178-6"><a href="sequencing-statistics.html#cb178-6" tabindex="-1"></a>pca_rownames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(neg_pos_PCA<span class="sc">$</span>df_clr_dist)</span>
<span id="cb178-7"><a href="sequencing-statistics.html#cb178-7" tabindex="-1"></a></span>
<span id="cb178-8"><a href="sequencing-statistics.html#cb178-8" tabindex="-1"></a><span class="co"># Filter and reorder metadata to match exactly</span></span>
<span id="cb178-9"><a href="sequencing-statistics.html#cb178-9" tabindex="-1"></a>neg_pos_metadata_adonis <span class="ot">&lt;-</span> plot_data_stats_temp <span class="sc">%&gt;%</span></span>
<span id="cb178-10"><a href="sequencing-statistics.html#cb178-10" tabindex="-1"></a>  <span class="fu">filter</span>(microsample <span class="sc">%in%</span> pca_rownames) <span class="sc">%&gt;%</span></span>
<span id="cb178-11"><a href="sequencing-statistics.html#cb178-11" tabindex="-1"></a>  <span class="co"># Create a factor with levels in the exact order of pca_rownames, then arrange</span></span>
<span id="cb178-12"><a href="sequencing-statistics.html#cb178-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">microsample_ordered =</span> <span class="fu">factor</span>(microsample, <span class="at">levels =</span> pca_rownames)) <span class="sc">%&gt;%</span></span>
<span id="cb178-13"><a href="sequencing-statistics.html#cb178-13" tabindex="-1"></a>  <span class="fu">arrange</span>(microsample_ordered) <span class="sc">%&gt;%</span></span>
<span id="cb178-14"><a href="sequencing-statistics.html#cb178-14" tabindex="-1"></a>  <span class="fu">select</span>(microsample, animal, treatment, age_category, filter_status, collection_type_simple)</span>
<span id="cb178-15"><a href="sequencing-statistics.html#cb178-15" tabindex="-1"></a></span>
<span id="cb178-16"><a href="sequencing-statistics.html#cb178-16" tabindex="-1"></a><span class="co"># Verify the order matches</span></span>
<span id="cb178-17"><a href="sequencing-statistics.html#cb178-17" tabindex="-1"></a><span class="fu">all</span>(neg_pos_metadata_adonis<span class="sc">$</span>microsample <span class="sc">==</span> pca_rownames)</span></code></pre></div>
<pre class="script-output"><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb180"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb180-1"><a href="sequencing-statistics.html#cb180-1" tabindex="-1"></a><span class="co"># Run PERMANOVA</span></span>
<span id="cb180-2"><a href="sequencing-statistics.html#cb180-2" tabindex="-1"></a>adonis2_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(</span>
<span id="cb180-3"><a href="sequencing-statistics.html#cb180-3" tabindex="-1"></a>  neg_pos_dist_matrix <span class="sc">~</span>  animal <span class="sc">+</span> filter_status <span class="sc">+</span> collection_type_simple ,</span>
<span id="cb180-4"><a href="sequencing-statistics.html#cb180-4" tabindex="-1"></a>  <span class="at">data =</span> neg_pos_metadata_adonis,</span>
<span id="cb180-5"><a href="sequencing-statistics.html#cb180-5" tabindex="-1"></a>  <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb180-6"><a href="sequencing-statistics.html#cb180-6" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">&quot;terms&quot;</span></span>
<span id="cb180-7"><a href="sequencing-statistics.html#cb180-7" tabindex="-1"></a>)</span>
<span id="cb180-8"><a href="sequencing-statistics.html#cb180-8" tabindex="-1"></a></span>
<span id="cb180-9"><a href="sequencing-statistics.html#cb180-9" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(adonis2_result)</span></code></pre></div>
<table>
<colgroup>
<col width="34%" />
<col width="7%" />
<col width="16%" />
<col width="14%" />
<col width="16%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Df</th>
<th align="right">SumOfSqs</th>
<th align="right">R2</th>
<th align="right">F</th>
<th align="right">Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">animal</td>
<td align="right">13</td>
<td align="right">1249612.98</td>
<td align="right">0.4935672</td>
<td align="right">111.680032</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">filter_status</td>
<td align="right">1</td>
<td align="right">149476.65</td>
<td align="right">0.0590397</td>
<td align="right">173.666766</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">collection_type_simple</td>
<td align="right">3</td>
<td align="right">14647.48</td>
<td align="right">0.0057854</td>
<td align="right">5.672637</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">Residual</td>
<td align="right">1299</td>
<td align="right">1118061.76</td>
<td align="right">0.4416077</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">Total</td>
<td align="right">1316</td>
<td align="right">2531798.87</td>
<td align="right">1.0000000</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb181"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb181-1"><a href="sequencing-statistics.html#cb181-1" tabindex="-1"></a>plot_data_stats_temp <span class="sc">%&gt;%</span></span>
<span id="cb181-2"><a href="sequencing-statistics.html#cb181-2" tabindex="-1"></a>  <span class="fu">select</span>(microsample, type_simple, filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb181-3"><a href="sequencing-statistics.html#cb181-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple) <span class="sc">%&gt;%</span></span>
<span id="cb181-4"><a href="sequencing-statistics.html#cb181-4" tabindex="-1"></a>  <span class="fu">count</span>(filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb181-5"><a href="sequencing-statistics.html#cb181-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> <span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n))</span></code></pre></div>
<pre class="script-output"><code># A tibble: 4 × 4
# Groups:   type_simple [2]
  type_simple filter_status               n percent
  &lt;chr&gt;       &lt;chr&gt;                   &lt;int&gt;   &lt;dbl&gt;
1 N           Excluded from filtering   153   91.1 
2 N           Retained by filtering      15    8.93
3 P           Excluded from filtering    49    4.17
4 P           Retained by filtering    1127   95.8 </code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb183-1"><a href="sequencing-statistics.html#cb183-1" tabindex="-1"></a>plot_data_stats_temp <span class="sc">%&gt;%</span></span>
<span id="cb183-2"><a href="sequencing-statistics.html#cb183-2" tabindex="-1"></a>  <span class="fu">select</span>(microsample, type_simple, collection, filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb183-3"><a href="sequencing-statistics.html#cb183-3" tabindex="-1"></a>  <span class="fu">group_by</span>(type_simple, collection) <span class="sc">%&gt;%</span></span>
<span id="cb183-4"><a href="sequencing-statistics.html#cb183-4" tabindex="-1"></a>  <span class="fu">count</span>(filter_status) <span class="sc">%&gt;%</span></span>
<span id="cb183-5"><a href="sequencing-statistics.html#cb183-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> <span class="dv">100</span> <span class="sc">*</span> n <span class="sc">/</span> <span class="fu">sum</span>(n))</span></code></pre></div>
<pre class="script-output"><code># A tibble: 11 × 5
# Groups:   type_simple, collection [6]
   type_simple collection filter_status               n percent
   &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;                   &lt;int&gt;   &lt;dbl&gt;
 1 N           Fail       Excluded from filtering    12   75   
 2 N           Fail       Retained by filtering       4   25   
 3 N           Success    Excluded from filtering    36   90   
 4 N           Success    Retained by filtering       4   10   
 5 N           &lt;NA&gt;       Excluded from filtering   105   93.8 
 6 N           &lt;NA&gt;       Retained by filtering       7    6.25
 7 P           Fail       Excluded from filtering    29   17.0 
 8 P           Fail       Retained by filtering     142   83.0 
 9 P           Success    Excluded from filtering    20    1.99
10 P           Success    Retained by filtering     984   98.0 
11 P           &lt;NA&gt;       Retained by filtering       1  100   </code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="histomonas-presence.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bacterial-load.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
